<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Kushal K Dey" />


<title>Circadian comparison with ash</title>

<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>




<link rel="stylesheet" href="cosmo.css" type="text/css" />

</head>

<body>


<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <link href="https://fonts.googleapis.com/css?family=Open+Sans:100,300,400,600" rel="stylesheet" type="text/css">
    <link href="../docs/cosmo.css" rel="stylesheet">
    <link href="../docs/cosmo.min.css" rel="stylesheet">
</head>
<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script>
<script src="../docs/bootstrap/js/bootstrap.min.js"></script>

<body>

<div class = "nav" align = "middle">
<div class= "p1"></div>
<p>
  <a class="btn btn-large btn btn-primary" type="button" href = "corshrink_front.html">HOME</a> 
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "workflow.html">WORKFLOW</a> 
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "tissue_corr.html">GTEx DEMO</a> 
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "about.html">METHODS</a> 
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "https://github.com/kkdey/CorShrink">SOFTWARE</a> 
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "license.html">LICENSE</a> 
   &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "https://github.com/kkdey/CorShrink_workflow">GITHUB</a> 
 </p>
</div>  
<!-- <div class="nav">
  <a href="index.html">Home page</a> &nbsp &nbsp
  <a href="musings.html">Musings</a> &nbsp &nbsp
  <a href="town.html">My town</a> &nbsp &nbsp
  <a href="links.html">Links</a>
</div> -->

<!-- Main content -->
</body>
</html>


<h1 class="title toc-ignore">Circadian comparison with ash</h1>
<h4 class="author"><em>Kushal K Dey</em></h4>
<h4 class="date"><em>6/8/2017</em></h4>


<div id="TOC">
<ul>
<li><a href="#lfsr-threshold-0.2">lfsr threshold 0.2</a></li>
<li><a href="#lfsr-threshold-0.5">lfsr threshold 0.5</a></li>
<li><a href="#review---important-genes">Review - Important genes</a></li>
<li><a href="#ensg00000133794-or-arntl">ENSG00000133794 or ARNTL</a></li>
<li><a href="#ensg00000132326-or-per2">ENSG00000132326 or PER2</a></li>
<li><a href="#ensg00000159208-or-ciart">ENSG00000159208 or CIART</a></li>
<li><a href="#ensg00000101194-or-slc17a7">ENSG00000101194 or SLC17A7</a></li>
</ul>
</div>

<p>In this script, we look at the death time comparison problem in the context of using lfsr for the adaptive shrinkage on the tissues for each gene. Adaptive shrinkage is performed using a normal distribution centered about 0.</p>
<pre class="r"><code>ash_cor_mat &lt;- get(load(&quot;../output/ash_cor_only_voom_pearson_halfuniform_gtex_tissues.rda&quot;))</code></pre>
<pre class="r"><code>common_samples &lt;- get(load(&quot;../output/common_samples.rda&quot;))
tissue_labels &lt;- read.table(file = &quot;../data/GTEX_V6/samples_id.txt&quot;)[,3]

#library(data.table)
#data &lt;- data.frame(fread(&quot;../data/GTEX_V6/cis_gene_expression.txt&quot;))
#matdata &lt;- t(data[,-c(1,2)])

gene_names &lt;- as.character(read.table(file = &quot;../data/GTEX_V6/gene_names_GTEX_V6.txt&quot;)[,1])
gene_names_1 &lt;- as.character(sapply(gene_names, function(x) return(strsplit(x, &quot;[.]&quot;)[[1]][1])))


U &lt;- unique(tissue_labels)</code></pre>
<pre class="r"><code>person_tissue_genes &lt;- get(load(&quot;../output/person_tissue_genes_voom.rda&quot;))
betahat_sebetahat &lt;- get(load(&quot;../output/death_time_cor_ash.rda&quot;))
sample_attributes &lt;- get(load(&quot;../output/sample_attributes_filtered.rda&quot;))
person_tissue_genes &lt;- get(load(&quot;../output/person_tissue_genes_voom.rda&quot;))</code></pre>
<pre class="r"><code>betahat_mat &lt;- betahat_sebetahat$post.mean
sebetahat_mat &lt;- betahat_sebetahat$post.sd
lfsr_mat &lt;- betahat_sebetahat$post.lfsr</code></pre>
<div id="lfsr-threshold-0.2" class="section level2">
<h2>lfsr threshold 0.2</h2>
<p>We look at which genes have the most number of lfsr values below a chosen threshold.</p>
<pre class="r"><code>num_lfsr_below_thresh &lt;- apply(lfsr_mat, 2, function(x) return(length(which(x &lt; 0.3))))

imp_genes &lt;- order(num_lfsr_below_thresh, decreasing = TRUE)[1:10]
num_lfsr_below_thresh[imp_genes]</code></pre>
<pre><code>## ENSG00000133794 ENSG00000132326 ENSG00000159208 ENSG00000101194 
##              19              12              11               7 
## ENSG00000255513 ENSG00000272375 ENSG00000049246 ENSG00000093072 
##               7               5               4               4 
## ENSG00000100395 ENSG00000105516 
##               4               4</code></pre>
<pre class="r"><code>names &lt;- colnames(betahat_mat)[imp_genes]</code></pre>
<pre class="r"><code>write.table(names, file = &quot;../utilities/circadian_ash/circadian_genes_ash_0_3.txt&quot;, quote = FALSE, row.names = FALSE, col.names = FALSE)
out &lt;- mygene::queryMany(names,  scopes=&quot;ensembl.gene&quot;, fields=c(&quot;name&quot;, &quot;summary&quot;), species=&quot;human&quot;);</code></pre>
<pre><code>## Finished</code></pre>
<pre class="r"><code>cbind.data.frame(out$name, out$summary)</code></pre>
<pre><code>##                                               out$name
## 1  aryl hydrocarbon receptor nuclear translocator like
## 2                             period circadian clock 2
## 3      circadian associated repressor of transcription
## 4                    solute carrier family 17 member 9
## 5                                                 &lt;NA&gt;
## 6                                                 &lt;NA&gt;
## 7                             period circadian clock 3
## 8                                adenosine deaminase 2
## 9       L3MBTL2, polycomb repressive complex 1 subunit
## 10         D-box binding PAR bZIP transcription factor
##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     out$summary
## 1  The protein encoded by this gene is a basic helix-loop-helix protein that forms a heterodimer with CLOCK. This heterodimer binds E-box enhancer elements upstream of Period (PER1, PER2, PER3) and Cryptochrome (CRY1, CRY2) genes and activates transcription of these genes. PER and CRY proteins heterodimerize and repress their own transcription by interacting in a feedback loop with CLOCK/ARNTL complexes. Defects in this gene have been linked to infertility, problems with gluconeogenesis and lipogenesis, and altered sleep patterns. Several transcript variants encoding different isoforms have been found for this gene.
## 2                                               This gene is a member of the Period family of genes and is expressed in a circadian pattern in the suprachiasmatic nucleus, the primary circadian pacemaker in the mammalian brain. Genes in this family encode components of the circadian rhythms of locomotor activity, metabolism, and behavior. This gene is upregulated by CLOCK/ARNTL heterodimers but then represses this upregulation in a feedback loop using PER/CRY heterodimers to interact with CLOCK/ARNTL. Polymorphisms in this gene may increase the risk of getting certain cancers and have been linked to sleep disorders.
## 3                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          &lt;NA&gt;
## 4                                                                                                                                                                                      This gene encodes a member of a family of transmembrane proteins that are involved in the transport of small molecules. The encoded protein participates in the vesicular uptake, storage, and secretion of adenoside triphosphate (ATP) and other nucleotides. A mutation in this gene was found in individuals with autosomal dominant disseminated superficial actinic porokeratosis-8. Alternative splicing results in multiple transcript variants.
## 5                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          &lt;NA&gt;
## 6                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          &lt;NA&gt;
## 7            This gene is a member of the Period family of genes and is expressed in a circadian pattern in the suprachiasmatic nucleus, the primary circadian pacemaker in the mammalian brain. Genes in this family encode components of the circadian rhythms of locomotor activity, metabolism, and behavior. This gene is upregulated by CLOCK/ARNTL heterodimers but then represses this upregulation in a feedback loop using PER/CRY heterodimers to interact with CLOCK/ARNTL. Polymorphisms in this gene have been linked to sleep disorders. Multiple transcript variants encoding different isoforms have been found for this gene.
## 8                                                                                                                  This gene encodes a member of a subfamily of the adenosine deaminase protein family. The encoded protein is one of two adenosine deaminases found in humans, which regulate levels of the signaling molecule, adenosine. The encoded protein is secreted from monocytes undergoing differentiation and may regulate cell proliferation and differentiation. This gene may be responsible for some of the phenotypic features associated with cat eye syndrome. Alternative splicing results in multiple transcript variants.
## 9                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          &lt;NA&gt;
## 10                                                                                                                                                                                                                                                                                                                The protein encoded by this gene is a member of the PAR bZIP transcription factor family and binds to specific sequences in the promoters of several genes, such as albumin, CYP2A4, and CYP2A5. The encoded protein can bind DNA as a homo- or heterodimer and is involved in the regulation of some circadian rhythm genes.</code></pre>
<pre class="r"><code>pathway2 &lt;- read.delim(&quot;../utilities/circadian_ash/circadian_genes_ash_0_3_pathways.tab&quot;)
if(length(which(pathway2$q.value &lt; 0.05)) &gt; 2){
  pathway2 &lt;- pathway2[which(pathway2$q.value &lt; 0.05),]
}
maxpath = 75
pathway &lt;- pathway2[1:min(maxpath, dim(pathway2)[1]), ]
ensembl_genes &lt;- pathway$members_input_overlap
pathway_list_genes &lt;- sapply(ensembl_genes, function(x) return(strsplit(as.character(x), &quot;; &quot;)))
all_genes &lt;- Reduce(union, pathway_list_genes)
pathway_mat &lt;- matrix(0, length(pathway_list_genes), length(all_genes))
for(l in 1:dim(pathway_mat)[1]){
  pathway_mat[l, match(pathway_list_genes[[l]], all_genes)] &lt;- 1
}

pathway_bimat &lt;- (pathway_mat)%*%t(pathway_mat)
pathway_bimat_prop &lt;- diag(1/sqrt(diag(pathway_bimat)), dim(pathway_bimat)[1]) %*% pathway_bimat %*% diag(1/sqrt(diag(pathway_bimat)), dim(pathway_bimat)[1])

col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
      rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
heatmap(pathway_bimat_prop, labRow = pathway$pathway, labCol = pathway$pathway, cexRow = 1, cexCol = 1, col = col, zlim = c(-1,1),
        scale = &quot;none&quot;, margins = c(5, 5))</code></pre>
<p><img src="circadian_compare_study_ash_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
</div>
<div id="lfsr-threshold-0.5" class="section level2">
<h2>lfsr threshold 0.5</h2>
<pre class="r"><code>num_lfsr_below_thresh &lt;- apply(lfsr_mat, 2, function(x) return(length(which(x &lt; 0.5))))

imp_genes &lt;- order(num_lfsr_below_thresh, decreasing = TRUE)[1:10]
num_lfsr_below_thresh[imp_genes]</code></pre>
<pre><code>## ENSG00000133794 ENSG00000132326 ENSG00000159208 ENSG00000101194 
##              24              17              13               9 
## ENSG00000255513 ENSG00000093072 ENSG00000105516 ENSG00000204248 
##               9               8               7               7 
## ENSG00000100372 ENSG00000126368 
##               6               6</code></pre>
<pre class="r"><code>names &lt;- colnames(betahat_mat)[imp_genes]</code></pre>
<pre class="r"><code>write.table(names, file = &quot;../utilities/circadian_ash/circadian_genes_ash_0_5.txt&quot;, quote = FALSE, row.names = FALSE, col.names = FALSE)
out &lt;- mygene::queryMany(names,  scopes=&quot;ensembl.gene&quot;, fields=c(&quot;name&quot;, &quot;summary&quot;), species=&quot;human&quot;);</code></pre>
<pre><code>## Finished</code></pre>
<pre class="r"><code>cbind.data.frame(out$name, out$summary)</code></pre>
<pre><code>##                                               out$name
## 1  aryl hydrocarbon receptor nuclear translocator like
## 2                             period circadian clock 2
## 3      circadian associated repressor of transcription
## 4                    solute carrier family 17 member 9
## 5                                                 &lt;NA&gt;
## 6                                adenosine deaminase 2
## 7          D-box binding PAR bZIP transcription factor
## 8                       collagen type XI alpha 2 chain
## 9                   solute carrier family 25 member 17
## 10       nuclear receptor subfamily 1 group D member 1
##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                out$summary
## 1                                                                                                                                                                                                                                                             The protein encoded by this gene is a basic helix-loop-helix protein that forms a heterodimer with CLOCK. This heterodimer binds E-box enhancer elements upstream of Period (PER1, PER2, PER3) and Cryptochrome (CRY1, CRY2) genes and activates transcription of these genes. PER and CRY proteins heterodimerize and repress their own transcription by interacting in a feedback loop with CLOCK/ARNTL complexes. Defects in this gene have been linked to infertility, problems with gluconeogenesis and lipogenesis, and altered sleep patterns. Several transcript variants encoding different isoforms have been found for this gene.
## 2                                                                                                                                                                                                                                                                                                          This gene is a member of the Period family of genes and is expressed in a circadian pattern in the suprachiasmatic nucleus, the primary circadian pacemaker in the mammalian brain. Genes in this family encode components of the circadian rhythms of locomotor activity, metabolism, and behavior. This gene is upregulated by CLOCK/ARNTL heterodimers but then represses this upregulation in a feedback loop using PER/CRY heterodimers to interact with CLOCK/ARNTL. Polymorphisms in this gene may increase the risk of getting certain cancers and have been linked to sleep disorders.
## 3                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     &lt;NA&gt;
## 4                                                                                                                                                                                                                                                                                                                                                                                                                                                 This gene encodes a member of a family of transmembrane proteins that are involved in the transport of small molecules. The encoded protein participates in the vesicular uptake, storage, and secretion of adenoside triphosphate (ATP) and other nucleotides. A mutation in this gene was found in individuals with autosomal dominant disseminated superficial actinic porokeratosis-8. Alternative splicing results in multiple transcript variants.
## 5                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     &lt;NA&gt;
## 6                                                                                                                                                                                                                                                                                                                                                                             This gene encodes a member of a subfamily of the adenosine deaminase protein family. The encoded protein is one of two adenosine deaminases found in humans, which regulate levels of the signaling molecule, adenosine. The encoded protein is secreted from monocytes undergoing differentiation and may regulate cell proliferation and differentiation. This gene may be responsible for some of the phenotypic features associated with cat eye syndrome. Alternative splicing results in multiple transcript variants.
## 7                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            The protein encoded by this gene is a member of the PAR bZIP transcription factor family and binds to specific sequences in the promoters of several genes, such as albumin, CYP2A4, and CYP2A5. The encoded protein can bind DNA as a homo- or heterodimer and is involved in the regulation of some circadian rhythm genes.
## 8  This gene encodes one of the two alpha chains of type XI collagen, a minor fibrillar collagen. It is located on chromosome 6 very close to but separate from the gene for retinoid X receptor beta. Type XI collagen is a heterotrimer but the third alpha chain is a post-translationally modified alpha 1 type II chain. Proteolytic processing of this type XI chain produces PARP, a proline/arginine-rich protein that is an amino terminal domain. Mutations in this gene are associated with type III Stickler syndrome, otospondylomegaepiphyseal dysplasia (OSMED syndrome), Weissenbacher-Zweymuller syndrome, autosomal dominant non-syndromic sensorineural type 13 deafness (DFNA13), and autosomal recessive non-syndromic sensorineural type 53 deafness (DFNB53). Alternative splicing results in multiple transcript variants. A related pseudogene is located nearby on chromosome 6.
## 9                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           This gene encodes a peroxisomal membrane protein that belongs to the family of mitochondrial solute carriers. It is expressed in the liver, and is likely involved in transport. Alternative splicing results in multiple transcript variants.
## 10                                                                                                                                                                                                                                                                                                                                                                                        This gene encodes a transcription factor that is a member of the nuclear receptor subfamily 1. The encoded protein is a ligand-sensitive transcription factor that negatively regulates the expression of core clock proteins. In particular this protein represses the circadian clock transcription factor aryl hydrocarbon receptor nuclear translocator-like protein 1 (ARNTL). This protein may also be involved in regulating genes that function in metabolic, inflammatory and cardiovascular processes.</code></pre>
<pre class="r"><code>pathway2 &lt;- read.delim(&quot;../utilities/circadian_ash/circadian_genes_ash_0_5_pathways.tab&quot;)
if(length(which(pathway2$q.value &lt; 0.05)) &gt; 2){
  pathway2 &lt;- pathway2[which(pathway2$q.value &lt; 0.05),]
}
maxpath = 75
pathway &lt;- pathway2[1:min(maxpath, dim(pathway2)[1]), ]
ensembl_genes &lt;- pathway$members_input_overlap
pathway_list_genes &lt;- sapply(ensembl_genes, function(x) return(strsplit(as.character(x), &quot;; &quot;)))
all_genes &lt;- Reduce(union, pathway_list_genes)
pathway_mat &lt;- matrix(0, length(pathway_list_genes), length(all_genes))
for(l in 1:dim(pathway_mat)[1]){
  pathway_mat[l, match(pathway_list_genes[[l]], all_genes)] &lt;- 1
}

pathway_bimat &lt;- (pathway_mat)%*%t(pathway_mat)
pathway_bimat_prop &lt;- diag(1/sqrt(diag(pathway_bimat)), dim(pathway_bimat)[1]) %*% pathway_bimat %*% diag(1/sqrt(diag(pathway_bimat)), dim(pathway_bimat)[1])

col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
      rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
heatmap(pathway_bimat_prop, labRow = pathway$pathway, labCol = pathway$pathway, cexRow = 1, cexCol = 1, col = col, zlim = c(-1,1),
        scale = &quot;none&quot;, margins = c(5, 5))</code></pre>
<p><img src="circadian_compare_study_ash_files/figure-html/unnamed-chunk-11-1.png" width="672" /></p>
</div>
<div id="review---important-genes" class="section level2">
<h2>Review - Important genes</h2>
<p>The important genes are <strong>ENSG00000133794 or ARNTL</strong>, <strong>ENSG00000132326 or PER2</strong>, <strong>ENSG00000159208</strong> or <strong>CIART</strong>, <strong>ENSG00000101194</strong> or <strong>SLC17A7</strong> and <strong>ENSG0000025513</strong>.</p>
<p>ARNTL encodes for a transcription factor with a basic helix-loop-helix (bHLH) and a PAS domain. The human ARNTL gene has a predicted 24 exons and is located on the p15 band of the 11th chromosome.[3] The BMAL1 protein is 626 amino acids long and plays a key role as one of the positive elements in the mammalian autoregulatory transcription translation negative feedback loop (TTFL), which is responsible for generating molecular circadian rhythms. One of ARNTL protein’s earliest discovered functions in circadian regulation was related to the CLOCK-BMAL1 (CLOCK-ARNTL) heterodimer, which would bind through an E-box enhancer to activate the transcription of the gene encoding vasopressin. <a href="https://en.wikipedia.org/wiki/ARNTL">Article</a>.</p>
<p>Circadian variations of clock gene Per2 and cell cycle genes in different stages of carcinogenesis in golden hamster buccal mucosa. Circadian (24-hour) rhythms and cell division are fundamental biological systems in most organisms. There is substantial evidence that, in mammals, circadian rhythms affect the timing of cell divisions in vivo. Day-night variations in both the mitotic index and DNA synthesis occur in many tissues (e.g., oral mucosa, tongue keratinocytes, intestinal epithelium, skin, and bone marrow) some of which persist even in constant darkness. However, how the circadian clock controls the timing of cell divisions is not known. <a href="https://www.wikigenes.org/e/gene/e/8864.html">Article</a>, <a href="https://en.wikipedia.org/wiki/PER2">Article</a>.</p>
<p>To determine whether the circadian clock continues to function during the massive entry of hepatocytes into the cell cycle, we examined expression profiles of clock components PER1, PER2, CRY1, and Bmal1 (ARNTL). <a href="https://www.nature.com/articles/srep09997">Article</a>.</p>
<p>The clock gene Per2 links the circadian system to the estrogen receptor. <a href="http://www.nature.com/onc/journal/v26/n57/full/1210585a.html">Article</a>.</p>
<p>CIART is a circadian associated repressor of transcription. <a href="https://www.wikigenes.org/e/gene/e/148523.html">Article</a>. I did not find much literature on CIART. But it lies in the superpathway of Circadian entrainment. <a href="http://www.genecards.org/cgi-bin/carddisp.pl?gene=SLC17A7">Source</a>.</p>
<pre class="r"><code>times &lt;- strsplit(as.character(sample_attributes$DTHTIME), &quot;[:]&quot;)
time_stamp &lt;- unlist(lapply(times, function(x){
  y &lt;- as.numeric(x[1])
  z &lt;- as.numeric(x[2])
  w &lt;- y*60+z
  return(w)
}))/(24*60)

cos_times &lt;- cos(2*pi*time_stamp)
uncos_times &lt;- 2*pi*time_stamp
na_indices &lt;- which(is.na(cos_times))

person_tissue_genes_1 &lt;- person_tissue_genes[-na_indices,,]
cos_times_1 &lt;- cos_times[-na_indices]
uncos_times_1 &lt;- uncos_times[-na_indices]</code></pre>
</div>
<div id="ensg00000133794-or-arntl" class="section level2">
<h2>ENSG00000133794 or ARNTL</h2>
<p>Check the read expression profile in GTEx here</p>
<p><a href="https://gtexportal.org/home/gene/ENSG00000133794">ENSG00000133794 or ARNTL</a></p>
<pre class="r"><code>betahat_sebetahat_2 &lt;- get(load(&quot;../output/death_time_cor_nonmash.rda&quot;))
tmat &lt;- betahat_sebetahat_2$betahat/betahat_sebetahat_2$sebetahat</code></pre>
<pre class="r"><code>indices &lt;- match(&quot;ENSG00000133794&quot;, colnames(betahat_mat))</code></pre>
<pre class="r"><code>plot(tmat[, indices], col=&quot;red&quot;)
abline(0,0)</code></pre>
<p><img src="circadian_compare_study_ash_files/figure-html/unnamed-chunk-15-1.png" width="672" /></p>
<p>Which are the tissues with lowest lfsr</p>
<pre class="r"><code>sort(lfsr_mat[,indices], decreasing = FALSE)[1:5]</code></pre>
<pre><code>##            Brain - Cerebellum                       Thyroid 
##                  1.264773e-09                  3.311120e-06 
## Brain - Cerebellar Hemisphere                     Pituitary 
##                  1.115171e-04                  5.031812e-04 
##             Muscle - Skeletal 
##                  5.103580e-04</code></pre>
<pre class="r"><code>par(mfrow=c(3,3))
for(j in 1:53){
  plot(uncos_times_1, person_tissue_genes_1[,j, indices],
       ylab = dimnames(person_tissue_genes)[[2]][j])
  lines(loess.smooth(uncos_times_1,person_tissue_genes_1[,j, indices[1]]), 
        col=&quot;red&quot;, lty=2, lwd=2)
}</code></pre>
<p><img src="circadian_compare_study_ash_files/figure-html/unnamed-chunk-17-1.png" width="672" /><img src="circadian_compare_study_ash_files/figure-html/unnamed-chunk-17-2.png" width="672" /><img src="circadian_compare_study_ash_files/figure-html/unnamed-chunk-17-3.png" width="672" /><img src="circadian_compare_study_ash_files/figure-html/unnamed-chunk-17-4.png" width="672" /><img src="circadian_compare_study_ash_files/figure-html/unnamed-chunk-17-5.png" width="672" /><img src="circadian_compare_study_ash_files/figure-html/unnamed-chunk-17-6.png" width="672" /></p>
<pre class="r"><code>col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
      rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
image(as.matrix(ash_cor_mat[order_index,order_index, indices]),
      col=col, main=paste0(&quot;corshrink: &quot;), cex.main=2,
      xaxt = &quot;n&quot;, yaxt = &quot;n&quot;, zlim=c(-1,1))
axis(1, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)
axis(2, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)</code></pre>
<p><img src="circadian_compare_study_ash_files/figure-html/unnamed-chunk-18-1.png" width="672" /></p>
</div>
<div id="ensg00000132326-or-per2" class="section level2">
<h2>ENSG00000132326 or PER2</h2>
<p>Check the read expression profile in GTEx here</p>
<p><a href="https://gtexportal.org/home/gene/ENSG00000132326">ENSG00000132326 or PER2</a></p>
<pre class="r"><code>betahat_sebetahat_2 &lt;- get(load(&quot;../output/death_time_cor_nonmash.rda&quot;))
tmat &lt;- betahat_sebetahat_2$betahat/betahat_sebetahat_2$sebetahat</code></pre>
<pre class="r"><code>indices &lt;- match(&quot;ENSG00000132326&quot;, colnames(betahat_mat))</code></pre>
<pre class="r"><code>plot(tmat[, indices], col=&quot;red&quot;)
abline(0,0)</code></pre>
<p><img src="circadian_compare_study_ash_files/figure-html/unnamed-chunk-21-1.png" width="672" /></p>
<p>Which are the tissues with lowest lfsr</p>
<pre class="r"><code>sort(lfsr_mat[,indices], decreasing = FALSE)[1:5]</code></pre>
<pre><code>##            Thyroid              Liver  Muscle - Skeletal 
##       5.553239e-07       1.070244e-04       6.092091e-04 
## Brain - Cerebellum          Pituitary 
##       9.325260e-04       3.476692e-03</code></pre>
<pre class="r"><code>par(mfrow=c(3,3))
for(j in 1:53){
  plot(uncos_times_1, person_tissue_genes_1[,j, indices],
       ylab = dimnames(person_tissue_genes)[[2]][j])
  lines(loess.smooth(uncos_times_1,person_tissue_genes_1[,j, indices[1]]), 
        col=&quot;red&quot;, lty=2, lwd=2)
}</code></pre>
<p><img src="circadian_compare_study_ash_files/figure-html/unnamed-chunk-23-1.png" width="672" /><img src="circadian_compare_study_ash_files/figure-html/unnamed-chunk-23-2.png" width="672" /><img src="circadian_compare_study_ash_files/figure-html/unnamed-chunk-23-3.png" width="672" /><img src="circadian_compare_study_ash_files/figure-html/unnamed-chunk-23-4.png" width="672" /><img src="circadian_compare_study_ash_files/figure-html/unnamed-chunk-23-5.png" width="672" /><img src="circadian_compare_study_ash_files/figure-html/unnamed-chunk-23-6.png" width="672" /></p>
<pre class="r"><code>col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
      rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
image(as.matrix(ash_cor_mat[order_index,order_index, indices]),
      col=col, main=paste0(&quot;corshrink: &quot;), cex.main=2,
      xaxt = &quot;n&quot;, yaxt = &quot;n&quot;, zlim=c(-1,1))
axis(1, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)
axis(2, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)</code></pre>
<p><img src="circadian_compare_study_ash_files/figure-html/unnamed-chunk-24-1.png" width="672" /></p>
</div>
<div id="ensg00000159208-or-ciart" class="section level2">
<h2>ENSG00000159208 or CIART</h2>
<p>Check the read expression profile in GTEx here</p>
<p><a href="https://gtexportal.org/home/gene/ENSG00000159208">ENSG00000159208 or CIART</a></p>
<pre class="r"><code>betahat_sebetahat_2 &lt;- get(load(&quot;../output/death_time_cor_nonmash.rda&quot;))
tmat &lt;- betahat_sebetahat_2$betahat/betahat_sebetahat_2$sebetahat</code></pre>
<pre class="r"><code>indices &lt;- match(&quot;ENSG00000159208&quot;, colnames(betahat_mat))</code></pre>
<pre class="r"><code>plot(tmat[, indices], col=&quot;red&quot;)
abline(0,0)</code></pre>
<p><img src="circadian_compare_study_ash_files/figure-html/unnamed-chunk-27-1.png" width="672" /></p>
<p>Which are the tissues with lowest lfsr</p>
<pre class="r"><code>sort(lfsr_mat[,indices], decreasing = FALSE)[1:5]</code></pre>
<pre><code>##             Brain - Frontal Cortex (BA9) 
##                             1.481171e-05 
##                                Pituitary 
##                             1.018459e-03 
##          Brain - Caudate (basal ganglia) 
##                             7.170894e-03 
## Brain - Anterior cingulate cortex (BA24) 
##                             1.046946e-02 
##            Brain - Cerebellar Hemisphere 
##                             1.660962e-02</code></pre>
<pre class="r"><code>par(mfrow=c(3,3))
for(j in 1:53){
  plot(uncos_times_1, person_tissue_genes_1[,j, indices],
       ylab = dimnames(person_tissue_genes)[[2]][j])
  lines(loess.smooth(uncos_times_1,person_tissue_genes_1[,j, indices[1]]), 
        col=&quot;red&quot;, lty=2, lwd=2)
}</code></pre>
<p><img src="circadian_compare_study_ash_files/figure-html/unnamed-chunk-29-1.png" width="672" /><img src="circadian_compare_study_ash_files/figure-html/unnamed-chunk-29-2.png" width="672" /><img src="circadian_compare_study_ash_files/figure-html/unnamed-chunk-29-3.png" width="672" /></p>
<pre><code>## Warning in simpleLoess(y, x, w, span, degree = degree, parametric =
## FALSE, : pseudoinverse used at 3.7088</code></pre>
<pre><code>## Warning in simpleLoess(y, x, w, span, degree = degree, parametric =
## FALSE, : neighborhood radius 1.0472</code></pre>
<pre><code>## Warning in simpleLoess(y, x, w, span, degree = degree, parametric =
## FALSE, : reciprocal condition number 0</code></pre>
<pre><code>## Warning in simpleLoess(y, x, w, span, degree = degree, parametric =
## FALSE, : There are other near singularities as well. 2.6064</code></pre>
<p><img src="circadian_compare_study_ash_files/figure-html/unnamed-chunk-29-4.png" width="672" /><img src="circadian_compare_study_ash_files/figure-html/unnamed-chunk-29-5.png" width="672" /><img src="circadian_compare_study_ash_files/figure-html/unnamed-chunk-29-6.png" width="672" /></p>
<pre class="r"><code>col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
      rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
image(as.matrix(ash_cor_mat[order_index,order_index, indices]),
      col=col, main=paste0(&quot;corshrink: &quot;), cex.main=2,
      xaxt = &quot;n&quot;, yaxt = &quot;n&quot;, zlim=c(-1,1))
axis(1, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)
axis(2, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)</code></pre>
<p><img src="circadian_compare_study_ash_files/figure-html/unnamed-chunk-30-1.png" width="672" /></p>
</div>
<div id="ensg00000101194-or-slc17a7" class="section level2">
<h2>ENSG00000101194 or SLC17A7</h2>
<p>Check the read expression profile in GTEx here</p>
<p><a href="https://gtexportal.org/home/gene/ENSG00000101194">ENSG00000101194 or SLC17A7</a></p>
<pre class="r"><code>betahat_sebetahat_2 &lt;- get(load(&quot;../output/death_time_cor_nonmash.rda&quot;))
tmat &lt;- betahat_sebetahat_2$betahat/betahat_sebetahat_2$sebetahat</code></pre>
<pre class="r"><code>indices &lt;- match(&quot;ENSG00000101194&quot;, colnames(betahat_mat))</code></pre>
<pre class="r"><code>plot(tmat[, indices], col=&quot;red&quot;)
abline(0,0)</code></pre>
<p><img src="circadian_compare_study_ash_files/figure-html/unnamed-chunk-33-1.png" width="672" /></p>
<p>Which are the tissues with lowest lfsr</p>
<pre class="r"><code>sort(lfsr_mat[,indices], decreasing = FALSE)[1:5]</code></pre>
<pre><code>## Brain - Frontal Cortex (BA9)                  Whole Blood 
##                   0.07343320                   0.07403959 
##               Artery - Aorta                        Liver 
##                   0.10475307                   0.10909545 
##       Heart - Left Ventricle 
##                   0.20746886</code></pre>
<pre class="r"><code>par(mfrow=c(3,3))
for(j in 1:53){
  plot(uncos_times_1, person_tissue_genes_1[,j, indices],
       ylab = dimnames(person_tissue_genes)[[2]][j])
  #lines(loess.smooth(uncos_times_1,person_tissue_genes_1[,j, indices[1]]), 
  #      col=&quot;red&quot;, lty=2, lwd=2)
}</code></pre>
<p><img src="circadian_compare_study_ash_files/figure-html/unnamed-chunk-35-1.png" width="672" /><img src="circadian_compare_study_ash_files/figure-html/unnamed-chunk-35-2.png" width="672" /><img src="circadian_compare_study_ash_files/figure-html/unnamed-chunk-35-3.png" width="672" /><img src="circadian_compare_study_ash_files/figure-html/unnamed-chunk-35-4.png" width="672" /><img src="circadian_compare_study_ash_files/figure-html/unnamed-chunk-35-5.png" width="672" /><img src="circadian_compare_study_ash_files/figure-html/unnamed-chunk-35-6.png" width="672" /></p>
<pre class="r"><code>col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
      rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
image(as.matrix(ash_cor_mat[order_index,order_index, indices]),
      col=col, main=paste0(&quot;corshrink: &quot;), cex.main=2,
      xaxt = &quot;n&quot;, yaxt = &quot;n&quot;, zlim=c(-1,1))
axis(1, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)
axis(2, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)</code></pre>
<p><img src="circadian_compare_study_ash_files/figure-html/unnamed-chunk-36-1.png" width="672" /></p>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
