<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Kushal K Dey" />


<title>corshrink: whats the deal with brain?</title>

<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>




<link rel="stylesheet" href="cosmo.css" type="text/css" />

</head>

<body>


<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <link href="https://fonts.googleapis.com/css?family=Open+Sans:100,300,400,600" rel="stylesheet" type="text/css">
    <link href="../docs/cosmo.css" rel="stylesheet">
    <link href="../docs/cosmo.min.css" rel="stylesheet">
</head>
<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script>
<script src="../docs/bootstrap/js/bootstrap.min.js"></script>

<body>

<div class = "nav" align = "middle">
<div class= "p1"></div>
<p>
  <a class="btn btn-large btn btn-primary" type="button" href = "corshrink_front.html">HOME</a> 
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "workflow.html">WORKFLOW</a> 
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "tissue_corr.html">GTEx DEMO</a> 
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "about.html">METHODS</a> 
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "https://github.com/kkdey/CorShrink">SOFTWARE</a> 
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "license.html">LICENSE</a> 
   &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "https://github.com/kkdey/CorShrink_workflow">GITHUB</a> 
 </p>
</div>  
<!-- <div class="nav">
  <a href="index.html">Home page</a> &nbsp &nbsp
  <a href="musings.html">Musings</a> &nbsp &nbsp
  <a href="town.html">My town</a> &nbsp &nbsp
  <a href="links.html">Links</a>
</div> -->

<!-- Main content -->
</body>
</html>


<h1 class="title toc-ignore">corshrink: whats the deal with brain?</h1>
<h4 class="author"><em>Kushal K Dey</em></h4>
<h4 class="date"><em>5/12/2017</em></h4>


<div id="TOC">
<ul>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2017-05-27</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> cedfc43</p>
<p>In this script, I try to tell a story about brains and corshrink.</p>
<pre class="r"><code>cor_result &lt;- get(load(&quot;../output/ash_cor_only_voom_pearson_halfuniform_gtex_tissues.rda&quot;))
common_samples &lt;- get(load(&quot;../output/common_samples.rda&quot;))
tissue_labels &lt;- read.table(file = &quot;../data/GTEX_V6/samples_id.txt&quot;)[,3]

#library(data.table)
#data &lt;- data.frame(fread(&quot;../data/GTEX_V6/cis_gene_expression.txt&quot;))
#matdata &lt;- t(data[,-c(1,2)])

gene_names &lt;- as.character(read.table(file = &quot;../data/GTEX_V6/gene_names_GTEX_V6.txt&quot;)[,1])
gene_names_1 &lt;- as.character(sapply(gene_names, function(x) return(strsplit(x, &quot;[.]&quot;)[[1]][1])))


U &lt;- unique(tissue_labels)</code></pre>
<pre class="r"><code>library(CorShrink)
source(&quot;../../CorShrink/R/CorShrinkML.R&quot;)

cor_result_mean &lt;- apply(cor_result, c(1,2), mean)
cor_result_min &lt;- apply(cor_result, c(1,2), min)
cor_result_max &lt;- apply(cor_result, c(1,2), max)

col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
      rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
image(as.matrix(cor_result_mean)[order_index, order_index],
      col=col, main=paste0(&quot;CorShrink mat: mean &quot;), cex.main=1,
      xaxt = &quot;n&quot;, yaxt = &quot;n&quot;, zlim=c(-1,1))
axis(1, at = seq(0, 1, length.out = ncol(cor_result[,,1])), labels = U[order_index], las=2, cex.axis = 0.4)
axis(2, at = seq(0, 1, length.out = ncol(cor_result[,,1])), labels = U[order_index], las=2, cex.axis = 0.4)</code></pre>
<p><img src="figure/corshrink_brain_func.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
      rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
image(as.matrix(cor_result_min)[order_index, order_index],
      col=col, main=paste0(&quot;CorShrink mat: min &quot;), cex.main=1,
      xaxt = &quot;n&quot;, yaxt = &quot;n&quot;, zlim=c(-1,1))
axis(1, at = seq(0, 1, length.out = ncol(cor_result[,,1])), labels = U[order_index], las=2, cex.axis = 0.4)
axis(2, at = seq(0, 1, length.out = ncol(cor_result[,,1])), labels = U[order_index], las=2, cex.axis = 0.4)</code></pre>
<p><img src="figure/corshrink_brain_func.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
      rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
image(as.matrix(cor_result_max)[order_index, order_index],
      col=col, main=paste0(&quot;CorShrink mat: max &quot;), cex.main=1,
      xaxt = &quot;n&quot;, yaxt = &quot;n&quot;, zlim=c(-1,1))
axis(1, at = seq(0, 1, length.out = ncol(cor_result[,,1])), labels = U[order_index], las=2, cex.axis = 0.4)
axis(2, at = seq(0, 1, length.out = ncol(cor_result[,,1])), labels = U[order_index], las=2, cex.axis = 0.4)</code></pre>
<p><img src="figure/corshrink_brain_func.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>indices_brain &lt;- grep(&quot;Brain&quot;, U)
cor_result_brain &lt;- cor_result[indices_brain, indices_brain, ]

bad_brain_gene_bool &lt;- apply(cor_result_brain, 3, function(x){
      z = x[row(x) &gt; col(x)]
      if(quantile(z, 0.8) &lt; 0.1) { a = 1;
      return (a)} else {
        a = 0;
        return (a)}
})

bad_brain_gene_indices &lt;- which(bad_brain_gene_bool == 1)
bad_brain_gene_names &lt;- gene_names_1[bad_brain_gene_indices]
bad_brain_gene_names_col &lt;- cbind.data.frame(bad_brain_gene_names)</code></pre>
<pre class="r"><code>length(bad_brain_gene_names)</code></pre>
<pre><code>[1] 256</code></pre>
<pre class="r"><code>num &lt;- 240
col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
      rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
image(as.matrix(cor_result[,,bad_brain_gene_indices[num]])[order_index, order_index],
      col=col, main=paste0(&quot;corshrink: &quot;, bad_brain_gene_names[num]), cex.main=1,
      xaxt = &quot;n&quot;, yaxt = &quot;n&quot;, zlim=c(-1,1))
axis(1, at = seq(0, 1, length.out = ncol(cor_result[,,1])), labels = U[order_index], las=2, cex.axis = 0.4)
axis(2, at = seq(0, 1, length.out = ncol(cor_result[,,1])), labels = U[order_index], las=2, cex.axis = 0.4)</code></pre>
<p><img src="figure/corshrink_brain_func.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>library(knitr)
out &lt;- mygene::queryMany(bad_brain_gene_names,  scopes=&quot;ensembl.gene&quot;, fields=c(&quot;name&quot;, &quot;summary&quot;), species=&quot;human&quot;);</code></pre>
<pre><code>Finished
Pass returnall=TRUE to return lists of duplicate or missing query terms.</code></pre>
<pre class="r"><code>kable(as.data.frame(out))</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">X_id</th>
<th align="right">X_score</th>
<th align="left">name</th>
<th align="left">summary</th>
<th align="left">query</th>
<th align="left">notfound</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">5318</td>
<td align="right">22.507269</td>
<td align="left">plakophilin 2</td>
<td align="left">This gene encodes a member of the arm-repeat (armadillo) and plakophilin gene families. Plakophilin proteins contain numerous armadillo repeats, localize to cell desmosomes and nuclei, and participate in linking cadherins to intermediate filaments in the cytoskeleton. This gene product may regulate the signaling activity of beta-catenin. Two alternately spliced transcripts encoding two protein isoforms have been identified. A processed pseudogene with high similarity to this locus has been mapped to chromosome 12p13.</td>
<td align="left">ENSG00000057294</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">10795</td>
<td align="right">22.933798</td>
<td align="left">zinc finger protein 268</td>
<td align="left">NA</td>
<td align="left">ENSG00000090612</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">688</td>
<td align="right">22.504370</td>
<td align="left">Kruppel like factor 5</td>
<td align="left">This gene encodes a member of the Kruppel-like factor subfamily of zinc finger proteins. The encoded protein is a transcriptional activator that binds directly to a specific recognition motif in the promoters of target genes. This protein acts downstream of multiple different signaling pathways and is regulated by post-translational modification. It may participate in both promoting and suppressing cell proliferation. Expression of this gene may be changed in a variety of different cancers and in cardiovascular disease. Alternative splicing results in multiple transcript variants.</td>
<td align="left">ENSG00000102554</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">3081</td>
<td align="right">22.950788</td>
<td align="left">homogentisate 1,2-dioxygenase</td>
<td align="left">This gene encodes the enzyme homogentisate 1,2 dioxygenase. This enzyme is involved in the catabolism of the amino acids tyrosine and phenylalanine. Mutations in this gene are the cause of the autosomal recessive metabolism disorder alkaptonuria.</td>
<td align="left">ENSG00000113924</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">6382</td>
<td align="right">22.950275</td>
<td align="left">syndecan 1</td>
<td align="left">The protein encoded by this gene is a transmembrane (type I) heparan sulfate proteoglycan and is a member of the syndecan proteoglycan family. The syndecans mediate cell binding, cell signaling, and cytoskeletal organization and syndecan receptors are required for internalization of the HIV-1 tat protein. The syndecan-1 protein functions as an integral membrane protein and participates in cell proliferation, cell migration and cell-matrix interactions via its receptor for extracellular matrix proteins. Altered syndecan-1 expression has been detected in several different tumor types. While several transcript variants may exist for this gene, the full-length natures of only two have been described to date. These two represent the major variants of this gene and encode the same protein.</td>
<td align="left">ENSG00000115884</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">11126</td>
<td align="right">22.949055</td>
<td align="left">CD160 molecule</td>
<td align="left">CD160 is an 27 kDa glycoprotein which was initially identified with the monoclonal antibody BY55. Its expression is tightly associated with peripheral blood NK cells and CD8 T lymphocytes with cytolytic effector activity. The cDNA sequence of CD160 predicts a cysteine-rich, glycosylphosphatidylinositol-anchored protein of 181 amino acids with a single Ig-like domain weakly homologous to KIR2DL4 molecule. CD160 is expressed at the cell surface as a tightly disulfide-linked multimer. RNA blot analysis revealed CD160 mRNAs of 1.5 and 1.6 kb whose expression was highly restricted to circulating NK and T cells, spleen and small intestine. Within NK cells CD160 is expressed by CD56dimCD16+ cells whereas among circulating T cells its expression is mainly restricted to TCRgd bearing cells and to TCRab+CD8brightCD95+CD56+CD28-CD27-cells. In tissues, CD160 is expressed on all intestinal intraepithelial lymphocytes. CD160 shows a broad specificity for binding to both classical and nonclassical MHC class I molecules.</td>
<td align="left">ENSG00000117281</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000124097</td>
<td align="right">7.605654</td>
<td align="left">high mobility group box 1 pseudogene 1</td>
<td align="left">NA</td>
<td align="left">ENSG00000124097</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">10930</td>
<td align="right">22.507347</td>
<td align="left">apolipoprotein B mRNA editing enzyme catalytic subunit 2</td>
<td align="left">NA</td>
<td align="left">ENSG00000124701</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">57152</td>
<td align="right">22.953253</td>
<td align="left">secreted LY6/PLAUR domain containing 1</td>
<td align="left">The protein encoded by this gene is a member of the Ly6/uPAR family but lacks a GPI-anchoring signal sequence. It is thought that this secreted protein contains antitumor activity. Mutations in this gene have been associated with Mal de Meleda, a rare autosomal recessive skin disorder. This gene maps to the same chromosomal region as several members of the Ly6/uPAR family of glycoprotein receptors.</td>
<td align="left">ENSG00000126233</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">57111</td>
<td align="right">22.505074</td>
<td align="left">RAB25, member RAS oncogene family</td>
<td align="left">The protein encoded by this gene is a member of the RAS superfamily of small GTPases. The encoded protein is involved in membrane trafficking and cell survival. This gene has been found to be a tumor suppressor and an oncogene, depending on the context. Two variants, one protein-coding and the other not, have been found for this gene.</td>
<td align="left">ENSG00000132698</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">4629</td>
<td align="right">14.343923</td>
<td align="left">myosin heavy chain 11</td>
<td align="left">The protein encoded by this gene is a smooth muscle myosin belonging to the myosin heavy chain family. The gene product is a subunit of a hexameric protein that consists of two heavy chain subunits and two pairs of non-identical light chain subunits. It functions as a major contractile protein, converting chemical energy into mechanical energy through the hydrolysis of ATP. The gene encoding a human ortholog of rat NUDE1 is transcribed from the reverse strand of this gene, and its 3’ end overlaps with that of the latter. The pericentric inversion of chromosome 16 [inv(16)(p13q22)] produces a chimeric transcript that encodes a protein consisting of the first 165 residues from the N terminus of core-binding factor beta in a fusion with the C-terminal portion of the smooth muscle myosin heavy chain. This chromosomal rearrangement is associated with acute myeloid leukemia of the M4Eo subtype. Alternative splicing generates isoforms that are differentially expressed, with ratios changing during muscle cell maturation. Alternatively spliced transcript variants encoding different isoforms have been identified.</td>
<td align="left">ENSG00000133392</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">79020</td>
<td align="right">22.933798</td>
<td align="left">chromosome 7 open reading frame 25</td>
<td align="left">NA</td>
<td align="left">ENSG00000136197</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">57159</td>
<td align="right">22.953253</td>
<td align="left">tripartite motif containing 54</td>
<td align="left">The protein encoded by this gene contains a RING finger motif and is highly similar to the ring finger proteins RNF28/MURF1 and RNF29/MURF2. In vitro studies demonstrated that this protein, RNF28, and RNF29 form heterodimers, which may be important for the regulation of titin kinase and microtubule-dependent signal pathways in striated muscles. Alternatively spliced transcript variants encoding distinct isoforms have been reported.</td>
<td align="left">ENSG00000138100</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">347902</td>
<td align="right">22.949055</td>
<td align="left">adhesion molecule with Ig like domain 2</td>
<td align="left">NA</td>
<td align="left">ENSG00000139211</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">50511</td>
<td align="right">23.577528</td>
<td align="left">synaptonemal complex protein 3</td>
<td align="left">This gene encodes an essential structural component of the synaptonemal complex. This complex is involved in synapsis, recombination and segregation of meiotic chromosomes. Mutations in this gene are associated with azoospermia in males and susceptibility to pregnancy loss in females. Alternate splicing results in multiple transcript variants that encode the same protein.</td>
<td align="left">ENSG00000139351</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">112840</td>
<td align="right">22.950275</td>
<td align="left">WD repeat domain 89</td>
<td align="left">NA</td>
<td align="left">ENSG00000140006</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">51601</td>
<td align="right">23.578710</td>
<td align="left">lipoyltransferase 1</td>
<td align="left">The process of transferring lipoic acid to proteins is a two-step process. The first step is the activation of lipoic acid by lipoate-activating enzyme to form lipoyl-AMP. For the second step, the protein encoded by this gene transfers the lipoyl moiety to apoproteins. Alternative splicing results in multiple transcript variants. A related pseudogene has been identified on chromosome 13. Read-through transcription also exists between this gene and the neighboring downstream mitochondrial ribosomal protein L30 (MRPL30) gene.</td>
<td align="left">ENSG00000144182</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">10788</td>
<td align="right">22.487890</td>
<td align="left">IQ motif containing GTPase activating protein 2</td>
<td align="left">This gene encodes a member of the IQGAP family. The encoded protein contains three IQ domains, one calponin homology domain, one Ras-GAP domain and one WW domain. This protein interacts with components of the cytoskeleton, with cell adhesion molecules, and with several signaling molecules to regulate cell morphology and motility. It also acts as a tumor suppressor and has been found to play a role in regulating innate antiviral responses. Alternative splicing results in multiple transcript variants.</td>
<td align="left">ENSG00000145703</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">144715</td>
<td align="right">22.504330</td>
<td align="left">RAD9 checkpoint clamp component B</td>
<td align="left">NA</td>
<td align="left">ENSG00000151164</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">10557</td>
<td align="right">22.507347</td>
<td align="left">ribonuclease P/MRP subunit p38</td>
<td align="left">NA</td>
<td align="left">ENSG00000152464</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">100310846</td>
<td align="right">22.949055</td>
<td align="left">ankyrin repeat domain 61</td>
<td align="left">NA</td>
<td align="left">ENSG00000157999</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">286207</td>
<td align="right">22.523668</td>
<td align="left">cilia and flagella associated protein 157</td>
<td align="left">NA</td>
<td align="left">ENSG00000160401</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">3699</td>
<td align="right">22.953176</td>
<td align="left">inter-alpha-trypsin inhibitor heavy chain 3</td>
<td align="left">This gene encodes the heavy chain subunit of the pre-alpha-trypsin inhibitor complex. This complex may stabilize the extracellular matrix through its ability to bind hyaluronic acid. Polymorphisms of this gene may be associated with increased risk for schizophrenia and major depressive disorder. This gene is present in an inter-alpha-trypsin inhibitor family gene cluster on chromosome 3.</td>
<td align="left">ENSG00000162267</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">6692</td>
<td align="right">22.950275</td>
<td align="left">serine peptidase inhibitor, Kunitz type 1</td>
<td align="left">The protein encoded by this gene is a member of the Kunitz family of serine protease inhibitors. The protein is a potent inhibitor specific for HGF activator and is thought to be involved in the regulation of the proteolytic activation of HGF in injured tissues. Alternative splicing results in multiple variants encoding different isoforms.</td>
<td align="left">ENSG00000166145</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">3856</td>
<td align="right">22.503149</td>
<td align="left">keratin 8</td>
<td align="left">This gene is a member of the type II keratin family clustered on the long arm of chromosome 12. Type I and type II keratins heteropolymerize to form intermediate-sized filaments in the cytoplasm of epithelial cells. The product of this gene typically dimerizes with keratin 18 to form an intermediate filament in simple single-layered epithelial cells. This protein plays a role in maintaining cellular structural integrity and also functions in signal transduction and cellular differentiation. Mutations in this gene cause cryptogenic cirrhosis. Alternatively spliced transcript variants have been found for this gene.</td>
<td align="left">ENSG00000170421</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">57228</td>
<td align="right">22.950980</td>
<td align="left">small cell adhesion glycoprotein</td>
<td align="left">NA</td>
<td align="left">ENSG00000170545</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">49861</td>
<td align="right">22.507347</td>
<td align="left">claudin 20</td>
<td align="left">This gene encodes a member of the claudin family. Claudins are integral membrane proteins and components of tight junction strands. Tight junction strands serve as a physical barrier to prevent solutes and water from passing freely through the paracellular space between epithelial or endothelial cell sheets, and also play critical roles in maintaining cell polarity and signal transductions.</td>
<td align="left">ENSG00000171217</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">121275</td>
<td align="right">22.504880</td>
<td align="left">olfactory receptor family 10 subfamily AD member 1</td>
<td align="left">Olfactory receptors interact with odorant molecules in the nose, to initiate a neuronal response that triggers the perception of a smell. The olfactory receptor proteins are members of a large family of G-protein-coupled receptors (GPCR) arising from single coding-exon genes. Olfactory receptors share a 7-transmembrane domain structure with many neurotransmitter and hormone receptors and are responsible for the recognition and G protein-mediated transduction of odorant signals. The olfactory receptor gene family is the largest in the genome. The nomenclature assigned to the olfactory receptor genes and proteins for this organism is independent of other organisms.</td>
<td align="left">ENSG00000172640</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">5936</td>
<td align="right">22.503149</td>
<td align="left">RNA binding motif protein 4</td>
<td align="left">NA</td>
<td align="left">ENSG00000173933</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">5174</td>
<td align="right">22.504880</td>
<td align="left">PDZ domain containing 1</td>
<td align="left">This gene encodes a PDZ domain-containing scaffolding protein. PDZ domain-containing molecules bind to and mediate the subcellular localization of target proteins. The encoded protein mediates the localization of cell surface proteins and plays a critical role in cholesterol metabolism by regulating the HDL receptor, scavenger receptor class B type 1. Single nucleotide polymorphisms in this gene may be associated with metabolic syndrome, and overexpression of this gene may play a role in drug resistance of multiple myeloma. Pseudogenes of this gene are located on the long arm of chromosome 1. Alternatively spliced transcript variants encoding multiple isoforms have been observed for this gene.</td>
<td align="left">ENSG00000174827</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">317671</td>
<td align="right">22.950275</td>
<td align="left">Rieske Fe-S domain containing</td>
<td align="left">NA</td>
<td align="left">ENSG00000175449</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">2877</td>
<td align="right">22.507269</td>
<td align="left">glutathione peroxidase 2</td>
<td align="left">The protein encoded by this gene belongs to the glutathione peroxidase family, members of which catalyze the reduction of organic hydroperoxides and hydrogen peroxide (H2O2) by glutathione, and thereby protect cells against oxidative damage. Several isozymes of this gene family exist in vertebrates, which vary in cellular location and substrate specificity. This isozyme is predominantly expressed in the gastrointestinal tract (also in liver in human), is localized in the cytoplasm, and whose preferred substrate is hydrogen peroxide. Overexpression of this gene is associated with increased differentiation and proliferation in colorectal cancer. This isozyme is also a selenoprotein, containing the rare amino acid selenocysteine (Sec) at its active site. Sec is encoded by the UGA codon, which normally signals translation termination. The 3’ UTRs of selenoprotein mRNAs contain a conserved stem-loop structure, designated the Sec insertion sequence (SECIS) element, that is necessary for the recognition of UGA as a Sec codon, rather than as a stop signal. Alternatively spliced transcript variants have been found for this gene.</td>
<td align="left">ENSG00000176153</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">339456</td>
<td align="right">22.952688</td>
<td align="left">transmembrane protein 52</td>
<td align="left">NA</td>
<td align="left">ENSG00000178821</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">135932</td>
<td align="right">22.950980</td>
<td align="left">transmembrane protein 139</td>
<td align="left">NA</td>
<td align="left">ENSG00000178826</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000180189</td>
<td align="right">7.600732</td>
<td align="left">high mobility group box 1 pseudogene 14</td>
<td align="left">NA</td>
<td align="left">ENSG00000180189</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000180211</td>
<td align="right">23.598043</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000180211</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000180389</td>
<td align="right">7.403303</td>
<td align="left">ATP synthase, H+ transporting, mitochondrial F1 complex, epsilon subunit pseudogene 2</td>
<td align="left">NA</td>
<td align="left">ENSG00000180389</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">55432</td>
<td align="right">22.950788</td>
<td align="left">YOD1 deubiquitinase</td>
<td align="left">Protein ubiquitination controls many intracellular processes, including cell cycle progression, transcriptional activation, and signal transduction. This dynamic process, involving ubiquitin conjugating enzymes and deubiquitinating enzymes, adds and removes ubiquitin. Deubiquitinating enzymes are cysteine proteases that specifically cleave ubiquitin from ubiquitin-conjugated protein substrates. The protein encoded by this gene belongs to a DUB subfamily characterized by an ovarian tumor (OTU) domain. Alternative splicing results in multiple transcript variants.</td>
<td align="left">ENSG00000180667</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">116729</td>
<td align="right">22.953253</td>
<td align="left">protein phosphatase 1 regulatory subunit 27</td>
<td align="left">NA</td>
<td align="left">ENSG00000182676</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">25880</td>
<td align="right">22.953253</td>
<td align="left">transmembrane protein 186</td>
<td align="left">This gene encodes a potential transmembrane protein.</td>
<td align="left">ENSG00000184857</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000187984</td>
<td align="right">7.606275</td>
<td align="left">ankyrin repeat domain 19, pseudogene</td>
<td align="left">NA</td>
<td align="left">ENSG00000187984</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">57474</td>
<td align="right">22.953176</td>
<td align="left">zinc finger protein 490</td>
<td align="left">NA</td>
<td align="left">ENSG00000188033</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000188873</td>
<td align="right">7.607008</td>
<td align="left">ribosomal protein L10a pseudogene 2</td>
<td align="left">NA</td>
<td align="left">ENSG00000188873</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">NA</td>
<td align="right">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000197830</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">26783</td>
<td align="right">22.953176</td>
<td align="left">small nucleolar RNA, H/ACA box 65</td>
<td align="left">NA</td>
<td align="left">ENSG00000201302</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000206624</td>
<td align="right">7.607008</td>
<td align="left">RNA, U1 small nuclear 39, pseudogene</td>
<td align="left">NA</td>
<td align="left">ENSG00000206624</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000213080</td>
<td align="right">23.577528</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000213080</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000213149</td>
<td align="right">7.607008</td>
<td align="left">calponin 2 pseudogene 9</td>
<td align="left">NA</td>
<td align="left">ENSG00000213149</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000213293</td>
<td align="right">23.562270</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000213293</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000213590</td>
<td align="right">23.581160</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000213590</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000213864</td>
<td align="right">7.600732</td>
<td align="left">eukaryotic translation elongation factor 1 beta 2 pseudogene 2</td>
<td align="left">NA</td>
<td align="left">ENSG00000213864</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000213881</td>
<td align="right">7.606275</td>
<td align="left">nucleophosmin 1 pseudogene 6</td>
<td align="left">NA</td>
<td align="left">ENSG00000213881</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000214719</td>
<td align="right">23.598043</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000214719</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000214973</td>
<td align="right">7.605654</td>
<td align="left">coiled-coil-helix-coiled-coil-helix domain containing 3 pseudogene 3</td>
<td align="left">NA</td>
<td align="left">ENSG00000214973</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="right">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000216090</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">ENSG00000217325</td>
<td align="right">7.607008</td>
<td align="left">PRELI domain containing 1 pseudogene 1</td>
<td align="left">NA</td>
<td align="left">ENSG00000217325</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000218350</td>
<td align="right">7.606213</td>
<td align="left">lysophospholipase I pseudogene 3</td>
<td align="left">NA</td>
<td align="left">ENSG00000218350</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000219085</td>
<td align="right">7.606036</td>
<td align="left">nucleophosmin 1 pseudogene 37</td>
<td align="left">NA</td>
<td align="left">ENSG00000219085</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000219355</td>
<td align="right">7.607008</td>
<td align="left">ribosomal protein L31 pseudogene 52</td>
<td align="left">NA</td>
<td align="left">ENSG00000219355</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000219642</td>
<td align="right">7.606826</td>
<td align="left">bone morphogenetic protein receptor type 1A pseudogene 1</td>
<td align="left">NA</td>
<td align="left">ENSG00000219642</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000220305</td>
<td align="right">7.605654</td>
<td align="left">heterogeneous nuclear ribonucleoprotein H1 pseudogene 1</td>
<td align="left">NA</td>
<td align="left">ENSG00000220305</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">NA</td>
<td align="right">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000221016</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="right">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000223244</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">ENSG00000224094</td>
<td align="right">7.606036</td>
<td align="left">ribosomal protein S24 pseudogene 8</td>
<td align="left">NA</td>
<td align="left">ENSG00000224094</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000224415</td>
<td align="right">23.579260</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000224415</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000224452</td>
<td align="right">7.606983</td>
<td align="left">ribosomal L24 domain containing 1 pseudogene 6</td>
<td align="left">NA</td>
<td align="left">ENSG00000224452</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000225113</td>
<td align="right">23.579260</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000225113</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000225131</td>
<td align="right">7.606275</td>
<td align="left">proteasome activator subunit 2 pseudogene 2</td>
<td align="left">NA</td>
<td align="left">ENSG00000225131</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000225225</td>
<td align="right">7.607008</td>
<td align="left">ADP ribosylation factor like GTPase 2 binding protein pseudogene 10</td>
<td align="left">NA</td>
<td align="left">ENSG00000225225</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000225408</td>
<td align="right">23.598043</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000225408</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="right">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000225410</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">ENSG00000225555</td>
<td align="right">23.578747</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000225555</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">100873935</td>
<td align="right">22.950275</td>
<td align="left">MTOR antisense RNA 1</td>
<td align="left">NA</td>
<td align="left">ENSG00000225602</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000225610</td>
<td align="right">23.581724</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000225610</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000225793</td>
<td align="right">23.577528</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000225793</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000225891</td>
<td align="right">23.562270</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000225891</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000225920</td>
<td align="right">7.606036</td>
<td align="left">ribosomal modification protein rimK like family member B pseudogene 2</td>
<td align="left">NA</td>
<td align="left">ENSG00000225920</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000226055</td>
<td align="right">7.612272</td>
<td align="left">phosphoribosylaminoimidazole carboxylase, phosphoribosylaminoimidazole succinocarboxamide synthetase pseudogene 1</td>
<td align="left">NA</td>
<td align="left">ENSG00000226055</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000226361</td>
<td align="right">7.600732</td>
<td align="left">telomeric repeat binding factor 1 pseudogene 5</td>
<td align="left">NA</td>
<td align="left">ENSG00000226361</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000226723</td>
<td align="right">23.577528</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000226723</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000227344</td>
<td align="right">7.606213</td>
<td align="left">HAUS augmin like complex subunit 6 pseudogene 1</td>
<td align="left">NA</td>
<td align="left">ENSG00000227344</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000227527</td>
<td align="right">23.598043</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000227527</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000228137</td>
<td align="right">23.562270</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000228137</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000228205</td>
<td align="right">23.577528</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000228205</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000228251</td>
<td align="right">23.598043</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000228251</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000228283</td>
<td align="right">7.606983</td>
<td align="left">katanin regulatory subunit B1 like 1 pseudogene 6</td>
<td align="left">NA</td>
<td align="left">ENSG00000228283</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000228335</td>
<td align="right">23.581160</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000228335</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000229122</td>
<td align="right">23.581648</td>
<td align="left">AGBL5 intronic transcript 1</td>
<td align="left">NA</td>
<td align="left">ENSG00000229122</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000229348</td>
<td align="right">23.581648</td>
<td align="left">HYI antisense RNA 1</td>
<td align="left">NA</td>
<td align="left">ENSG00000229348</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000229659</td>
<td align="right">7.606826</td>
<td align="left">ribosomal protein L26 pseudogene 6</td>
<td align="left">NA</td>
<td align="left">ENSG00000229659</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="right">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000229789</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">ENSG00000229795</td>
<td align="right">7.607008</td>
<td align="left">ribosomal protein S21 pseudogene 1</td>
<td align="left">NA</td>
<td align="left">ENSG00000229795</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000229870</td>
<td align="right">7.612272</td>
<td align="left">ribosomal protein L21 pseudogene 89</td>
<td align="left">NA</td>
<td align="left">ENSG00000229870</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000229894</td>
<td align="right">7.403480</td>
<td align="left">glycerol kinase 3 pseudogene</td>
<td align="left">NA</td>
<td align="left">ENSG00000229894</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000230185</td>
<td align="right">23.581648</td>
<td align="left">chromosome 9 open reading frame 147</td>
<td align="left">NA</td>
<td align="left">ENSG00000230185</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000230289</td>
<td align="right">23.577528</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000230289</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000230303</td>
<td align="right">23.581724</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000230303</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000230358</td>
<td align="right">7.606983</td>
<td align="left">speedy/RINGO cell cycle regulator family member E21, pseudogene</td>
<td align="left">NA</td>
<td align="left">ENSG00000230358</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">101927795</td>
<td align="right">22.507269</td>
<td align="left">uncharacterized LOC101927795</td>
<td align="left">NA</td>
<td align="left">ENSG00000230408</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000230457</td>
<td align="right">7.605654</td>
<td align="left">proliferation-associated 2G4 pseudogene 4</td>
<td align="left">NA</td>
<td align="left">ENSG00000230457</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000231369</td>
<td align="right">23.579260</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000231369</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">100132215</td>
<td align="right">22.507269</td>
<td align="left">uncharacterized LOC100132215</td>
<td align="left">NA</td>
<td align="left">ENSG00000231609</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000231840</td>
<td align="right">23.578747</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000231840</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000231841</td>
<td align="right">23.579260</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000231841</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="right">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000232222</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">ENSG00000232344</td>
<td align="right">23.577528</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000232344</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000232531</td>
<td align="right">23.581648</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000232531</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000232640</td>
<td align="right">23.577528</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000232640</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000232797</td>
<td align="right">7.606213</td>
<td align="left">family with sequence similarity 207 member C, pseudogene</td>
<td align="left">NA</td>
<td align="left">ENSG00000232797</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">389791</td>
<td align="right">22.523668</td>
<td align="left">PTGES2 antisense RNA 1 (head to head)</td>
<td align="left">NA</td>
<td align="left">ENSG00000232850</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000232860</td>
<td align="right">23.581648</td>
<td align="left">SMG7 antisense RNA 1</td>
<td align="left">NA</td>
<td align="left">ENSG00000232860</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000233330</td>
<td align="right">23.578710</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000233330</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000233837</td>
<td align="right">7.606826</td>
<td align="left">eukaryotic translation initiation factor 3 subunit L pseudogene 2</td>
<td align="left">NA</td>
<td align="left">ENSG00000233837</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000233846</td>
<td align="right">23.581724</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000233846</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000233859</td>
<td align="right">7.605654</td>
<td align="left">alcohol dehydrogenase 5 (class III), chi polypeptide, pseudogene 4</td>
<td align="left">NA</td>
<td align="left">ENSG00000233859</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000234136</td>
<td align="right">23.578747</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000234136</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000234241</td>
<td align="right">23.579260</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000234241</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000234332</td>
<td align="right">7.606048</td>
<td align="left">breast carcinoma amplified sequence 2 pseudogene 2</td>
<td align="left">NA</td>
<td align="left">ENSG00000234332</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">83940</td>
<td align="right">14.344243</td>
<td align="left">TatD DNase domain containing 1</td>
<td align="left">NA</td>
<td align="left">ENSG00000234369</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000234424</td>
<td align="right">23.578747</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000234424</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000234500</td>
<td align="right">23.562270</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000234500</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000234638</td>
<td align="right">23.581724</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000234638</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000234981</td>
<td align="right">23.581724</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000234981</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000235036</td>
<td align="right">23.581724</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000235036</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000235042</td>
<td align="right">23.581648</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000235042</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000235308</td>
<td align="right">23.598043</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000235308</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000235313</td>
<td align="right">23.578747</td>
<td align="left">HM13 intronic transcript 1</td>
<td align="left">NA</td>
<td align="left">ENSG00000235313</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000235363</td>
<td align="right">7.606826</td>
<td align="left">small nuclear ribonucleoprotein polypeptide G pseudogene 10</td>
<td align="left">NA</td>
<td align="left">ENSG00000235363</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000235411</td>
<td align="right">23.581160</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000235411</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000235555</td>
<td align="right">7.606983</td>
<td align="left">SUMO1 pseudogene 4</td>
<td align="left">NA</td>
<td align="left">ENSG00000235555</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000235795</td>
<td align="right">23.578747</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000235795</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">644591</td>
<td align="right">23.577528</td>
<td align="left">peptidylprolyl isomerase A like 4G</td>
<td align="left">NA</td>
<td align="left">ENSG00000236334</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000236519</td>
<td align="right">23.581724</td>
<td align="left">long intergenic non-protein coding RNA 1424</td>
<td align="left">NA</td>
<td align="left">ENSG00000236519</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000236992</td>
<td align="right">7.606036</td>
<td align="left">ribosomal protein L12 pseudogene 12</td>
<td align="left">NA</td>
<td align="left">ENSG00000236992</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000237214</td>
<td align="right">23.581160</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000237214</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000237259</td>
<td align="right">23.562270</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000237259</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000237611</td>
<td align="right">23.578710</td>
<td align="left">Metazoan signal recognition particle RNA</td>
<td align="left">NA</td>
<td align="left">ENSG00000237611</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000237705</td>
<td align="right">23.578747</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000237705</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000237842</td>
<td align="right">23.581160</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000237842</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000238000</td>
<td align="right">23.581724</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000238000</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000238061</td>
<td align="right">23.577528</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000238061</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">100874201</td>
<td align="right">22.523668</td>
<td align="left">B4GALT4 antisense RNA 1</td>
<td align="left">NA</td>
<td align="left">ENSG00000240254</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000240376</td>
<td align="right">23.562270</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000240376</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000240695</td>
<td align="right">23.581648</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000240695</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000240710</td>
<td align="right">23.579450</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000240710</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000240898</td>
<td align="right">23.581724</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000240898</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">27297</td>
<td align="right">22.504370</td>
<td align="left">CGRP receptor component</td>
<td align="left">This gene encodes a membrane protein that functions as part of a receptor complex for a small neuropeptide that increases intracellular cAMP levels. Alternate transcriptional splice variants, encoding different isoforms, have been characterized.</td>
<td align="left">ENSG00000241258</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000241547</td>
<td align="right">7.600732</td>
<td align="left">actin gamma 1 pseudogene 20</td>
<td align="left">NA</td>
<td align="left">ENSG00000241547</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000241782</td>
<td align="right">23.578747</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000241782</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000242017</td>
<td align="right">7.606213</td>
<td align="left">asparagine-linked glycosylation 1-like 15, pseudogene</td>
<td align="left">NA</td>
<td align="left">ENSG00000242017</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000242553</td>
<td align="right">23.579260</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000242553</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000242571</td>
<td align="right">7.600732</td>
<td align="left">ribosomal protein L21 pseudogene 11</td>
<td align="left">NA</td>
<td align="left">ENSG00000242571</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">116143</td>
<td align="right">22.950275</td>
<td align="left">WD repeat domain 92</td>
<td align="left">This gene encodes a protein with two WD40 repeat domains thought to be involved in an apoptosis via activation of caspase-3. Multiple transcript variants encoding different isoforms have been found for this gene.</td>
<td align="left">ENSG00000243667</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000244332</td>
<td align="right">23.581648</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000244332</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000244582</td>
<td align="right">7.606983</td>
<td align="left">ribosomal protein L21 pseudogene 120</td>
<td align="left">NA</td>
<td align="left">ENSG00000244582</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">100507300</td>
<td align="right">23.581648</td>
<td align="left">ALKBH3 antisense RNA 1</td>
<td align="left">NA</td>
<td align="left">ENSG00000244926</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">101927480</td>
<td align="right">22.504330</td>
<td align="left">uncharacterized LOC101927480</td>
<td align="left">NA</td>
<td align="left">ENSG00000245694</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000247735</td>
<td align="right">23.579450</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000247735</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000248254</td>
<td align="right">23.578710</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000248254</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000248356</td>
<td align="right">23.577528</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000248356</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000248641</td>
<td align="right">7.606983</td>
<td align="left">high mobility group AT-hook 1 pseudogene 2</td>
<td align="left">NA</td>
<td align="left">ENSG00000248641</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000248968</td>
<td align="right">23.579450</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000248968</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000249142</td>
<td align="right">23.598043</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000249142</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000249492</td>
<td align="right">23.578710</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000249492</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000249502</td>
<td align="right">23.578747</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000249502</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">101929340</td>
<td align="right">22.503149</td>
<td align="left">uncharacterized LOC101929340</td>
<td align="left">NA</td>
<td align="left">ENSG00000250073</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000251224</td>
<td align="right">23.577528</td>
<td align="left">CNOT10 antisense RNA 1</td>
<td align="left">NA</td>
<td align="left">ENSG00000251224</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">677772</td>
<td align="right">14.345735</td>
<td align="left">small Cajal body-specific RNA 6</td>
<td align="left">NA</td>
<td align="left">ENSG00000251791</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000254244</td>
<td align="right">7.607008</td>
<td align="left">phosphoribosylaminoimidazole carboxylase, phosphoribosylaminoimidazole succinocarboxamide synthetase pseudogene 4</td>
<td align="left">NA</td>
<td align="left">ENSG00000254244</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000254571</td>
<td align="right">23.578710</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000254571</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000255038</td>
<td align="right">23.578747</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000255038</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000255585</td>
<td align="right">23.577528</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000255585</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">104169670</td>
<td align="right">22.507269</td>
<td align="left">TMEM5 antisense RNA 1</td>
<td align="left">NA</td>
<td align="left">ENSG00000255850</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000256210</td>
<td align="right">23.562270</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000256210</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000256448</td>
<td align="right">23.578747</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000256448</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000257243</td>
<td align="right">23.579260</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000257243</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">100422960</td>
<td align="right">23.578747</td>
<td align="left">microRNA 3179-1</td>
<td align="left">microRNAs (miRNAs) are short (20-24 nt) non-coding RNAs that are involved in post-transcriptional regulation of gene expression in multicellular organisms by affecting both the stability and translation of mRNAs. miRNAs are transcribed by RNA polymerase II as part of capped and polyadenylated primary transcripts (pri-miRNAs) that can be either protein-coding or non-coding. The primary transcript is cleaved by the Drosha ribonuclease III enzyme to produce an approximately 70-nt stem-loop precursor miRNA (pre-miRNA), which is further cleaved by the cytoplasmic Dicer ribonuclease to generate the mature miRNA and antisense miRNA star (miRNA*) products. The mature miRNA is incorporated into a RNA-induced silencing complex (RISC), which recognizes target mRNAs through imperfect base pairing with the miRNA and most commonly results in translational inhibition or destabilization of the target mRNA. The RefSeq represents the predicted microRNA stem-loop.</td>
<td align="left">ENSG00000257264</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000257307</td>
<td align="right">23.579450</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000257307</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">100422886</td>
<td align="right">22.950240</td>
<td align="left">microRNA 3179-2</td>
<td align="left">microRNAs (miRNAs) are short (20-24 nt) non-coding RNAs that are involved in post-transcriptional regulation of gene expression in multicellular organisms by affecting both the stability and translation of mRNAs. miRNAs are transcribed by RNA polymerase II as part of capped and polyadenylated primary transcripts (pri-miRNAs) that can be either protein-coding or non-coding. The primary transcript is cleaved by the Drosha ribonuclease III enzyme to produce an approximately 70-nt stem-loop precursor miRNA (pre-miRNA), which is further cleaved by the cytoplasmic Dicer ribonuclease to generate the mature miRNA and antisense miRNA star (miRNA*) products. The mature miRNA is incorporated into a RNA-induced silencing complex (RISC), which recognizes target mRNAs through imperfect base pairing with the miRNA and most commonly results in translational inhibition or destabilization of the target mRNA. The RefSeq represents the predicted microRNA stem-loop.</td>
<td align="left">ENSG00000257381</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000257489</td>
<td align="right">23.598043</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000257489</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000257613</td>
<td align="right">23.562270</td>
<td align="left">long intergenic non-protein coding RNA 1481</td>
<td align="left">NA</td>
<td align="left">ENSG00000257613</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000257773</td>
<td align="right">7.600732</td>
<td align="left">suppression of tumorigenicity 13 (colon carcinoma) (Hsp70 interacting protein) pseudogene 3</td>
<td align="left">NA</td>
<td align="left">ENSG00000257773</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000257839</td>
<td align="right">23.579260</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000257839</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000258099</td>
<td align="right">23.562270</td>
<td align="left">ATXN2 antisense RNA</td>
<td align="left">NA</td>
<td align="left">ENSG00000258099</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000258345</td>
<td align="right">23.579450</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000258345</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000258359</td>
<td align="right">7.612272</td>
<td align="left">PEST containing nuclear protein pseudogene 1</td>
<td align="left">NA</td>
<td align="left">ENSG00000258359</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000258443</td>
<td align="right">23.562270</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000258443</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000258568</td>
<td align="right">7.606275</td>
<td align="left">ras homolog family member Q pseudogene 1</td>
<td align="left">NA</td>
<td align="left">ENSG00000258568</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000258782</td>
<td align="right">23.598043</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000258782</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000259086</td>
<td align="right">23.581160</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000259086</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000259516</td>
<td align="right">7.605654</td>
<td align="left">acidic nuclear phosphoprotein 32 family member A pseudogene 1</td>
<td align="left">NA</td>
<td align="left">ENSG00000259516</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000259556</td>
<td align="right">23.578747</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000259556</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000259790</td>
<td align="right">7.605654</td>
<td align="left">acidic nuclear phosphoprotein 32 family member B pseudogene 1</td>
<td align="left">NA</td>
<td align="left">ENSG00000259790</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000259804</td>
<td align="right">23.598043</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000259804</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000259924</td>
<td align="right">23.581160</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000259924</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000260185</td>
<td align="right">23.581648</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000260185</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">101928707</td>
<td align="right">22.950240</td>
<td align="left">uncharacterized LOC101928707</td>
<td align="left">NA</td>
<td align="left">ENSG00000260219</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000260235</td>
<td align="right">23.579260</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000260235</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000260646</td>
<td align="right">23.579450</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000260646</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000260647</td>
<td align="right">23.581648</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000260647</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000260927</td>
<td align="right">23.581160</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000260927</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000260949</td>
<td align="right">23.578710</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000260949</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000261187</td>
<td align="right">23.581160</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000261187</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">NA</td>
<td align="right">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000261962</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000262777</td>
<td align="right">23.581648</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000262777</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000263065</td>
<td align="right">23.578710</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000263065</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000263220</td>
<td align="right">23.581160</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000263220</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000263388</td>
<td align="right">23.598043</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000263388</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000264924</td>
<td align="right">23.581648</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000264924</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000265451</td>
<td align="right">23.578747</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000265451</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000265791</td>
<td align="right">23.578710</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000265791</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">100861412</td>
<td align="right">22.953253</td>
<td align="left">fibrinogen silencer binding protein</td>
<td align="left">NA</td>
<td align="left">ENSG00000265817</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000266373</td>
<td align="right">23.579260</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000266373</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000266501</td>
<td align="right">23.562270</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000266501</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000267047</td>
<td align="right">23.581724</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000267047</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000267226</td>
<td align="right">23.598043</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000267226</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000267274</td>
<td align="right">23.578747</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000267274</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000267340</td>
<td align="right">23.581648</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000267340</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000267437</td>
<td align="right">23.581724</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000267437</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000267549</td>
<td align="right">23.579260</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000267549</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">NA</td>
<td align="right">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000267668</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">101928571</td>
<td align="right">22.933798</td>
<td align="left">ZNF649 antisense RNA 1</td>
<td align="left">NA</td>
<td align="left">ENSG00000268095</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000268288</td>
<td align="right">23.598043</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000268288</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000268543</td>
<td align="right">23.579450</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000268543</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000268798</td>
<td align="right">22.950980</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000268798</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000269038</td>
<td align="right">23.579450</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000269038</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000269106</td>
<td align="right">23.577528</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000269106</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000269903</td>
<td align="right">23.581724</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000269903</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000269921</td>
<td align="right">23.578710</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000269921</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000269924</td>
<td align="right">23.578747</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000269924</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000269938</td>
<td align="right">23.562270</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000269938</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000270061</td>
<td align="right">23.579450</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000270061</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000270062</td>
<td align="right">23.581160</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000270062</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000270115</td>
<td align="right">23.578747</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000270115</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000270130</td>
<td align="right">23.581648</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000270130</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000270750</td>
<td align="right">23.581724</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000270750</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000271360</td>
<td align="right">23.562270</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000271360</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000271580</td>
<td align="right">23.581648</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000271580</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000271664</td>
<td align="right">23.581160</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000271664</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">729096</td>
<td align="right">7.259639</td>
<td align="left">BMS1, ribosome biogenesis factor pseudogene 4</td>
<td align="left">NA</td>
<td align="left">ENSG00000271816</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000272186</td>
<td align="right">23.562270</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000272186</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000272226</td>
<td align="right">23.579260</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000272226</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000272263</td>
<td align="right">23.578710</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000272263</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000272389</td>
<td align="right">23.562270</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000272389</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000272432</td>
<td align="right">23.581724</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000272432</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000272498</td>
<td align="right">23.581724</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000272498</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000272510</td>
<td align="right">23.562270</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000272510</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000272716</td>
<td align="right">23.581160</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000272716</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000272735</td>
<td align="right">23.579260</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000272735</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000272814</td>
<td align="right">23.581724</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000272814</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000272843</td>
<td align="right">23.581648</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000272843</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000273004</td>
<td align="right">23.578747</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000273004</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000273174</td>
<td align="right">23.581160</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000273174</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000273245</td>
<td align="right">23.562270</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000273245</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000273447</td>
<td align="right">23.578710</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000273447</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">ENSG00000273485</td>
<td align="right">23.598043</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">ENSG00000273485</td>
<td align="left">NA</td>
</tr>
</tbody>
</table>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<!-- Insert the session information into the document -->
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.3.3 (2017-03-06)
Platform: x86_64-apple-darwin13.4.0 (64-bit)
Running under: macOS Sierra 10.12

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] knitr_1.15.1     CorShrink_0.99.0

loaded via a namespace (and not attached):
 [1] Biobase_2.34.0             httr_1.2.1                
 [3] jsonlite_1.4               splines_3.3.3             
 [5] foreach_1.4.3              gsubfn_0.6-6              
 [7] Formula_1.2-1              expm_0.999-2              
 [9] highr_0.6                  stats4_3.3.3              
[11] latticeExtra_0.6-28        Rsamtools_1.26.1          
[13] yaml_2.1.14                RSQLite_1.1-2             
[15] backports_1.0.5            lattice_0.20-35           
[17] chron_2.3-50               digest_0.6.12             
[19] GenomicRanges_1.26.1       RColorBrewer_1.1-2        
[21] XVector_0.14.0             checkmate_1.8.2           
[23] colorspace_1.3-2           htmltools_0.3.6           
[25] Matrix_1.2-8               plyr_1.8.4                
[27] XML_3.98-1.6               biomaRt_2.30.0            
[29] zlibbioc_1.20.0            scales_0.4.1              
[31] BiocParallel_1.8.2         git2r_0.18.0              
[33] htmlTable_1.9              tibble_1.3.1              
[35] IRanges_2.8.1              ggplot2_2.2.1             
[37] sqldf_0.4-10               SummarizedExperiment_1.4.0
[39] GenomicFeatures_1.26.0     ashr_2.1-15               
[41] nnet_7.3-12                BiocGenerics_0.20.0       
[43] lazyeval_0.2.0             proto_1.0.0               
[45] survival_2.41-3            magrittr_1.5              
[47] memoise_1.1.0              evaluate_0.10             
[49] doParallel_1.0.10          MASS_7.3-45               
[51] foreign_0.8-67             truncnorm_1.0-7           
[53] tools_3.3.3                data.table_1.10.4         
[55] stringr_1.2.0              S4Vectors_0.12.0          
[57] munsell_0.4.3              cluster_2.0.6             
[59] AnnotationDbi_1.36.0       Biostrings_2.42.0         
[61] GenomeInfoDb_1.10.1        rlang_0.1.1.9000          
[63] grid_3.3.3                 RCurl_1.95-4.8            
[65] iterators_1.0.8            htmlwidgets_0.8           
[67] bitops_1.0-6               base64enc_0.1-3           
[69] rmarkdown_1.5              gtable_0.2.0              
[71] codetools_0.2-15           curl_2.6                  
[73] DBI_0.6-1                  mygene_1.10.0             
[75] reshape2_1.4.2.9000        R6_2.2.1                  
[77] GenomicAlignments_1.10.1   gridExtra_2.2.1           
[79] rtracklayer_1.34.1         Hmisc_4.0-2               
[81] rprojroot_1.2              stringi_1.1.5             
[83] pscl_1.4.9                 parallel_3.3.3            
[85] SQUAREM_2016.8-2           Rcpp_0.12.11              
[87] rpart_4.1-10               acepack_1.4.1             </code></pre>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
