<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Kushal K Dey" />


<title>Death time comparison study</title>

<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>




<link rel="stylesheet" href="cosmo.css" type="text/css" />

</head>

<body>


<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <link href="https://fonts.googleapis.com/css?family=Open+Sans:100,300,400,600" rel="stylesheet" type="text/css">
    <link href="../docs/cosmo.css" rel="stylesheet">
    <link href="../docs/cosmo.min.css" rel="stylesheet">
</head>
<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script>
<script src="../docs/bootstrap/js/bootstrap.min.js"></script>

<body>

<div class = "nav" align = "middle">
<div class= "p1"></div>
<p>
  <a class="btn btn-large btn btn-primary" type="button" href = "corshrink_front.html">HOME</a> 
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "workflow.html">WORKFLOW</a> 
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "tissue_corr.html">GTEx DEMO</a> 
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "about.html">METHODS</a> 
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "https://github.com/kkdey/CorShrink">SOFTWARE</a> 
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "license.html">LICENSE</a> 
   &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "https://github.com/kkdey/CorShrink_workflow">GITHUB</a> 
 </p>
</div>  
<!-- <div class="nav">
  <a href="index.html">Home page</a> &nbsp &nbsp
  <a href="musings.html">Musings</a> &nbsp &nbsp
  <a href="town.html">My town</a> &nbsp &nbsp
  <a href="links.html">Links</a>
</div> -->

<!-- Main content -->
</body>
</html>


<h1 class="title toc-ignore">Death time comparison study</h1>
<h4 class="author"><em>Kushal K Dey</em></h4>
<h4 class="date"><em>6/4/2017</em></h4>


<div id="TOC">
<ul>
<li><a href="#death-time-data-reading-and-t-score-calculation">Death time data reading and t-score calculation</a></li>
<li><a href="#pathway-analysis-of-seemingly-significant-genes">Pathway analysis of seemingly significant genes</a></li>
<li><a href="#pathway-analysis-of-genes-above-quantile-threshold">Pathway analysis of genes above quantile threshold</a></li>
<li><a href="#t-score-analysis">t-score analysis</a><ul>
<li><a href="#t-score-plots-for-circadian-clock-genes">t-score plots for circadian clock genes</a></li>
<li><a href="#t-score-plots-for-circadian-rhythm-genes">t-score plots for circadian rhythm genes</a></li>
<li><a href="#t-score-plots-for-top-quantile-genes">t-score plots for top quantile genes</a></li>
<li><a href="#t-score-plots-for-random-set-of-genes">t-score plots for random set of genes</a></li>
</ul></li>
</ul>
</div>

<div id="death-time-data-reading-and-t-score-calculation" class="section level2">
<h2>Death time data reading and t-score calculation</h2>
<pre class="r"><code>betahat_sebetahat &lt;- get(load(&quot;../output/death_time_cor_nonmash.rda&quot;))
sample_attributes &lt;- get(load(&quot;../output/sample_attributes_filtered.rda&quot;))
person_tissue_genes &lt;- get(load(&quot;../output/person_tissue_genes_voom.rda&quot;))</code></pre>
<pre class="r"><code>betahat_mat &lt;- betahat_sebetahat$betahat
sebetahat_mat &lt;- betahat_sebetahat$sebetahat
pval_mat &lt;- betahat_sebetahat$pval
rsquare_mat &lt;- betahat_sebetahat$rsquare</code></pre>
<pre class="r"><code>tmat &lt;- betahat_mat/sebetahat_mat
abs_tmat &lt;- abs(tmat)
column_sum_1 &lt;- apply(abs_tmat, 2, function(x) return(quantile(x, 0.7)))</code></pre>
</div>
<div id="pathway-analysis-of-seemingly-significant-genes" class="section level2">
<h2>Pathway analysis of seemingly significant genes</h2>
<pre class="r"><code>names &lt;- dimnames(person_tissue_genes)[[3]][which(column_sum_1 &gt; 1.8)]
write.table(names, file = &quot;../utilities/circadian_genes.txt&quot;, quote = FALSE, row.names = FALSE, col.names = FALSE)
ensembl = biomaRt::useMart(&quot;ensembl&quot;, dataset=&quot;hsapiens_gene_ensembl&quot;, host=&quot;www.ensembl.org&quot;)
res = biomaRt::getBM(attributes = c(&quot;chromosome_name&quot;, &quot;ensembl_gene_id&quot;, &quot;external_gene_name&quot;, &quot;description&quot;), filters = c(&quot;ensembl_gene_id&quot;), values = names,  
mart = ensembl)
out &lt;- mygene::queryMany(names,  scopes=&quot;ensembl.gene&quot;, fields=c(&quot;name&quot;, &quot;summary&quot;), species=&quot;human&quot;);</code></pre>
<pre><code>## Finished
## Pass returnall=TRUE to return lists of duplicate or missing query terms.</code></pre>
<pre class="r"><code>cbind.data.frame(out[[4]], out[[5]])</code></pre>
<pre><code>##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   out[[4]]
## 1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        PITPNM2 belongs to a family of membrane-associated phosphatidylinositol transfer domain-containing proteins that share homology with the Drosophila retinal degeneration B (rdgB) protein (Ocaka et al., 2005 [PubMed 15627748]).
## 2                                                                                                                                                                                                                                                                                                                                                                             This gene encodes a member of a subfamily of the adenosine deaminase protein family. The encoded protein is one of two adenosine deaminases found in humans, which regulate levels of the signaling molecule, adenosine. The encoded protein is secreted from monocytes undergoing differentiation and may regulate cell proliferation and differentiation. This gene may be responsible for some of the phenotypic features associated with cat eye syndrome. Alternative splicing results in multiple transcript variants.
## 3                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       The protein encoded by this gene is involved in the synthesis of tetrahydrofolate (THF) in the mitochondrion. THF is important in the de novo synthesis of purines and thymidylate and in the regeneration of methionine from homocysteine. Several transcript variants encoding different isoforms have been found for this gene.
## 4                                                                                                                                                                                                                                                                                                          This gene is a member of the Period family of genes and is expressed in a circadian pattern in the suprachiasmatic nucleus, the primary circadian pacemaker in the mammalian brain. Genes in this family encode components of the circadian rhythms of locomotor activity, metabolism, and behavior. This gene is upregulated by CLOCK/ARNTL heterodimers but then represses this upregulation in a feedback loop using PER/CRY heterodimers to interact with CLOCK/ARNTL. Polymorphisms in this gene may increase the risk of getting certain cancers and have been linked to sleep disorders.
## 5                                                                                                                                                                                                                                                             The protein encoded by this gene is a basic helix-loop-helix protein that forms a heterodimer with CLOCK. This heterodimer binds E-box enhancer elements upstream of Period (PER1, PER2, PER3) and Cryptochrome (CRY1, CRY2) genes and activates transcription of these genes. PER and CRY proteins heterodimerize and repress their own transcription by interacting in a feedback loop with CLOCK/ARNTL complexes. Defects in this gene have been linked to infertility, problems with gluconeogenesis and lipogenesis, and altered sleep patterns. Several transcript variants encoding different isoforms have been found for this gene.
## 6                                                                                                                                                                                                                                                                                                                                                                                                          This gene encodes a member of a subfamily of basic helix-loop-helix transcription repressors that have homology to the Drosophila enhancer of split genes. Members of this gene family regulate cell differentiation in numerous cell types. The protein encoded by this gene functions as a cofactor, interacting with other transcription factors through a tetrapeptide domain in its C-terminus. Alternatively spliced transcript variants encoding different isoforms have been described.
## 7                                                                                                                                                                                                                                                                                                                                                                 Malic enzyme catalyzes the oxidative decarboxylation of malate to pyruvate using either NAD+ or NADP+ as a cofactor. Mammalian tissues contain 3 distinct isoforms of malic enzyme: a cytosolic NADP(+)-dependent isoform, a mitochondrial NADP(+)-dependent isoform, and a mitochondrial NAD(+)-dependent isoform. This gene encodes a mitochondrial NADP(+)-dependent isoform. Multiple alternatively spliced transcript variants have been found for this gene, but the biological validity of some variants has not been determined.
## 8                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     &lt;NA&gt;
## 9                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     &lt;NA&gt;
## 10                                                                                                                                                                                                                                                                                                                                                                                                                       This gene encodes a zinc metalloaminopeptidase of the M1 protease family that resides in the endoplasmic reticulum and functions in N-terminal trimming antigenic epitopes for presentation by major histocompatibility complex (MHC) class I molecules. Certain mutations in this gene are associated with the inflammatory arthritis syndrome ankylosing spondylitis and pre-eclampsia. This gene is located adjacent to a closely related aminopeptidase gene on chromosome 5.
## 11                                                                                                                                                                                   Autophagy is the process by which endogenous proteins and damaged organelles are destroyed intracellularly. Autophagy is postulated to be essential for cell homeostasis and cell remodeling during differentiation, metamorphosis, non-apoptotic cell death, and aging. Reduced levels of autophagy have been described in some malignant tumors, and a role for autophagy in controlling the unregulated cell growth linked to cancer has been proposed. This gene encodes a member of the autophagin protein family. The encoded protein is also designated as a member of the C-54 family of cysteine proteases. Alternate transcriptional splice variants, encoding different isoforms, have been characterized.
## 12                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    &lt;NA&gt;
## 13                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             This gene encodes a member of the nuclear hormone receptor family, specifically the NR1 subfamily of receptors. The encoded protein functions as a transcriptional repressor and may play a role in circadian rhythms and carbohydrate and lipid metabolism. Alternatively spliced transcript variants have been described.
## 14 This gene encodes one of the two alpha chains of type XI collagen, a minor fibrillar collagen. It is located on chromosome 6 very close to but separate from the gene for retinoid X receptor beta. Type XI collagen is a heterotrimer but the third alpha chain is a post-translationally modified alpha 1 type II chain. Proteolytic processing of this type XI chain produces PARP, a proline/arginine-rich protein that is an amino terminal domain. Mutations in this gene are associated with type III Stickler syndrome, otospondylomegaepiphyseal dysplasia (OSMED syndrome), Weissenbacher-Zweymuller syndrome, autosomal dominant non-syndromic sensorineural type 13 deafness (DFNA13), and autosomal recessive non-syndromic sensorineural type 53 deafness (DFNB53). Alternative splicing results in multiple transcript variants. A related pseudogene is located nearby on chromosome 6.
## 15                                                                                                                                                                                                                             Calcium-dependent membrane-binding proteins may regulate molecular events at the interface of the cell membrane and cytoplasm. This gene encodes a calcium-dependent protein that also contains two N-terminal type II C2 domains and an integrin A domain-like sequence in the C-terminus. However, the encoded protein does not contain a predicted signal sequence or transmembrane domains. This protein has a broad tissue distribution and it may function in membrane trafficking. This gene and the gene for RNA binding motif protein 12 overlap at map location 20q11.21. Alternate splicing results in multiple transcript variants encoding different proteins.
## 16                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    &lt;NA&gt;
## 17                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    &lt;NA&gt;
## 18                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    &lt;NA&gt;
## 19                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    &lt;NA&gt;
## 20                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    &lt;NA&gt;
##           out[[5]]
## 1  ENSG00000090975
## 2  ENSG00000093072
## 3  ENSG00000120254
## 4  ENSG00000132326
## 5  ENSG00000133794
## 6  ENSG00000144485
## 7  ENSG00000151376
## 8  ENSG00000158717
## 9  ENSG00000159208
## 10 ENSG00000164308
## 11 ENSG00000168397
## 12 ENSG00000171806
## 13 ENSG00000174738
## 14 ENSG00000204248
## 15 ENSG00000214078
## 16 ENSG00000214706
## 17 ENSG00000250379
## 18 ENSG00000255513
## 19 ENSG00000270028
## 20 ENSG00000272782</code></pre>
<pre class="r"><code>names</code></pre>
<pre><code>##  [1] &quot;ENSG00000090975&quot; &quot;ENSG00000093072&quot; &quot;ENSG00000120254&quot;
##  [4] &quot;ENSG00000132326&quot; &quot;ENSG00000133794&quot; &quot;ENSG00000144485&quot;
##  [7] &quot;ENSG00000151376&quot; &quot;ENSG00000158717&quot; &quot;ENSG00000159208&quot;
## [10] &quot;ENSG00000164308&quot; &quot;ENSG00000168397&quot; &quot;ENSG00000171806&quot;
## [13] &quot;ENSG00000174738&quot; &quot;ENSG00000204248&quot; &quot;ENSG00000214078&quot;
## [16] &quot;ENSG00000214706&quot; &quot;ENSG00000250379&quot; &quot;ENSG00000255513&quot;
## [19] &quot;ENSG00000270028&quot; &quot;ENSG00000272782&quot;</code></pre>
<p>The pathway data we read.</p>
<pre class="r"><code>circadian &lt;- read.delim(&quot;../utilities/circadian_pathways.tab&quot;)
cbind.data.frame(circadian$q.value, circadian$pathway, circadian$members_input_overlap_geneids)</code></pre>
<pre><code>##   circadian$q.value                                     circadian$pathway
## 1      7.464974e-05                 Exercise-induced Circadian Regulation
## 2      1.543851e-04                                       Circadian Clock
## 3      2.615618e-04                              Circadian rhythm pathway
## 4      2.615618e-04 BMAL1:CLOCK,NPAS2 activates circadian gene expression
## 5      4.932426e-04                      Melatonin metabolism and effects
## 6      4.932426e-04               Circadian rhythm - Homo sapiens (human)
## 7      6.107379e-04                                       Circadian Clock
## 8      1.354877e-02       Herpes simplex infection - Homo sapiens (human)
##   circadian$members_input_overlap_geneids
## 1                         8864; 406; 9975
## 2                               8864; 406
## 3                               8864; 406
## 4                               8864; 406
## 5                               8864; 406
## 6                               406; 8864
## 7                               8864; 406
## 8                               8864; 406</code></pre>
<pre class="r"><code>pathway2 &lt;- read.delim(&quot;../utilities/circadian_pathways.tab&quot;)
if(length(which(pathway2$q.value &lt; 0.05)) &gt; 2){
  pathway2 &lt;- pathway2[which(pathway2$q.value &lt; 0.05),]
}
maxpath = 75
pathway &lt;- pathway2[1:min(maxpath, dim(pathway2)[1]), ]
ensembl_genes &lt;- pathway$members_input_overlap
pathway_list_genes &lt;- sapply(ensembl_genes, function(x) return(strsplit(as.character(x), &quot;; &quot;)))
all_genes &lt;- Reduce(union, pathway_list_genes)
pathway_mat &lt;- matrix(0, length(pathway_list_genes), length(all_genes))
for(l in 1:dim(pathway_mat)[1]){
  pathway_mat[l, match(pathway_list_genes[[l]], all_genes)] &lt;- 1
}

pathway_bimat &lt;- (pathway_mat)%*%t(pathway_mat)
pathway_bimat_prop &lt;- diag(1/sqrt(diag(pathway_bimat)), dim(pathway_bimat)[1]) %*% pathway_bimat %*% diag(1/sqrt(diag(pathway_bimat)), dim(pathway_bimat)[1])

col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
      rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
heatmap(pathway_bimat_prop, labRow = pathway$pathway, labCol = pathway$pathway, cexRow = 1, cexCol = 1, col = col, zlim = c(-1,1),
        scale = &quot;none&quot;, margins = c(5, 5))</code></pre>
<p><img src="death_time_comparison_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
</div>
<div id="pathway-analysis-of-genes-above-quantile-threshold" class="section level2">
<h2>Pathway analysis of genes above quantile threshold</h2>
<pre class="r"><code>column_sum_1 &lt;- apply(abs_tmat, 2, function(x) return(quantile(x, 0.7)))
names &lt;- dimnames(person_tissue_genes)[[3]][order(column_sum_1, decreasing = TRUE)[1:100]]
write.table(names, file = &quot;../utilities/circadian_genes_2.txt&quot;, quote = FALSE, row.names = FALSE, col.names = FALSE)
ensembl = biomaRt::useMart(&quot;ensembl&quot;, dataset=&quot;hsapiens_gene_ensembl&quot;, host=&quot;www.ensembl.org&quot;)
res = biomaRt::getBM(attributes = c(&quot;chromosome_name&quot;, &quot;ensembl_gene_id&quot;, &quot;external_gene_name&quot;, &quot;description&quot;), filters = c(&quot;ensembl_gene_id&quot;), values = names,  
mart = ensembl)
out &lt;- mygene::queryMany(names,  scopes=&quot;ensembl.gene&quot;, fields=c(&quot;name&quot;, &quot;summary&quot;), species=&quot;human&quot;);</code></pre>
<pre><code>## Finished
## Pass returnall=TRUE to return lists of duplicate or missing query terms.</code></pre>
<pre class="r"><code>names</code></pre>
<pre><code>##   [1] &quot;ENSG00000133794&quot; &quot;ENSG00000132326&quot; &quot;ENSG00000159208&quot;
##   [4] &quot;ENSG00000120254&quot; &quot;ENSG00000174738&quot; &quot;ENSG00000151376&quot;
##   [7] &quot;ENSG00000164308&quot; &quot;ENSG00000204248&quot; &quot;ENSG00000272782&quot;
##  [10] &quot;ENSG00000255513&quot; &quot;ENSG00000250379&quot; &quot;ENSG00000158717&quot;
##  [13] &quot;ENSG00000214706&quot; &quot;ENSG00000144485&quot; &quot;ENSG00000168397&quot;
##  [16] &quot;ENSG00000090975&quot; &quot;ENSG00000214078&quot; &quot;ENSG00000171806&quot;
##  [19] &quot;ENSG00000270028&quot; &quot;ENSG00000093072&quot; &quot;ENSG00000131797&quot;
##  [22] &quot;ENSG00000013573&quot; &quot;ENSG00000257122&quot; &quot;ENSG00000168255&quot;
##  [25] &quot;ENSG00000261691&quot; &quot;ENSG00000166762&quot; &quot;ENSG00000131591&quot;
##  [28] &quot;ENSG00000170727&quot; &quot;ENSG00000233961&quot; &quot;ENSG00000126368&quot;
##  [31] &quot;ENSG00000138449&quot; &quot;ENSG00000105245&quot; &quot;ENSG00000188554&quot;
##  [34] &quot;ENSG00000160271&quot; &quot;ENSG00000270659&quot; &quot;ENSG00000258451&quot;
##  [37] &quot;ENSG00000249572&quot; &quot;ENSG00000199785&quot; &quot;ENSG00000137802&quot;
##  [40] &quot;ENSG00000103942&quot; &quot;ENSG00000176125&quot; &quot;ENSG00000234028&quot;
##  [43] &quot;ENSG00000163795&quot; &quot;ENSG00000004660&quot; &quot;ENSG00000105516&quot;
##  [46] &quot;ENSG00000072071&quot; &quot;ENSG00000007376&quot; &quot;ENSG00000108733&quot;
##  [49] &quot;ENSG00000101194&quot; &quot;ENSG00000261971&quot; &quot;ENSG00000178623&quot;
##  [52] &quot;ENSG00000133275&quot; &quot;ENSG00000171503&quot; &quot;ENSG00000179021&quot;
##  [55] &quot;ENSG00000167074&quot; &quot;ENSG00000179094&quot; &quot;ENSG00000182173&quot;
##  [58] &quot;ENSG00000223774&quot; &quot;ENSG00000213390&quot; &quot;ENSG00000100058&quot;
##  [61] &quot;ENSG00000130772&quot; &quot;ENSG00000127415&quot; &quot;ENSG00000214562&quot;
##  [64] &quot;ENSG00000198598&quot; &quot;ENSG00000008405&quot; &quot;ENSG00000168209&quot;
##  [67] &quot;ENSG00000166135&quot; &quot;ENSG00000183682&quot; &quot;ENSG00000110799&quot;
##  [70] &quot;ENSG00000103326&quot; &quot;ENSG00000255717&quot; &quot;ENSG00000169385&quot;
##  [73] &quot;ENSG00000243749&quot; &quot;ENSG00000269926&quot; &quot;ENSG00000155324&quot;
##  [76] &quot;ENSG00000272949&quot; &quot;ENSG00000254860&quot; &quot;ENSG00000177453&quot;
##  [79] &quot;ENSG00000204775&quot; &quot;ENSG00000214425&quot; &quot;ENSG00000214401&quot;
##  [82] &quot;ENSG00000178921&quot; &quot;ENSG00000172409&quot; &quot;ENSG00000270081&quot;
##  [85] &quot;ENSG00000174100&quot; &quot;ENSG00000213923&quot; &quot;ENSG00000203602&quot;
##  [88] &quot;ENSG00000128973&quot; &quot;ENSG00000055483&quot; &quot;ENSG00000238198&quot;
##  [91] &quot;ENSG00000174996&quot; &quot;ENSG00000092199&quot; &quot;ENSG00000049246&quot;
##  [94] &quot;ENSG00000188312&quot; &quot;ENSG00000131931&quot; &quot;ENSG00000103253&quot;
##  [97] &quot;ENSG00000105771&quot; &quot;ENSG00000167962&quot; &quot;ENSG00000215190&quot;
## [100] &quot;ENSG00000177764&quot;</code></pre>
<pre class="r"><code>circadian &lt;- read.delim(&quot;../utilities/circadian_pathways_2.tab&quot;)
cbind.data.frame(circadian$q.value, circadian$pathway, circadian$members_input_overlap_geneids)</code></pre>
<pre><code>##    circadian$q.value                                     circadian$pathway
## 1       5.231518e-08               Circadian rhythm - Homo sapiens (human)
## 2       5.231518e-08 BMAL1-CLOCK,NPAS2 activates circadian gene expression
## 3       5.231518e-08                              Circadian rhythm pathway
## 4       5.563338e-08                                       Circadian Clock
## 5       6.122722e-08 BMAL1:CLOCK,NPAS2 activates circadian gene expression
## 6       3.714580e-07                                       Circadian Clock
## 7       5.877847e-07                      Melatonin metabolism and effects
## 8       2.325424e-04                 Exercise-induced Circadian Regulation
## 9       3.819026e-02          Circadian entrainment - Homo sapiens (human)
## 10      4.441017e-02                                   Pyruvate metabolism
##    circadian$members_input_overlap_geneids
## 1       8863; 8864; 1454; 5187; 9572; 1407
## 2             8864; 5187; 1628; 9572; 1407
## 3             5187; 1454; 9572; 1407; 8864
## 4       1454; 1628; 8864; 1407; 5187; 9572
## 5             9572; 5187; 8864; 1628; 1407
## 6                   8864; 5187; 9572; 1407
## 7             8864; 1407; 8863; 5187; 1454
## 8                   8864; 9975; 1407; 5187
## 9                         5187; 8863; 8864
## 10                            10873; 84264</code></pre>
<p>The pathway heatmap in this case</p>
<pre class="r"><code>pathway2 &lt;- read.delim(&quot;../utilities/circadian_pathways_2.tab&quot;)
if(length(which(pathway2$q.value &lt; 0.05)) &gt; 2){
  pathway2 &lt;- pathway2[which(pathway2$q.value &lt; 0.05),]
}
maxpath = 75
pathway &lt;- pathway2[1:min(maxpath, dim(pathway2)[1]), ]
ensembl_genes &lt;- pathway$members_input_overlap
pathway_list_genes &lt;- sapply(ensembl_genes, function(x) return(strsplit(as.character(x), &quot;; &quot;)))
all_genes &lt;- Reduce(union, pathway_list_genes)
pathway_mat &lt;- matrix(0, length(pathway_list_genes), length(all_genes))
for(l in 1:dim(pathway_mat)[1]){
  pathway_mat[l, match(pathway_list_genes[[l]], all_genes)] &lt;- 1
}

pathway_bimat &lt;- (pathway_mat)%*%t(pathway_mat)
pathway_bimat_prop &lt;- diag(1/sqrt(diag(pathway_bimat)), dim(pathway_bimat)[1]) %*% pathway_bimat %*% diag(1/sqrt(diag(pathway_bimat)), dim(pathway_bimat)[1])

col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
      rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
heatmap(pathway_bimat_prop, labRow = pathway$pathway, labCol = pathway$pathway, cexRow = 1, cexCol = 1, col = col, zlim = c(-1,1),
        scale = &quot;none&quot;, margins = c(5, 5))</code></pre>
<p><img src="death_time_comparison_files/figure-html/unnamed-chunk-11-1.png" width="672" /></p>
</div>
<div id="t-score-analysis" class="section level2">
<h2>t-score analysis</h2>
<p>The plots of the t-scores for some of the genes which we know are part of the Circadian rthym.</p>
<pre class="r"><code>pathways &lt;- read.delim(&quot;../utilities/CPDB_pathways_genes.tab&quot;)
pathway_names &lt;- pathways[,1]
grep(&quot;Circadian&quot;, pathway_names)</code></pre>
<pre><code>## [1]  139  162 1165 1396 1513 2452 4090</code></pre>
<pre class="r"><code>circadian_pathways &lt;- pathways[grep(&quot;Circadian&quot;, pathway_names),]
gene_ids &lt;- circadian_pathways[,4]
circadian_rhythm_genes_pre &lt;- gene_ids[2]
circadian_clock_genes_pre &lt;- gene_ids[3]</code></pre>
<pre class="r"><code>circadian_clock_genes &lt;- strsplit(as.character(circadian_clock_genes_pre), &quot;[,]&quot;)[[1]]
circadian_rhythm_genes &lt;- strsplit(as.character(circadian_rhythm_genes_pre), &quot;[,]&quot;)[[1]]</code></pre>
<div id="t-score-plots-for-circadian-clock-genes" class="section level3">
<h3>t-score plots for circadian clock genes</h3>
<p>Plots of the t-scores for the circadian clock genes and the circadian rthym genes.</p>
<pre class="r"><code>clock_indices &lt;- match(circadian_clock_genes, dimnames(person_tissue_genes)[[3]])
clock_indices &lt;- clock_indices[!is.na(clock_indices)]</code></pre>
<pre class="r"><code>par(mfrow=c(3,3))
for(j in 1:length(clock_indices)){
  plot(tmat[,clock_indices[j]], pch=20, cex=1, ylim=c(-3, 3),
       ylab =paste0(dimnames(person_tissue_genes)[[3]][clock_indices[j]]))
  abline(0,0)
}</code></pre>
<p><img src="death_time_comparison_files/figure-html/unnamed-chunk-16-1.png" width="672" /></p>
</div>
<div id="t-score-plots-for-circadian-rhythm-genes" class="section level3">
<h3>t-score plots for circadian rhythm genes</h3>
<pre class="r"><code>rhythm_indices &lt;- match(circadian_rhythm_genes, dimnames(person_tissue_genes)[[3]])
rhythm_indices &lt;- rhythm_indices[!is.na(rhythm_indices)]</code></pre>
<pre class="r"><code>par(mfrow=c(3,3))
for(j in 1:length(rhythm_indices)){
  plot(tmat[,rhythm_indices[j]], pch=20, cex=1, ylim=c(-3, 3),
       ylab =paste0(dimnames(person_tissue_genes)[[3]][rhythm_indices[j]]))
  abline(0,0)
}</code></pre>
<p><img src="death_time_comparison_files/figure-html/unnamed-chunk-18-1.png" width="672" /><img src="death_time_comparison_files/figure-html/unnamed-chunk-18-2.png" width="672" /><img src="death_time_comparison_files/figure-html/unnamed-chunk-18-3.png" width="672" /></p>
</div>
<div id="t-score-plots-for-top-quantile-genes" class="section level3">
<h3>t-score plots for top quantile genes</h3>
<p>The top t-score values we checked for out pathway analysis, we see how the scatter plots look like for them.</p>
<pre class="r"><code>our_indices &lt;- order(column_sum_1, decreasing = TRUE)[1:25]
dimnames(person_tissue_genes)[[3]][our_indices]</code></pre>
<pre><code>##  [1] &quot;ENSG00000133794&quot; &quot;ENSG00000132326&quot; &quot;ENSG00000159208&quot;
##  [4] &quot;ENSG00000120254&quot; &quot;ENSG00000174738&quot; &quot;ENSG00000151376&quot;
##  [7] &quot;ENSG00000164308&quot; &quot;ENSG00000204248&quot; &quot;ENSG00000272782&quot;
## [10] &quot;ENSG00000255513&quot; &quot;ENSG00000250379&quot; &quot;ENSG00000158717&quot;
## [13] &quot;ENSG00000214706&quot; &quot;ENSG00000144485&quot; &quot;ENSG00000168397&quot;
## [16] &quot;ENSG00000090975&quot; &quot;ENSG00000214078&quot; &quot;ENSG00000171806&quot;
## [19] &quot;ENSG00000270028&quot; &quot;ENSG00000093072&quot; &quot;ENSG00000131797&quot;
## [22] &quot;ENSG00000013573&quot; &quot;ENSG00000257122&quot; &quot;ENSG00000168255&quot;
## [25] &quot;ENSG00000261691&quot;</code></pre>
<pre class="r"><code>par(mfrow=c(3,3))
for(j in 1:length(our_indices)){
  plot(tmat[,our_indices[j]], pch=20, cex=1, ylim=c(-3, 3),
       ylab =paste0(dimnames(person_tissue_genes)[[3]][our_indices[j]]))
  abline(0,0)
}</code></pre>
<p><img src="death_time_comparison_files/figure-html/unnamed-chunk-20-1.png" width="672" /><img src="death_time_comparison_files/figure-html/unnamed-chunk-20-2.png" width="672" /><img src="death_time_comparison_files/figure-html/unnamed-chunk-20-3.png" width="672" /></p>
</div>
<div id="t-score-plots-for-random-set-of-genes" class="section level3">
<h3>t-score plots for random set of genes</h3>
<p>How does a random set of 25 genes look like in its patterns across the tissues.</p>
<pre class="r"><code>random_indices &lt;- sample(1:16069, 25, replace=FALSE)
par(mfrow=c(3,3))
for(j in 1:length(random_indices)){
  plot(tmat[,random_indices[j]], pch=20, cex=1, ylim=c(-3, 3),
       ylab =paste0(dimnames(person_tissue_genes)[[3]][random_indices[j]]))
  abline(0,0)
}</code></pre>
<p><img src="death_time_comparison_files/figure-html/unnamed-chunk-21-1.png" width="672" /><img src="death_time_comparison_files/figure-html/unnamed-chunk-21-2.png" width="672" /><img src="death_time_comparison_files/figure-html/unnamed-chunk-21-3.png" width="672" /></p>
</div>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
