<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Kushal K Dey" />


<title>Blood and Spleen : Tissue wide CorShrink</title>

<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>




<link rel="stylesheet" href="cosmo.css" type="text/css" />

</head>

<body>


<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <link href="https://fonts.googleapis.com/css?family=Open+Sans:100,300,400,600" rel="stylesheet" type="text/css">
    <link href="../docs/cosmo.css" rel="stylesheet">
    <link href="../docs/cosmo.min.css" rel="stylesheet">
</head>
<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script>
<script src="../docs/bootstrap/js/bootstrap.min.js"></script>

<body>

<div class = "nav" align = "middle">
<div class= "p1"></div>
<p>
  <a class="btn btn-large btn btn-primary" type="button" href = "corshrink_front.html">HOME</a> 
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "workflow.html">WORKFLOW</a> 
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "tissue_corr.html">GTEx DEMO</a> 
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "about.html">METHODS</a> 
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "https://github.com/kkdey/CorShrink">SOFTWARE</a> 
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "license.html">LICENSE</a> 
   &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "https://github.com/kkdey/CorShrink_workflow">GITHUB</a> 
 </p>
</div>  
<!-- <div class="nav">
  <a href="index.html">Home page</a> &nbsp &nbsp
  <a href="musings.html">Musings</a> &nbsp &nbsp
  <a href="town.html">My town</a> &nbsp &nbsp
  <a href="links.html">Links</a>
</div> -->

<!-- Main content -->
</body>
</html>


<h1 class="title toc-ignore">Blood and Spleen : Tissue wide CorShrink</h1>
<h4 class="author"><em>Kushal K Dey</em></h4>
<h4 class="date"><em>5/25/2017</em></h4>



<pre class="r"><code>rm(list=ls())
corshrink_data &lt;- get(load(&quot;../output/ash_cor_only_voom_pearson_halfuniform_gtex_tissues.rda&quot;))
cor_data &lt;- get(load(&quot;../output/cor_tissues_non_ash_voom_pearson.rda&quot;))</code></pre>
<pre class="r"><code>common_samples &lt;- get(load(&quot;../output/common_samples.rda&quot;))
tissue_labels &lt;- read.table(file = &quot;../data/GTEX_V6/samples_id.txt&quot;)[,3]

#library(data.table)
#data &lt;- data.frame(fread(&quot;../data/GTEX_V6/cis_gene_expression.txt&quot;))
#matdata &lt;- t(data[,-c(1,2)])

gene_names &lt;- as.character(read.table(file = &quot;../data/GTEX_V6/gene_names_GTEX_V6.txt&quot;)[,1])
gene_names_1 &lt;- as.character(sapply(gene_names, function(x) return(strsplit(x, &quot;[.]&quot;)[[1]][1])))


U &lt;- unique(tissue_labels)</code></pre>
<pre class="r"><code>tab &lt;- array(0, dim(corshrink_data)[3])
blood_spleen_set &lt;- match(c(&quot;Whole Blood&quot;, &quot;Spleen&quot;), U)
for(m in 1:dim(corshrink_data)[3]){
  temp &lt;- corshrink_data[blood_spleen_set, blood_spleen_set, m]
  temp1 &lt;-  corshrink_data[order_index[-(1:13)], order_index[-(1:13)],m]
  tab[m] &lt;-  median(temp[row(temp) &gt; col(temp)]) - median(temp1[row(temp1) &gt; col(temp1)])
}

ordering_high &lt;- order(tab, decreasing =  TRUE)[1:100]
ordering_low &lt;- order(tab, decreasing =  FALSE)[1:100]</code></pre>
<pre class="r"><code>top_gene &lt;- gene_names_1[ordering_high[1]]
col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
      rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
image(as.matrix(corshrink_data[order_index,order_index, ordering_high[1]]),
      col=col, main=paste0(&quot;corshrink: &quot;), cex.main=2,
      xaxt = &quot;n&quot;, yaxt = &quot;n&quot;, zlim=c(-1,1))
axis(1, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)
axis(2, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)</code></pre>
<p><img src="enrichment_blood_spleen_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
<pre class="r"><code>bottom_gene &lt;- gene_names_1[ordering_low[1]]
col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
      rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
image(as.matrix(corshrink_data[order_index,order_index, ordering_low[1]]),
      col=col, main=paste0(&quot;corshrink: &quot;), cex.main=2,
      xaxt = &quot;n&quot;, yaxt = &quot;n&quot;, zlim=c(-1,1))
axis(1, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)
axis(2, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)</code></pre>
<p><img src="enrichment_blood_spleen_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
<pre class="r"><code>high_blood_spleen_genes &lt;- gene_names_1[ordering_high]
low_blood_spleen_genes &lt;- gene_names_1[ordering_low]

write.table(cbind.data.frame(high_blood_spleen_genes), quote = FALSE, file = &quot;../utilities/blood_spleen/high_blood_spleen.txt&quot;, row.names = FALSE, col.names = FALSE)

write.table(cbind.data.frame(low_blood_spleen_genes), quote = FALSE, file = &quot;../utilities/blood_spleen/low_blood_spleen.txt&quot;, row.names = FALSE, col.names = FALSE)</code></pre>
<pre class="r"><code>tab &lt;- read.delim(&quot;../utilities/blood_spleen/pathway_high.tab&quot;)
head(cbind.data.frame(tab$pathway, tab$q.value), 10)</code></pre>
<pre><code>##                                            tab$pathway  tab$q.value
## 1                             Neutrophil degranulation 8.551864e-09
## 2                                 Innate Immune System 1.704004e-06
## 3                                        Immune System 1.461926e-04
## 4                   ATF-2 transcription factor network 1.223741e-02
## 5                                   Complement cascade 1.655519e-02
## 6      ECM-receptor interaction - Homo sapiens (human) 2.291675e-02
## 7       Cell surface interactions at the vascular wall 2.291675e-02
## 8                     Initial triggering of complement 4.218208e-02
## 9                                    ECM proteoglycans 4.922193e-02
## 10 Validated transcriptional targets of TAp63 isoforms 4.922193e-02</code></pre>
<pre class="r"><code>pathway2 &lt;- read.delim(&quot;../utilities/blood_spleen/pathway_high.tab&quot;)
if(length(which(pathway2$q.value &lt; 0.05)) &gt; 2){
  pathway2 &lt;- pathway2[which(pathway2$q.value &lt; 0.05),]
}
maxpath = 75
pathway &lt;- pathway2[1:min(maxpath, dim(pathway2)[1]), ]
ensembl_genes &lt;- pathway$members_input_overlap
pathway_list_genes &lt;- sapply(ensembl_genes, function(x) return(strsplit(as.character(x), &quot;; &quot;)))
all_genes &lt;- Reduce(union, pathway_list_genes)
pathway_mat &lt;- matrix(0, length(pathway_list_genes), length(all_genes))
for(l in 1:dim(pathway_mat)[1]){
  pathway_mat[l, match(pathway_list_genes[[l]], all_genes)] &lt;- 1
}

pathway_bimat &lt;- (pathway_mat)%*%t(pathway_mat)
pathway_bimat_prop &lt;- diag(1/sqrt(diag(pathway_bimat)), dim(pathway_bimat)[1]) %*% pathway_bimat %*% diag(1/sqrt(diag(pathway_bimat)), dim(pathway_bimat)[1])

col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
      rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
heatmap(pathway_bimat_prop, labRow = pathway$pathway, labCol = pathway$pathway, cexRow = 0.3, cexCol = 0.3, col = col, zlim = c(-1,1),
        scale = &quot;none&quot;, margins = c(5, 5))</code></pre>
<p><img src="enrichment_blood_spleen_files/figure-html/unnamed-chunk-9-1.png" width="672" /></p>
<pre class="r"><code>tab &lt;- read.delim(&quot;../utilities/blood_spleen/GO_high.tab&quot;)
head(cbind.data.frame(tab$term_name, tab$q.value), 10)</code></pre>
<pre><code>##                          tab$term_name  tab$q.value
## 1  immunoglobulin complex, circulating 6.448253e-10
## 2               immunoglobulin complex 1.638637e-09
## 3      immunoglobulin receptor binding 1.756445e-09
## 4                extracellular exosome 2.330844e-09
## 5                extracellular vesicle 1.638637e-09
## 6              extracellular organelle 1.557219e-09
## 7                  blood microparticle 3.438330e-09
## 8            phagocytosis, recognition 7.513661e-08
## 9     external side of plasma membrane 2.518414e-08
## 10                             vesicle 4.411240e-08</code></pre>
<pre class="r"><code>tab &lt;- read.delim(&quot;../utilities/blood_spleen/pathway_low.tab&quot;)
head(cbind.data.frame(tab$pathway, tab$q.value), 10)</code></pre>
<pre><code>##                                              tab$pathway  tab$q.value
## 1                           Metallothioneins bind metals 3.749645e-07
## 2                                 Response to metal ions 3.749645e-07
## 3                                       Zinc homeostasis 3.179130e-05
## 4              Mineral absorption - Homo sapiens (human) 9.496566e-05
## 5                                       Oxidative Stress 4.799253e-04
## 6                                     Copper homeostasis 2.467279e-03
## 7    Platinum Pathway, Pharmacokinetics/Pharmacodynamics 2.659553e-03
## 8                                            Methylation 2.710343e-03
## 9  BMAL1:CLOCK,NPAS2 activates circadian gene expression 4.023149e-03
## 10                      Melatonin metabolism and effects 8.255106e-03</code></pre>
<pre class="r"><code>pathway2 &lt;- read.delim(&quot;../utilities/blood_spleen/pathway_low.tab&quot;)
if(length(which(pathway2$q.value &lt; 0.05)) &gt; 2){
  pathway2 &lt;- pathway2[which(pathway2$q.value &lt; 0.05),]
}
maxpath = 75
pathway &lt;- pathway2[1:min(maxpath, dim(pathway2)[1]), ]
ensembl_genes &lt;- pathway$members_input_overlap
pathway_list_genes &lt;- sapply(ensembl_genes, function(x) return(strsplit(as.character(x), &quot;; &quot;)))
all_genes &lt;- Reduce(union, pathway_list_genes)
pathway_mat &lt;- matrix(0, length(pathway_list_genes), length(all_genes))
for(l in 1:dim(pathway_mat)[1]){
  pathway_mat[l, match(pathway_list_genes[[l]], all_genes)] &lt;- 1
}

pathway_bimat &lt;- (pathway_mat)%*%t(pathway_mat)
pathway_bimat_prop &lt;- diag(1/sqrt(diag(pathway_bimat)), dim(pathway_bimat)[1]) %*% pathway_bimat %*% diag(1/sqrt(diag(pathway_bimat)), dim(pathway_bimat)[1])

col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
      rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
heatmap(pathway_bimat_prop, labRow = pathway$pathway, labCol = pathway$pathway, cexRow = 0.3, cexCol = 0.3, col = col, zlim = c(-1,1),
        scale = &quot;none&quot;, margins = c(5, 5))</code></pre>
<p><img src="enrichment_blood_spleen_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
<pre class="r"><code>tab &lt;- read.delim(&quot;../utilities/blood_spleen/GO_low.tab&quot;)
head(cbind.data.frame(tab$term_name, tab$q.value), 10)</code></pre>
<pre><code>##                               tab$term_name tab$q.value
## 1                      response to zinc ion 0.001647947
## 2     9-cis-retinoic acid metabolic process 0.002016197
## 3                response to erythropoietin 0.002236401
## 4        retinoic acid biosynthetic process 0.004672481
## 5  maternal process involved in parturition 0.044432649
## 6            terpenoid biosynthetic process 0.007305029
## 7            cellular response to metal ion 0.008228319
## 8                                   nucleus 0.008641997
## 9                               preribosome 0.008641997
## 10 cellular response to inorganic substance 0.057261256</code></pre>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
