<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Kushal K Dey" />


<title>p53 and p63 gene analysis</title>

<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>




<link rel="stylesheet" href="cosmo.css" type="text/css" />

</head>

<body>


<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <link href="https://fonts.googleapis.com/css?family=Open+Sans:100,300,400,600" rel="stylesheet" type="text/css">
    <link href="../docs/cosmo.css" rel="stylesheet">
    <link href="../docs/cosmo.min.css" rel="stylesheet">
</head>
<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script>
<script src="../docs/bootstrap/js/bootstrap.min.js"></script>

<body>

<div class = "nav" align = "middle">
<div class= "p1"></div>
<p>
  <a class="btn btn-large btn btn-primary" type="button" href = "corshrink_front.html">HOME</a> 
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "workflow.html">WORKFLOW</a> 
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "tissue_corr.html">GTEx DEMO</a> 
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "about.html">METHODS</a> 
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "https://github.com/kkdey/CorShrink">SOFTWARE</a> 
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "license.html">LICENSE</a> 
   &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "https://github.com/kkdey/CorShrink_workflow">GITHUB</a> 
 </p>
</div>  
<!-- <div class="nav">
  <a href="index.html">Home page</a> &nbsp &nbsp
  <a href="musings.html">Musings</a> &nbsp &nbsp
  <a href="town.html">My town</a> &nbsp &nbsp
  <a href="links.html">Links</a>
</div> -->

<!-- Main content -->
</body>
</html>


<h1 class="title toc-ignore">p53 and p63 gene analysis</h1>
<h4 class="author"><em>Kushal K Dey</em></h4>
<h4 class="date"><em>6/5/2017</em></h4>


<div id="TOC">
<ul>
<li><a href="#getting-to-know-these-genes">Getting to know these genes</a></li>
<li><a href="#reads-expression-profile">Reads expression profile</a></li>
<li><a href="#ensg00000161513">ENSG00000161513</a></li>
<li><a href="#ensg00000130513">ENSG00000130513</a></li>
<li><a href="#ensg00000134574">ENSG00000134574</a></li>
<li><a href="#ensg00000185088">ENSG00000185088</a></li>
<li><a href="#ensg00000147889">ENSG00000147889</a></li>
<li><a href="#ensg00000172667">ENSG00000172667</a></li>
<li><a href="#corshrink-matrices">CorShrink matrices</a></li>
<li><a href="#correlation-scatter-plot-fibroblasts">Correlation scatter plot (Fibroblasts)</a><ul>
<li><a href="#ensg00000130513-1">ENSG00000130513</a></li>
<li><a href="#ensg00000185088-1">ENSG00000185088</a></li>
</ul></li>
<li><a href="#correlation-scatter-plot-lymphoblasts">Correlation scatter plot (Lymphoblasts)</a><ul>
<li><a href="#ensg00000130513-2">ENSG00000130513</a></li>
<li><a href="#ensg00000185088-2">ENSG00000185088</a></li>
</ul></li>
<li><a href="#correlation-plots-for-brain-spinal-cord">Correlation plots for brain spinal cord</a><ul>
<li><a href="#ensg00000130513-3">ENSG00000130513</a></li>
</ul></li>
</ul>
</div>

<div id="getting-to-know-these-genes" class="section level2">
<h2>Getting to know these genes</h2>
<p>In this script, we take a close look at the p53 and p63 genes that were found to be highly correlated with age in the GTEx data.</p>
<p>The genes related to p53 are <em>ENSG00000161513</em>, <em>ENSG00000130513</em>, <em>ENSG00000134574</em>, <em>ENSG00000185088</em>, <em>ENSG00000147889</em>, <em>ENSG00000172667</em>.</p>
<p><em>ENSG00000161513</em>, also called <strong>FDXR</strong>, sensitizes cells to oxidative stress-induced apoptosis and is regulated by the p53 family <a href="https://www.ncbi.nlm.nih.gov/pubmed/12370809">Article</a>. So, it makes sense it is associated with aging since p53 family plays a role in aging.</p>
<p><em>ENSG00000130513</em> also called <strong>GDF-15</strong> (or growth differentiation factor) is a p53- and demethylation-upregulating gene represses cell proliferation, invasion, and tumorigenesis in bladder carcinoma cells. <a href="https://www.nature.com/articles/srep12870">Article</a>, <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1975780/">Article2</a>.</p>
<p><em>ENSG00000134574</em> or Xeroderma Pigmentosum DDB2 <strong>DDB2</strong> gene is activated by p53 only in humans and not in mice. <a href="http://mcb.asm.org/content/22/10/3247">Article</a>.</p>
<p><em>ENSG00000185088</em> or <strong>RPS27L</strong> or Ribosomal protein S27-like and S27 interplay with p53-MDM2 axis s a target, a substrate, and a regulator. <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3077453/">Article</a>.</p>
<p><em>ENSG00000147889</em> or <strong>CDKN2A</strong> . Somatic mutations of CDKN2A are common in the majority of human cancers, with estimates that CDKN2A is the second most commonly inactivated gene in cancer after p53. The gene codes for two proteins, including the INK4 family member p16 (or p16INK4a) and p14arf. Both act as tumor suppressors by regulating the cell cycle. p14ARF (known as p19ARF in the mouse) activates the p53 tumor suppressor.</p>
<p><em>ENSG00000172667</em> orZMAT3 or Wig-1 is a p53 target gene that encodes an RNA-binding zinc-finger protein involved in the regulation of mRNA stability through binding to AU-rich elements (AREs). Wig-1 regulates cell cycle arrest and cell death through the p53 targets FAS. <a href="http://www.nature.com/onc/journal/v33/n35/full/onc2013594a.html">Article</a>.</p>
</div>
<div id="reads-expression-profile" class="section level2">
<h2>Reads expression profile</h2>
<p>We take these genes one by one and look for the reads expression across the tissues. Here are the plots from the GTEx portal site.</p>
<p><a href="https://gtexportal.org/home/gene/ENSG00000172667">ENSG00000172667</a> <a href="https://gtexportal.org/home/gene/ENSG00000147889">ENSG00000147889</a> <a href="https://gtexportal.org/home/gene/ENSG00000185088">ENSG00000185088</a> <a href="https://gtexportal.org/home/gene/ENSG00000134574">ENSG00000134574</a> <a href="https://gtexportal.org/home/gene/ENSG00000130513">ENSG00000130513</a> <a href="https://gtexportal.org/home/gene/ENSG00000161513">ENSG00000161513</a></p>
<p>In all these examples, it seems the cell lymphocytes and cell fibroblasts seem to be higher expressed compared to most tissues.</p>
<p>We know p53 prevent apoptosis and aids in the survival of stem cells. We also know that it plays a role in the survival of the stem cells. Now we know that fibroblasts play a very important role in wound healing which is linked with stem cells, so there seems to be a connection there.</p>
<p>We also look at the p63 genes in the pathways we found and their reads expression profiles in GTEx. The genes we consider are <em>ENSG00000161513</em>, <em>ENSG00000130513</em> and <em>ENSG00000147889</em>. All these genes are already studied in their patterns above.</p>
<p>We now look at the patterns of expression of each of these genes for the different samples across the 53 tissues of interest.</p>
<pre class="r"><code>sample_attributes &lt;- get(load(&quot;../output/sample_attributes_filtered.rda&quot;))
person_tissue_genes &lt;- get(load(&quot;../output/person_tissue_genes_voom.rda&quot;))</code></pre>
<pre class="r"><code>indices &lt;- match(c(&quot;ENSG00000161513&quot;, &quot;ENSG00000130513&quot;,
                   &quot;ENSG00000134574&quot;, &quot;ENSG00000185088&quot;,
                   &quot;ENSG00000147889&quot;, &quot;ENSG00000172667&quot;), dimnames(person_tissue_genes)[[3]])</code></pre>
</div>
<div id="ensg00000161513" class="section level2">
<h2>ENSG00000161513</h2>
<pre class="r"><code>par(mfrow = c(3,3))
for(j in 1:53){
  temp1 &lt;- person_tissue_genes[,j,indices[1]]
  plot(temp1, sample_attributes$AGE, ylab = dimnames(person_tissue_genes)[[2]][j])
  lines(loess.smooth(temp1,sample_attributes$AGE, family = &quot;gaussian&quot;), 
        col=&quot;red&quot;, lty=2, lwd=2)
}</code></pre>
<p><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-3-1.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-3-2.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-3-3.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-3-4.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-3-5.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-3-6.png" width="672" /></p>
</div>
<div id="ensg00000130513" class="section level2">
<h2>ENSG00000130513</h2>
<pre class="r"><code>par(mfrow = c(3,3))
for(j in 1:53){
  temp1 &lt;- person_tissue_genes[,j,indices[2]]
  plot(temp1, sample_attributes$AGE, ylab = dimnames(person_tissue_genes)[[2]][j])
  lines(loess.smooth(temp1,sample_attributes$AGE, family = &quot;gaussian&quot;), 
        col=&quot;red&quot;, lty=2, lwd=2)
}</code></pre>
<p><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-4-1.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-4-2.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-4-3.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-4-4.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-4-5.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-4-6.png" width="672" /></p>
</div>
<div id="ensg00000134574" class="section level2">
<h2>ENSG00000134574</h2>
<pre class="r"><code>par(mfrow = c(3,3))
for(j in 1:53){
  temp1 &lt;- person_tissue_genes[,j,indices[3]]
  plot(temp1, sample_attributes$AGE, ylab = dimnames(person_tissue_genes)[[2]][j])
  lines(loess.smooth(temp1,sample_attributes$AGE, family = &quot;gaussian&quot;), 
        col=&quot;red&quot;, lty=2, lwd=2)
}</code></pre>
<p><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-5-1.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-5-2.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-5-3.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-5-4.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-5-5.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-5-6.png" width="672" /></p>
</div>
<div id="ensg00000185088" class="section level2">
<h2>ENSG00000185088</h2>
<pre class="r"><code>par(mfrow = c(3,3))
for(j in 1:53){
  temp1 &lt;- person_tissue_genes[,j,indices[4]]
  plot(temp1, sample_attributes$AGE, ylab = dimnames(person_tissue_genes)[[2]][j])
  lines(loess.smooth(temp1,sample_attributes$AGE, family = &quot;gaussian&quot;), 
        col=&quot;red&quot;, lty=2, lwd=2)
}</code></pre>
<p><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-6-1.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-6-2.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-6-3.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-6-4.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-6-5.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-6-6.png" width="672" /></p>
</div>
<div id="ensg00000147889" class="section level2">
<h2>ENSG00000147889</h2>
<pre class="r"><code>par(mfrow = c(3,3))
for(j in 1:53){
  temp1 &lt;- person_tissue_genes[,j,indices[5]]
  plot(temp1, sample_attributes$AGE, ylab = dimnames(person_tissue_genes)[[2]][j])
  lines(loess.smooth(temp1,sample_attributes$AGE, family = &quot;gaussian&quot;), 
        col=&quot;red&quot;, lty=2, lwd=2)
}</code></pre>
<p><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-7-1.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-7-2.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-7-3.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-7-4.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-7-5.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-7-6.png" width="672" /></p>
</div>
<div id="ensg00000172667" class="section level2">
<h2>ENSG00000172667</h2>
<pre class="r"><code>par(mfrow = c(3,3))
for(j in 1:53){
  temp1 &lt;- person_tissue_genes[,j,indices[6]]
  plot(temp1, sample_attributes$AGE, ylab = dimnames(person_tissue_genes)[[2]][j])
  lines(loess.smooth(temp1,sample_attributes$AGE, family = &quot;gaussian&quot;), 
        col=&quot;red&quot;, lty=2, lwd=2)
}</code></pre>
<p><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-8-1.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-8-2.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-8-3.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-8-4.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-8-5.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-8-6.png" width="672" /></p>
</div>
<div id="corshrink-matrices" class="section level2">
<h2>CorShrink matrices</h2>
<p>Now we check how the CorShrink matrices look like for these genes.</p>
<pre class="r"><code>ash_cor_mat &lt;- get(load(&quot;../output/ash_cor_only_voom_pearson_halfuniform_gtex_tissues.rda&quot;))</code></pre>
<pre class="r"><code>common_samples &lt;- get(load(&quot;../output/common_samples.rda&quot;))
tissue_labels &lt;- read.table(file = &quot;../data/GTEX_V6/samples_id.txt&quot;)[,3]

#library(data.table)
#data &lt;- data.frame(fread(&quot;../data/GTEX_V6/cis_gene_expression.txt&quot;))
#matdata &lt;- t(data[,-c(1,2)])

gene_names &lt;- as.character(read.table(file = &quot;../data/GTEX_V6/gene_names_GTEX_V6.txt&quot;)[,1])
gene_names_1 &lt;- as.character(sapply(gene_names, function(x) return(strsplit(x, &quot;[.]&quot;)[[1]][1])))


U &lt;- unique(tissue_labels)</code></pre>
<pre class="r"><code>col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
      rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
image(as.matrix(ash_cor_mat[order_index,order_index, indices[1]]),
      col=col, main=paste0(&quot;corshrink: &quot;), cex.main=2,
      xaxt = &quot;n&quot;, yaxt = &quot;n&quot;, zlim=c(-1,1))
axis(1, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)
axis(2, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)</code></pre>
<p><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
<pre class="r"><code>col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
      rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
image(as.matrix(ash_cor_mat[order_index,order_index, indices[2]]),
      col=col, main=paste0(&quot;corshrink: &quot;), cex.main=2,
      xaxt = &quot;n&quot;, yaxt = &quot;n&quot;, zlim=c(-1,1))
axis(1, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)
axis(2, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)</code></pre>
<p><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-13-1.png" width="672" /></p>
<pre class="r"><code>col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
      rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
image(as.matrix(ash_cor_mat[order_index,order_index, indices[3]]),
      col=col, main=paste0(&quot;corshrink: &quot;), cex.main=2,
      xaxt = &quot;n&quot;, yaxt = &quot;n&quot;, zlim=c(-1,1))
axis(1, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)
axis(2, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)</code></pre>
<p><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-14-1.png" width="672" /></p>
<pre class="r"><code>col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
      rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
image(as.matrix(ash_cor_mat[order_index,order_index, indices[4]]),
      col=col, main=paste0(&quot;corshrink: &quot;), cex.main=2,
      xaxt = &quot;n&quot;, yaxt = &quot;n&quot;, zlim=c(-1,1))
axis(1, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)
axis(2, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)</code></pre>
<p><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-15-1.png" width="672" /></p>
<pre class="r"><code>col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
      rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
image(as.matrix(ash_cor_mat[order_index,order_index, indices[5]]),
      col=col, main=paste0(&quot;corshrink: &quot;), cex.main=2,
      xaxt = &quot;n&quot;, yaxt = &quot;n&quot;, zlim=c(-1,1))
axis(1, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)
axis(2, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)</code></pre>
<p><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-16-1.png" width="672" /></p>
<pre class="r"><code>col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
      rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
image(as.matrix(ash_cor_mat[order_index,order_index, indices[6]]),
      col=col, main=paste0(&quot;corshrink: &quot;), cex.main=2,
      xaxt = &quot;n&quot;, yaxt = &quot;n&quot;, zlim=c(-1,1))
axis(1, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)
axis(2, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)</code></pre>
<p><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-17-1.png" width="672" /></p>
</div>
<div id="correlation-scatter-plot-fibroblasts" class="section level2">
<h2>Correlation scatter plot (Fibroblasts)</h2>
<div id="ensg00000130513-1" class="section level3">
<h3>ENSG00000130513</h3>
<pre class="r"><code>##  23 is the index for fibroblasts in person_tissue_genes tissue labels
par(mfrow=c(3,3))
for(j in 1:53){
  plot(person_tissue_genes[,23, indices[[2]]], person_tissue_genes[,j, indices[2]], col=&quot;red&quot;, pch=20, cex=1, xlab = &quot;Fibroblasts&quot;,
     ylab = dimnames(person_tissue_genes)[[2]][j])
}</code></pre>
<p><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-18-1.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-18-2.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-18-3.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-18-4.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-18-5.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-18-6.png" width="672" /></p>
</div>
<div id="ensg00000185088-1" class="section level3">
<h3>ENSG00000185088</h3>
<pre class="r"><code>##  23 is the index for fibroblasts in person_tissue_genes tissue labels
par(mfrow=c(3,3))
for(j in 1:53){
  plot(person_tissue_genes[,23, indices[[4]]], person_tissue_genes[,j, indices[2]], col=&quot;red&quot;, pch=20, cex=1, xlab = &quot;Fibroblasts&quot;,
     ylab = dimnames(person_tissue_genes)[[2]][j])
}</code></pre>
<p><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-19-1.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-19-2.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-19-3.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-19-4.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-19-5.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-19-6.png" width="672" /></p>
</div>
</div>
<div id="correlation-scatter-plot-lymphoblasts" class="section level2">
<h2>Correlation scatter plot (Lymphoblasts)</h2>
<div id="ensg00000130513-2" class="section level3">
<h3>ENSG00000130513</h3>
<pre class="r"><code>##  22 is the index for lymphoblasts in person_tissue_genes tissue labels
par(mfrow=c(3,3))
for(j in 1:53){
  plot(person_tissue_genes[,22, indices[[2]]], person_tissue_genes[,j, indices[2]], col=&quot;red&quot;, pch=20, cex=1, xlab = dimnames(person_tissue_genes)[[2]][22],
     ylab = dimnames(person_tissue_genes)[[2]][j])
}</code></pre>
<p><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-20-1.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-20-2.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-20-3.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-20-4.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-20-5.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-20-6.png" width="672" /></p>
</div>
<div id="ensg00000185088-2" class="section level3">
<h3>ENSG00000185088</h3>
<pre class="r"><code>##  23 is the index for fibroblasts in person_tissue_genes tissue labels
par(mfrow=c(3,3))
for(j in 1:53){
  plot(person_tissue_genes[,22, indices[[4]]], person_tissue_genes[,j, indices[2]], col=&quot;red&quot;, pch=20, cex=1, xlab = dimnames(person_tissue_genes)[[2]][22],
     ylab = dimnames(person_tissue_genes)[[2]][j])
}</code></pre>
<p><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-21-1.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-21-2.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-21-3.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-21-4.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-21-5.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-21-6.png" width="672" /></p>
<p>It seems that for fibroblasts and lymphocytes, there seems to be slightly negative to 0 correlation between the tissues based on the samples for the genes of interest. We have to make sure</p>
</div>
</div>
<div id="correlation-plots-for-brain-spinal-cord" class="section level2">
<h2>Correlation plots for brain spinal cord</h2>
<div id="ensg00000130513-3" class="section level3">
<h3>ENSG00000130513</h3>
<pre class="r"><code>##  19 is the index for brain spinal cord in person_tissue_genes tissue labels
par(mfrow=c(3,3))
for(j in 1:53){
  plot(person_tissue_genes[,19, indices[[2]]], person_tissue_genes[,j, indices[2]], col=&quot;red&quot;, pch=20, cex=1, xlab = dimnames(person_tissue_genes)[[2]][19],
     ylab = dimnames(person_tissue_genes)[[2]][j])
}</code></pre>
<p><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-22-1.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-22-2.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-22-3.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-22-4.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-22-5.png" width="672" /><img src="p53_p63_analysis_files/figure-html/unnamed-chunk-22-6.png" width="672" /></p>
</div>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
