<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Kushal K Dey" />


<title>Reads expression vs CorShrink</title>

<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>




<link rel="stylesheet" href="cosmo.css" type="text/css" />

</head>

<body>


<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <link href="https://fonts.googleapis.com/css?family=Open+Sans:100,300,400,600" rel="stylesheet" type="text/css">
    <link href="../docs/cosmo.css" rel="stylesheet">
    <link href="../docs/cosmo.min.css" rel="stylesheet">
</head>
<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script>
<script src="../docs/bootstrap/js/bootstrap.min.js"></script>

<body>

<div class = "nav" align = "middle">
<div class= "p1"></div>
<p>
  <a class="btn btn-large btn btn-primary" type="button" href = "corshrink_front.html">HOME</a> 
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "index.html">WORKFLOW</a> 
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "tissue_corr.html">GTEx DEMO</a> 
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "about.html">METHODS</a> 
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "https://github.com/kkdey/CorShrink">SOFTWARE</a> 
  &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "license.html">LICENSE</a> 
   &nbsp &nbsp
  <a class="btn btn-large btn btn-primary" type="button" href = "https://github.com/kkdey/CorShrink_workflow">GITHUB</a> 
 </p>
</div>  
<!-- <div class="nav">
  <a href="index.html">Home page</a> &nbsp &nbsp
  <a href="musings.html">Musings</a> &nbsp &nbsp
  <a href="town.html">My town</a> &nbsp &nbsp
  <a href="links.html">Links</a>
</div> -->

<!-- Main content -->
</body>
</html>


<h1 class="title toc-ignore">Reads expression vs CorShrink</h1>
<h4 class="author"><em>Kushal K Dey</em></h4>
<h4 class="date"><em>5/18/2017</em></h4>


<div id="TOC">
<ul>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2017-05-19</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> 8b3b17c</p>
<p>In this script, we look at how the correlations between tissues is affected by the reads expression of the genes and is it true that higher read expression tend to drive higher correlations.</p>
<p>We normalized the reads expression data by library size to obtain the log CPM data and then looked at the correlations between the normalized expression.</p>
<p>The log CPM normalized data per person per tissue for each gene.</p>
<pre class="r"><code>person_tissue_genes &lt;- get(load(&quot;../output/person_tissue_genes_voom.rda&quot;))</code></pre>
<p>Next we load the CorShrink data (Pearson correlation).</p>
<pre class="r"><code>corshrink_data &lt;- get(load(&quot;../output/ash_cor_only_voom_pearson_gtex_tissues.rda&quot;))
cor_data &lt;- get(load(&quot;../output/cor_tissues_non_ash_voom_pearson.rda&quot;))</code></pre>
<pre class="r"><code>gene_names &lt;- as.character(read.table(file = &quot;../data/GTEX_V6/gene_names_GTEX_V6.txt&quot;)[,1])
gene_names_1 &lt;- as.character(sapply(gene_names, function(x) return(strsplit(x, &quot;[.]&quot;)[[1]][1])))
 
person_label=read.table(&quot;../data/GTEX_V6/person_identifier_labels_with_numbers.txt&quot;);
samples_id &lt;- read.table(file = &quot;../data/GTEX_V6/samples_id.txt&quot;)[,1]

samples_person &lt;- sapply(samples_id, function(x) return(paste0(strsplit(as.character(x), &quot;-&quot;)[[1]][1:2], collapse =&quot;-&quot;)))
tissue_labels &lt;- read.table(file = &quot;../data/GTEX_V6/samples_id.txt&quot;)[,3]

unique_persons &lt;- unique(samples_person)
unique_tissues &lt;- unique(tissue_labels)</code></pre>
<pre class="r"><code>plot.new()
num &lt;- grep(&quot;ENSG00000244734&quot;, gene_names_1)
par(mfrow=c(1,1))
col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
          rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
    image(diag(1, 53) + cor_data[order_index, order_index,num],
          col=col, main=paste0(&quot;corr structure: &quot;, gene_names_1[num]), cex.main=0.7,
          xaxt = &quot;n&quot;, yaxt = &quot;n&quot;, zlim=c(-1,+1))
    axis(1, at = seq(0, 1, length.out = 53), labels = unique_tissues[order_index], las=2, cex.axis = 0.5)
    axis(2, at = seq(0, 1, length.out = 53), labels = unique_tissues[order_index], las=2, cex.axis = 0.5)</code></pre>
<p><img src="figure/reads_expression_CorShrink.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot.new()
par(mfrow=c(1,1))
col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
          rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
    image(corshrink_data[order_index, order_index,num],
          col=col, main=paste0(&quot;corr structure: &quot;, gene_names_1[num]), cex.main=0.7,
          xaxt = &quot;n&quot;, yaxt = &quot;n&quot;, zlim=c(-1,+1))
    axis(1, at = seq(0, 1, length.out = 53), labels = unique_tissues[order_index], las=2, cex.axis = 0.5)
    axis(2, at = seq(0, 1, length.out = 53), labels = unique_tissues[order_index], las=2, cex.axis = 0.5)</code></pre>
<p><img src="figure/reads_expression_CorShrink.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>We get a lot of red !! Is this for real. We take up a couple of tissues to test.</p>
<pre class="r"><code>mat &lt;- person_tissue_genes[,,num]
par(mfrow=c(6,2))
k &lt;- 53
l &lt;- 36
plot(mat[,k], mat[,l], col=&quot;red&quot;, pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0(&quot;corshrink est: &quot;, round(corshrink_data[k,l,num], 2)))

k &lt;- 53
l &lt;- 14
plot(mat[,k], mat[,l], col=&quot;red&quot;, pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0(&quot;corshrink est: &quot;, round(corshrink_data[k,l,num], 2)))

k &lt;- 53
l &lt;- 8
plot(mat[,k], mat[,l], col=&quot;red&quot;, pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0(&quot;corshrink est: &quot;, round(corshrink_data[k,l,num], 2)))

k &lt;- 53
l &lt;- 47
plot(mat[,k], mat[,l],  col=&quot;red&quot;, pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0(&quot;corshrink est: &quot;, round(corshrink_data[k,l,num], 2)))

k &lt;- 53
l &lt;- 23
plot(mat[,k], mat[,l], col=&quot;red&quot;, pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0(&quot;corshrink est: &quot;, round(corshrink_data[k,l,num], 2)))

k &lt;- 53
l &lt;- 22
plot(mat[,k], mat[,l], col=&quot;red&quot;, pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0(&quot;corshrink est: &quot;, round(corshrink_data[k,l,num], 2)))

k &lt;- 13
l &lt;- 14
plot(mat[,k], mat[,l], col=&quot;red&quot;, pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0(&quot;corshrink est: &quot;, round(corshrink_data[k,l,num], 2)))

k &lt;- 12
l &lt;- 11
plot(mat[,k], mat[,l], col=&quot;red&quot;, pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0(&quot;corshrink est: &quot;, round(corshrink_data[k,l,num], 2)))


k &lt;- 41
l &lt;- 1
plot(mat[,k], mat[,l], col=&quot;red&quot;, pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0(&quot;corshrink est: &quot;, round(corshrink_data[k,l,num], 2)))

k &lt;- 41
l &lt;- 8
plot(mat[,k], mat[,l], col=&quot;red&quot;, pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0(&quot;corshrink est: &quot;, round(cor_data[k,l,num], 2)))

k &lt;- 53
l &lt;- 41
plot(mat[,k], mat[,l], col=&quot;red&quot;, pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0(&quot;corshrink est: &quot;, round(cor_data[k,l,num], 2)))

k &lt;- 41
l &lt;- 47
plot(mat[,k], mat[,l], col=&quot;red&quot;, pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0(&quot;corshrink est: &quot;, round(cor_data[k,l,num], 2)))</code></pre>
<p><img src="figure/reads_expression_CorShrink.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Do we get back the gene expression distribution as expected from the GTEx plot.</p>
<p>Now we look at another gene, a testis specific gene called PRM1 (ENSG00000175646).</p>
<pre class="r"><code>plot.new()
num &lt;- grep(&quot;ENSG00000175646&quot;, gene_names_1)
par(mfrow=c(1,1))
col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
          rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
    image(diag(1, 53) + cor_data[order_index, order_index,num],
          col=col, main=paste0(&quot;corr structure: &quot;, gene_names_1[num]), cex.main=0.7,
          xaxt = &quot;n&quot;, yaxt = &quot;n&quot;, zlim=c(-1,+1))
    axis(1, at = seq(0, 1, length.out = 53), labels = unique_tissues[order_index], las=2, cex.axis = 0.5)
    axis(2, at = seq(0, 1, length.out = 53), labels = unique_tissues[order_index], las=2, cex.axis = 0.5)</code></pre>
<p><img src="figure/reads_expression_CorShrink.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot.new()
par(mfrow=c(1,1))
col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
          rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
    image(corshrink_data[order_index, order_index,num],
          col=col, main=paste0(&quot;corr structure: &quot;, gene_names_1[num]), cex.main=0.7,
          xaxt = &quot;n&quot;, yaxt = &quot;n&quot;, zlim=c(-1,+1))
    axis(1, at = seq(0, 1, length.out = 53), labels = unique_tissues[order_index], las=2, cex.axis = 0.5)
    axis(2, at = seq(0, 1, length.out = 53), labels = unique_tissues[order_index], las=2, cex.axis = 0.5)</code></pre>
<p><img src="figure/reads_expression_CorShrink.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>mat &lt;- person_tissue_genes[,,num]
par(mfrow=c(4,2))
k &lt;- 49
l &lt;- 22
plot(mat[,k], mat[,l], col=&quot;red&quot;, pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0(&quot;corshrink est: &quot;, round(corshrink_data[k,l,num], 2)))

k &lt;- 49
l &lt;- 53
plot(mat[,k], mat[,l], col=&quot;red&quot;, pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0(&quot;corshrink est: &quot;, round(corshrink_data[k,l,num], 2)))

k &lt;- 49
l &lt;- 14
plot(mat[,k], mat[,l], col=&quot;red&quot;, pch=20, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0(&quot;corshrink est: &quot;, round(corshrink_data[k,l,num], 2)))

k &lt;- 14
l &lt;- 13
plot(mat[,k], mat[,l],  col=&quot;red&quot;, pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0(&quot;corshrink est: &quot;, round(corshrink_data[k,l,num], 2)))

k &lt;- 8
l &lt;- 14
plot(mat[,k], mat[,l],  col=&quot;red&quot;, pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0(&quot;corshrink est: &quot;, round(corshrink_data[k,l,num], 2)))


k &lt;- 16
l &lt;- 12
plot(mat[,k], mat[,l],  col=&quot;red&quot;, pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0(&quot;corshrink est: &quot;, round(corshrink_data[k,l,num], 2)))</code></pre>
<p><img src="figure/reads_expression_CorShrink.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>The final example we consider os of the MBP gene (ENSG00000197971).</p>
<pre class="r"><code>plot.new()
num &lt;- grep(&quot;ENSG00000197971&quot;, gene_names_1)
par(mfrow=c(1,1))
col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
          rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
    image(diag(1, 53) + cor_data[order_index, order_index,num],
          col=col, main=paste0(&quot;corr structure: &quot;, gene_names_1[num]), cex.main=0.7,
          xaxt = &quot;n&quot;, yaxt = &quot;n&quot;, zlim=c(-1,+1))
    axis(1, at = seq(0, 1, length.out = 53), labels = unique_tissues[order_index], las=2, cex.axis = 0.5)
    axis(2, at = seq(0, 1, length.out = 53), labels = unique_tissues[order_index], las=2, cex.axis = 0.5)</code></pre>
<p><img src="figure/reads_expression_CorShrink.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot.new()
par(mfrow=c(1,1))
col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
          rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
    image(corshrink_data[order_index, order_index,num],
          col=col, main=paste0(&quot;corr structure: &quot;, gene_names_1[num]), cex.main=0.7,
          xaxt = &quot;n&quot;, yaxt = &quot;n&quot;, zlim=c(-1,+1))
    axis(1, at = seq(0, 1, length.out = 53), labels = unique_tissues[order_index], las=2, cex.axis = 0.5)
    axis(2, at = seq(0, 1, length.out = 53), labels = unique_tissues[order_index], las=2, cex.axis = 0.5)</code></pre>
<p><img src="figure/reads_expression_CorShrink.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>mat &lt;- person_tissue_genes[,,num]
par(mfrow=c(4,2))
k &lt;- 8
l &lt;- 12
plot(mat[,k], mat[,l], col=&quot;red&quot;, pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0(&quot;corshrink est: &quot;, round(corshrink_data[k,l,num], 2)))

k &lt;- 32
l &lt;- 33
plot(mat[,k], mat[,l], col=&quot;red&quot;, pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0(&quot;corshrink est: &quot;, round(corshrink_data[k,l,num], 2)))

k &lt;- 31
l &lt;- 7
plot(mat[,k], mat[,l], col=&quot;red&quot;, pch=20, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0(&quot;corshrink est: &quot;, round(corshrink_data[k,l,num], 2)))

k &lt;- 35
l &lt;- 24
plot(mat[,k], mat[,l],  col=&quot;red&quot;, pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0(&quot;corshrink est: &quot;, round(corshrink_data[k,l,num], 2)))

k &lt;- 15
l &lt;- 19
plot(mat[,k], mat[,l],  col=&quot;red&quot;, pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0(&quot;corshrink est: &quot;, round(corshrink_data[k,l,num], 2)))

k &lt;- 12
l &lt;- 19
plot(mat[,k], mat[,l],  col=&quot;red&quot;, pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0(&quot;corshrink est: &quot;, round(corshrink_data[k,l,num], 2)))</code></pre>
<p><img src="figure/reads_expression_CorShrink.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<!-- Insert the session information into the document -->
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.3.3 (2017-03-06)
Platform: x86_64-apple-darwin13.4.0 (64-bit)
Running under: macOS Sierra 10.12

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] fgsea_1.1.2      Rcpp_0.12.10     softImpute_1.4   Matrix_1.2-8    
[5] knitr_1.15.1     CorShrink_0.99.0 workflowr_0.4.0  rmarkdown_1.5   

loaded via a namespace (and not attached):
 [1] Biobase_2.34.0             httr_1.2.1                
 [3] jsonlite_1.4               splines_3.3.3             
 [5] foreach_1.4.3              gsubfn_0.6-6              
 [7] Formula_1.2-1              expm_0.999-2              
 [9] highr_0.6                  stats4_3.3.3              
[11] latticeExtra_0.6-28        Rsamtools_1.26.1          
[13] yaml_2.1.14                RSQLite_1.1-2             
[15] backports_1.0.5            lattice_0.20-35           
[17] chron_2.3-50               digest_0.6.12             
[19] GenomicRanges_1.26.1       RColorBrewer_1.1-2        
[21] XVector_0.14.0             checkmate_1.8.2           
[23] colorspace_1.3-2           htmltools_0.3.6           
[25] plyr_1.8.4                 XML_3.98-1.6              
[27] biomaRt_2.30.0             zlibbioc_1.20.0           
[29] corpcor_1.6.9              scales_0.4.1              
[31] glasso_1.8                 BiocParallel_1.8.2        
[33] git2r_0.18.0               htmlTable_1.9             
[35] tibble_1.3.1               sqldf_0.4-10              
[37] IRanges_2.8.1              ggplot2_2.2.1             
[39] SummarizedExperiment_1.4.0 GenomicFeatures_1.26.0    
[41] ashr_2.0.5                 nnet_7.3-12               
[43] BiocGenerics_0.20.0        lazyeval_0.2.0            
[45] proto_1.0.0                survival_2.41-3           
[47] magrittr_1.5               memoise_1.1.0             
[49] evaluate_0.10              doParallel_1.0.10         
[51] MASS_7.3-45                foreign_0.8-67            
[53] truncnorm_1.0-7            rsconnect_0.7             
[55] tools_3.3.3                data.table_1.10.4         
[57] REBayes_0.73               stringr_1.2.0             
[59] S4Vectors_0.12.0           munsell_0.4.3             
[61] cluster_2.0.6              AnnotationDbi_1.36.0      
[63] Biostrings_2.42.0          GenomeInfoDb_1.10.1       
[65] rlang_0.1.1.9000           grid_3.3.3                
[67] RCurl_1.95-4.8             iterators_1.0.8           
[69] htmlwidgets_0.8            bitops_1.0-6              
[71] base64enc_0.1-3            gtable_0.2.0              
[73] codetools_0.2-15           curl_2.6                  
[75] DBI_0.6-1                  mygene_1.10.0             
[77] reshape2_1.4.2.9000        R6_2.2.1                  
[79] GenomicAlignments_1.10.1   gridExtra_2.2.1           
[81] rtracklayer_1.34.1         fastmatch_1.1-0           
[83] Hmisc_4.0-2                rprojroot_1.2             
[85] stringi_1.1.5              pscl_1.4.9                
[87] parallel_3.3.3             SQUAREM_2016.8-2          
[89] rpart_4.1-10               acepack_1.4.1             </code></pre>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
