---
title: "Age comparison analysis: GTEx"
author: "Kushal K Dey"
date: "6/3/2017"
output: html_document
---

In this script, we check for the correlation structure between age as a continuous variable (in inches) and gene expression of the samples for the different tissues.

We first load the sample attributes file.

```{r}
betahat_sebetahat <- get(load("../output/age_cor_nonmash.rda"))
sample_attributes <- get(load("../output/sample_attributes_filtered.rda"))
person_tissue_genes <- get(load("../output/person_tissue_genes_voom.rda"))
```

```{r}
betahat_mat <- betahat_sebetahat$betahat
sebetahat_mat <- betahat_sebetahat$sebetahat
pval_mat <- betahat_sebetahat$pval
rsquare_mat <- betahat_sebetahat$rsquare
```

```{r}
tmat <- betahat_mat/sebetahat_mat
abs_tmat <- abs(tmat)
column_sum <- apply(abs_tmat, 2, function(x) return(quantile(x, 0.7)))
sort(column_sum, decreasing = TRUE)[1:10]
indices <- order(column_sum, decreasing = TRUE)[1:10]
```

```{r}
names <- dimnames(person_tissue_genes)[[3]][order(column_sum, decreasing = TRUE)[1:10]]
names
```

## ENSG00000147889

```{r}
tmat[,indices[1]]
```

```{r}
library(vioplot)
par(mfrow = c(3,3))
for(j in 1:53){
#  ll <-lm(person_tissue_genes[,j,indices[3]] ~ sample_attributes$AGE)
  temp1 <- person_tissue_genes[,j,indices[1]]
  plot(temp1, sample_attributes$AGE, ylab = dimnames(person_tissue_genes)[[2]][j])
  lines(loess.smooth(temp1,sample_attributes$AGE, family = "gaussian"), 
        col="red", lty=2, lwd=2)
}
```

## ENSG00000172667

```{r}
tmat[,indices[2]]
```

```{r}
library(vioplot)
par(mfrow = c(3,3))
for(j in 1:53){
#  ll <-lm(person_tissue_genes[,j,indices[3]] ~ sample_attributes$AGE)
  temp1 <- person_tissue_genes[,j,indices[2]]
  plot(temp1, sample_attributes$AGE, ylab = dimnames(person_tissue_genes)[[2]][j])
  lines(loess.smooth(temp1,sample_attributes$AGE), 
        col="red", lty=2, lwd=2)
}
```

## ENSG00000178163

```{r}
tmat[,indices[3]]
```

```{r}
library(vioplot)
par(mfrow = c(3,3))
for(j in 1:53){
#  ll <-lm(person_tissue_genes[,j,indices[3]] ~ sample_attributes$AGE)
  temp1 <- person_tissue_genes[,j,indices[3]]
  plot(temp1, sample_attributes$AGE, ylab = dimnames(person_tissue_genes)[[2]][j])
  lines(loess.smooth(temp1,sample_attributes$AGE), 
        col="red", lty=2, lwd=2)
}
```

## ENSG00000176871

```{r}
tmat[,indices[4]]
```

```{r}
library(vioplot)
par(mfrow = c(3,3))
for(j in 1:53){
#  ll <-lm(person_tissue_genes[,j,indices[3]] ~ sample_attributes$AGE)
  temp1 <- person_tissue_genes[,j,indices[4]]
  plot(temp1, sample_attributes$AGE, ylab = dimnames(person_tissue_genes)[[2]][j])
  lines(loess.smooth(temp1,sample_attributes$AGE), 
        col="red", lty=2, lwd=2)
}
```

## Pathway analysis of Genes with high t-scores

```{r}
names <- dimnames(person_tissue_genes)[[3]][which(column_sum > 3)]
write.table(names, file = "../utilities/age_genes.txt", quote = FALSE, row.names = FALSE, col.names = FALSE)
ensembl = biomaRt::useMart("ensembl", dataset="hsapiens_gene_ensembl", host="www.ensembl.org")
res = biomaRt::getBM(attributes = c("chromosome_name", "ensembl_gene_id", "external_gene_name", "description"), filters = c("ensembl_gene_id"), values = names,  
mart = ensembl)
out <- mygene::queryMany(names,  scopes="ensembl.gene", fields=c("name", "summary"), species="human");
cbind.data.frame(out$name, out$summary)
```

```{r}
names
```

```{r}
age_paths <- read.delim("../utilities/age_pathways.tab")
cbind.data.frame(age_paths$q.value, age_paths$pathway, age_paths$members_input_overlap_geneids)
```

```{r fig.height = 10}
pathway2 <- read.delim("../utilities/age_pathways.tab")
if(length(which(pathway2$q.value < 0.05)) > 2){
  pathway2 <- pathway2[which(pathway2$q.value < 0.05),]
}
maxpath = 75
pathway <- pathway2[1:min(maxpath, dim(pathway2)[1]), ]
ensembl_genes <- pathway$members_input_overlap
pathway_list_genes <- sapply(ensembl_genes, function(x) return(strsplit(as.character(x), "; ")))
all_genes <- Reduce(union, pathway_list_genes)
pathway_mat <- matrix(0, length(pathway_list_genes), length(all_genes))
for(l in 1:dim(pathway_mat)[1]){
  pathway_mat[l, match(pathway_list_genes[[l]], all_genes)] <- 1
}

pathway_bimat <- (pathway_mat)%*%t(pathway_mat)
pathway_bimat_prop <- diag(1/sqrt(diag(pathway_bimat)), dim(pathway_bimat)[1]) %*% pathway_bimat %*% diag(1/sqrt(diag(pathway_bimat)), dim(pathway_bimat)[1])

col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
      rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
heatmap(pathway_bimat_prop, labRow = pathway$pathway, labCol = pathway$pathway, cexRow = 1, cexCol = 1, col = col, zlim = c(-1,1),
        scale = "none", margins = c(5, 5))
```

## Pathway analysis of top genes beyond quantile threshold

```{r}
column_sum_1 <- apply(abs_tmat, 2, function(x) return(quantile(x, 0.7)))
names <- dimnames(person_tissue_genes)[[3]][order(column_sum_1, decreasing = TRUE)[1:50]]
write.table(names, file = "../utilities/age_genes_2.txt", quote = FALSE, row.names = FALSE, col.names = FALSE)
ensembl = biomaRt::useMart("ensembl", dataset="hsapiens_gene_ensembl", host="www.ensembl.org")
res = biomaRt::getBM(attributes = c("chromosome_name", "ensembl_gene_id", "external_gene_name", "description"), filters = c("ensembl_gene_id"), values = names,  
mart = ensembl)
out <- mygene::queryMany(names,  scopes="ensembl.gene", fields=c("name", "summary"), species="human");
cbind.data.frame(out$name, out$summary)
```

```{r}
names
```

```{r}
age_paths <- read.delim("../utilities/age_pathways_2.tab")
cbind.data.frame(age_paths$q.value, age_paths$pathway, age_paths$members_input_overlap_geneids)
```

```{r fig.height = 10}
pathway2 <- read.delim("../utilities/age_pathways_2.tab")
if(length(which(pathway2$q.value < 0.05)) > 2){
  pathway2 <- pathway2[which(pathway2$q.value < 0.05),]
}
maxpath = 75
pathway <- pathway2[1:min(maxpath, dim(pathway2)[1]), ]
ensembl_genes <- pathway$members_input_overlap
pathway_list_genes <- sapply(ensembl_genes, function(x) return(strsplit(as.character(x), "; ")))
all_genes <- Reduce(union, pathway_list_genes)
pathway_mat <- matrix(0, length(pathway_list_genes), length(all_genes))
for(l in 1:dim(pathway_mat)[1]){
  pathway_mat[l, match(pathway_list_genes[[l]], all_genes)] <- 1
}

pathway_bimat <- (pathway_mat)%*%t(pathway_mat)
pathway_bimat_prop <- diag(1/sqrt(diag(pathway_bimat)), dim(pathway_bimat)[1]) %*% pathway_bimat %*% diag(1/sqrt(diag(pathway_bimat)), dim(pathway_bimat)[1])

col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
      rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
heatmap(pathway_bimat_prop, labRow = pathway$pathway, labCol = pathway$pathway, cexRow = 1, cexCol = 1, col = col, zlim = c(-1,1),
        scale = "none", margins = c(5, 5))
```


