---
title: "Circadian comparison with ash"
author: "Kushal K Dey"
date: "6/8/2017"
output: html_document
---

In this script, we look at the death time comparison problem in the context of using lfsr for the adaptive shrinkage on the tissues for each gene. Adaptive shrinkage is performed using a normal distribution centered about 0. 

```{r}
ash_cor_mat <- get(load("../output/ash_cor_only_voom_pearson_halfuniform_gtex_tissues.rda"))
```

```{r}
common_samples <- get(load("../output/common_samples.rda"))
tissue_labels <- read.table(file = "../data/GTEX_V6/samples_id.txt")[,3]

#library(data.table)
#data <- data.frame(fread("../data/GTEX_V6/cis_gene_expression.txt"))
#matdata <- t(data[,-c(1,2)])

gene_names <- as.character(read.table(file = "../data/GTEX_V6/gene_names_GTEX_V6.txt")[,1])
gene_names_1 <- as.character(sapply(gene_names, function(x) return(strsplit(x, "[.]")[[1]][1])))


U <- unique(tissue_labels)
```

```{r echo=FALSE, eval=TRUE}
order_index <- c();
order_index <- c(order_index, grep("Brain", U))
order_index <- c(order_index, grep("fibroblast", U))
order_index <- c(order_index, grep("EBV", U))
order_index <- c(order_index, grep("Spleen", U))
order_index <- c(order_index, grep("Whole Blood", U))
order_index <- c(order_index, grep("Muscle - Skeletal", U))
order_index <- c(order_index, grep("Liver", U))
order_index <- c(order_index, grep("Pancreas", U))
order_index <- c(order_index, grep("Stomach", U))
order_index <- c(order_index, grep("Kidney - Cortex", U))
order_index <- c(order_index, grep("Adrenal Gland", U))
order_index <- c(order_index, grep("Colon - Transverse", U))
order_index <- c(order_index, grep("Small Intestine - Terminal Ileum", U))
order_index <- c(order_index, grep("Heart - Atrial Appendage", U))
order_index <- c(order_index, grep("Heart - Left Ventricle", U))
order_index <- c(order_index, grep("Minor Salivary Gland", U))
order_index <- c(order_index, grep("Skin - Sun Exposed", U))
order_index <- c(order_index, grep("Skin - Not Sun Exposed", U))
order_index <- c(order_index, grep("Lung", U))
order_index <- c(order_index, grep("Ovary", U))
order_index <- c(order_index, grep("Thyroid", U))
order_index <- c(order_index, grep("Pituitary", U))
order_index <- c(order_index, grep("Testis", U))
order_index <- c(order_index, grep("Nerve - Tibial", U))
order_index <- c(order_index, grep("Breast - Mammary Tissue", U))
order_index <- c(order_index, grep("Adipose - Visceral", U))
order_index <- c(order_index, grep("Adipose - Subcutaneous", U))
order_index <- c(order_index, grep("Artery - Coronary", U))
order_index <- c(order_index, grep("Artery - Tibial", U))
order_index <- c(order_index, grep("Artery - Aorta", U))
order_index <- c(order_index, grep("Esophagus - Mucosa", U))
order_index <- c(order_index, grep("Vagina", U))
order_index <- c(order_index, grep("Cervix - Endocervix", U))
order_index <- c(order_index, grep("Esophagus - Gastroesophageal Junction", U))
order_index <- c(order_index, grep("Colon - Sigmoid", U))
order_index <- c(order_index, grep("Esophagus - Muscularis", U))
order_index <- c(order_index, grep("Cervix - Ectocervix", U))
order_index <- c(order_index, grep("Fallopian", U))
order_index <- c(order_index, grep("Prostate", U))
order_index <- c(order_index, grep("Uterus", U))
order_index <- c(order_index, grep("Bladder", U))
```

```{r}
person_tissue_genes <- get(load("../output/person_tissue_genes_voom.rda"))
betahat_sebetahat <- get(load("../output/death_time_cor_ash.rda"))
sample_attributes <- get(load("../output/sample_attributes_filtered.rda"))
person_tissue_genes <- get(load("../output/person_tissue_genes_voom.rda"))
```

```{r}
betahat_mat <- betahat_sebetahat$post.mean
sebetahat_mat <- betahat_sebetahat$post.sd
lfsr_mat <- betahat_sebetahat$post.lfsr
```

##  lfsr threshold 0.2

We look at which genes have the most number of lfsr values below a chosen threshold.

```{r}
num_lfsr_below_thresh <- apply(lfsr_mat, 2, function(x) return(length(which(x < 0.3))))

imp_genes <- order(num_lfsr_below_thresh, decreasing = TRUE)[1:10]
num_lfsr_below_thresh[imp_genes]

names <- colnames(betahat_mat)[imp_genes]
```

```{r}
write.table(names, file = "../utilities/circadian_ash/circadian_genes_ash_0_3.txt", quote = FALSE, row.names = FALSE, col.names = FALSE)
out <- mygene::queryMany(names,  scopes="ensembl.gene", fields=c("name", "summary"), species="human");
cbind.data.frame(out$name, out$summary)
```
```{r}
pathway2 <- read.delim("../utilities/circadian_ash/circadian_genes_ash_0_3_pathways.tab")
if(length(which(pathway2$q.value < 0.05)) > 2){
  pathway2 <- pathway2[which(pathway2$q.value < 0.05),]
}
maxpath = 75
pathway <- pathway2[1:min(maxpath, dim(pathway2)[1]), ]
ensembl_genes <- pathway$members_input_overlap
pathway_list_genes <- sapply(ensembl_genes, function(x) return(strsplit(as.character(x), "; ")))
all_genes <- Reduce(union, pathway_list_genes)
pathway_mat <- matrix(0, length(pathway_list_genes), length(all_genes))
for(l in 1:dim(pathway_mat)[1]){
  pathway_mat[l, match(pathway_list_genes[[l]], all_genes)] <- 1
}

pathway_bimat <- (pathway_mat)%*%t(pathway_mat)
pathway_bimat_prop <- diag(1/sqrt(diag(pathway_bimat)), dim(pathway_bimat)[1]) %*% pathway_bimat %*% diag(1/sqrt(diag(pathway_bimat)), dim(pathway_bimat)[1])

col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
      rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
heatmap(pathway_bimat_prop, labRow = pathway$pathway, labCol = pathway$pathway, cexRow = 1, cexCol = 1, col = col, zlim = c(-1,1),
        scale = "none", margins = c(5, 5))
```

##  lfsr threshold 0.5

```{r}
num_lfsr_below_thresh <- apply(lfsr_mat, 2, function(x) return(length(which(x < 0.5))))

imp_genes <- order(num_lfsr_below_thresh, decreasing = TRUE)[1:10]
num_lfsr_below_thresh[imp_genes]

names <- colnames(betahat_mat)[imp_genes]
```

```{r}
write.table(names, file = "../utilities/circadian_ash/circadian_genes_ash_0_5.txt", quote = FALSE, row.names = FALSE, col.names = FALSE)
out <- mygene::queryMany(names,  scopes="ensembl.gene", fields=c("name", "summary"), species="human");
cbind.data.frame(out$name, out$summary)
```
```{r}
pathway2 <- read.delim("../utilities/circadian_ash/circadian_genes_ash_0_5_pathways.tab")
if(length(which(pathway2$q.value < 0.05)) > 2){
  pathway2 <- pathway2[which(pathway2$q.value < 0.05),]
}
maxpath = 75
pathway <- pathway2[1:min(maxpath, dim(pathway2)[1]), ]
ensembl_genes <- pathway$members_input_overlap
pathway_list_genes <- sapply(ensembl_genes, function(x) return(strsplit(as.character(x), "; ")))
all_genes <- Reduce(union, pathway_list_genes)
pathway_mat <- matrix(0, length(pathway_list_genes), length(all_genes))
for(l in 1:dim(pathway_mat)[1]){
  pathway_mat[l, match(pathway_list_genes[[l]], all_genes)] <- 1
}

pathway_bimat <- (pathway_mat)%*%t(pathway_mat)
pathway_bimat_prop <- diag(1/sqrt(diag(pathway_bimat)), dim(pathway_bimat)[1]) %*% pathway_bimat %*% diag(1/sqrt(diag(pathway_bimat)), dim(pathway_bimat)[1])

col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
      rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
heatmap(pathway_bimat_prop, labRow = pathway$pathway, labCol = pathway$pathway, cexRow = 1, cexCol = 1, col = col, zlim = c(-1,1),
        scale = "none", margins = c(5, 5))
```

## Review - Important genes


The important genes are **ENSG00000133794 or ARNTL**,  **ENSG00000132326 or PER2**, **ENSG00000159208** or **CIART**, **ENSG00000101194** or **SLC17A7** and  **ENSG0000025513**.

ARNTL encodes for a transcription factor with a basic helix-loop-helix (bHLH) and a PAS domain. The human ARNTL gene has a predicted 24 exons and is located on the p15 band of the 11th chromosome.[3] The BMAL1 protein is 626 amino acids long and plays a key role as one of the positive elements in the mammalian autoregulatory transcription translation negative feedback loop (TTFL), which is responsible for generating molecular circadian rhythms.
One of ARNTL protein's earliest discovered functions in circadian regulation was related to the CLOCK-BMAL1 (CLOCK-ARNTL) heterodimer, which would bind through an E-box enhancer to activate the transcription of the gene encoding vasopressin. [Article](https://en.wikipedia.org/wiki/ARNTL).

Circadian variations of clock gene Per2 and cell cycle genes in different stages of carcinogenesis in golden hamster buccal mucosa. Circadian (24-hour) rhythms and cell division are fundamental biological systems in most organisms. There is substantial evidence that, in mammals, circadian rhythms affect the timing of cell divisions in vivo.
Day-night variations in both the mitotic index and DNA synthesis occur in many tissues
(e.g., oral mucosa, tongue keratinocytes, intestinal epithelium, skin, and bone marrow)
some of which persist even in constant darkness. However, how the circadian
clock controls the timing of cell divisions is not known. [Article](https://www.wikigenes.org/e/gene/e/8864.html), [Article](https://en.wikipedia.org/wiki/PER2).

To determine whether the circadian clock continues to function during the massive
entry of hepatocytes into the cell cycle, we examined expression profiles of clock
components PER1, PER2, CRY1, and Bmal1 (ARNTL). [Article](https://www.nature.com/articles/srep09997).


The clock gene Per2 links the circadian system to the estrogen receptor. [Article](http://www.nature.com/onc/journal/v26/n57/full/1210585a.html).

CIART is a circadian associated repressor of transcription. [Article](https://www.wikigenes.org/e/gene/e/148523.html). I did not find much literature on CIART. But it lies in the superpathway of Circadian entrainment. [Source](http://www.genecards.org/cgi-bin/carddisp.pl?gene=SLC17A7).


```{r}
times <- strsplit(as.character(sample_attributes$DTHTIME), "[:]")
time_stamp <- unlist(lapply(times, function(x){
  y <- as.numeric(x[1])
  z <- as.numeric(x[2])
  w <- y*60+z
  return(w)
}))/(24*60)

cos_times <- cos(2*pi*time_stamp)
uncos_times <- 2*pi*time_stamp
na_indices <- which(is.na(cos_times))

person_tissue_genes_1 <- person_tissue_genes[-na_indices,,]
cos_times_1 <- cos_times[-na_indices]
uncos_times_1 <- uncos_times[-na_indices]
```


## ENSG00000133794 or ARNTL

Check the read expression profile in GTEx here 

[ENSG00000133794 or ARNTL](https://gtexportal.org/home/gene/ENSG00000133794)

```{r}
betahat_sebetahat_2 <- get(load("../output/death_time_cor_nonmash.rda"))
tmat <- betahat_sebetahat_2$betahat/betahat_sebetahat_2$sebetahat
```

```{r}
indices <- match("ENSG00000133794", colnames(betahat_mat))
```


```{r}
plot(tmat[, indices], col="red")
abline(0,0)
```

Which are the tissues with lowest lfsr 

```{r}
sort(lfsr_mat[,indices], decreasing = FALSE)[1:5]
```

```{r}
par(mfrow=c(3,3))
for(j in 1:53){
  plot(uncos_times_1, person_tissue_genes_1[,j, indices],
       ylab = dimnames(person_tissue_genes)[[2]][j])
  lines(loess.smooth(uncos_times_1,person_tissue_genes_1[,j, indices[1]]), 
        col="red", lty=2, lwd=2)
}
```

```{r}
col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
      rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
image(as.matrix(ash_cor_mat[order_index,order_index, indices]),
      col=col, main=paste0("corshrink: "), cex.main=2,
      xaxt = "n", yaxt = "n", zlim=c(-1,1))
axis(1, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)
axis(2, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)
```


## ENSG00000132326 or PER2

Check the read expression profile in GTEx here 

[ENSG00000132326 or PER2](https://gtexportal.org/home/gene/ENSG00000132326)

```{r}
betahat_sebetahat_2 <- get(load("../output/death_time_cor_nonmash.rda"))
tmat <- betahat_sebetahat_2$betahat/betahat_sebetahat_2$sebetahat
```

```{r}
indices <- match("ENSG00000132326", colnames(betahat_mat))
```


```{r}
plot(tmat[, indices], col="red")
abline(0,0)
```

Which are the tissues with lowest lfsr 

```{r}
sort(lfsr_mat[,indices], decreasing = FALSE)[1:5]
```


```{r}
par(mfrow=c(3,3))
for(j in 1:53){
  plot(uncos_times_1, person_tissue_genes_1[,j, indices],
       ylab = dimnames(person_tissue_genes)[[2]][j])
  lines(loess.smooth(uncos_times_1,person_tissue_genes_1[,j, indices[1]]), 
        col="red", lty=2, lwd=2)
}
```

```{r}
col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
      rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
image(as.matrix(ash_cor_mat[order_index,order_index, indices]),
      col=col, main=paste0("corshrink: "), cex.main=2,
      xaxt = "n", yaxt = "n", zlim=c(-1,1))
axis(1, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)
axis(2, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)
```

## ENSG00000159208 or CIART

Check the read expression profile in GTEx here 

[ENSG00000159208 or CIART](https://gtexportal.org/home/gene/ENSG00000159208)

```{r}
betahat_sebetahat_2 <- get(load("../output/death_time_cor_nonmash.rda"))
tmat <- betahat_sebetahat_2$betahat/betahat_sebetahat_2$sebetahat
```

```{r}
indices <- match("ENSG00000159208", colnames(betahat_mat))
```


```{r}
plot(tmat[, indices], col="red")
abline(0,0)
```

Which are the tissues with lowest lfsr 

```{r}
sort(lfsr_mat[,indices], decreasing = FALSE)[1:5]
```

```{r}
par(mfrow=c(3,3))
for(j in 1:53){
  plot(uncos_times_1, person_tissue_genes_1[,j, indices],
       ylab = dimnames(person_tissue_genes)[[2]][j])
  lines(loess.smooth(uncos_times_1,person_tissue_genes_1[,j, indices[1]]), 
        col="red", lty=2, lwd=2)
}
```

```{r}
col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
      rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
image(as.matrix(ash_cor_mat[order_index,order_index, indices]),
      col=col, main=paste0("corshrink: "), cex.main=2,
      xaxt = "n", yaxt = "n", zlim=c(-1,1))
axis(1, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)
axis(2, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)
```


## ENSG00000101194 or SLC17A7

Check the read expression profile in GTEx here 

[ENSG00000101194 or SLC17A7](https://gtexportal.org/home/gene/ENSG00000101194)

```{r}
betahat_sebetahat_2 <- get(load("../output/death_time_cor_nonmash.rda"))
tmat <- betahat_sebetahat_2$betahat/betahat_sebetahat_2$sebetahat
```

```{r}
indices <- match("ENSG00000101194", colnames(betahat_mat))
```

```{r}
plot(tmat[, indices], col="red")
abline(0,0)
```

Which are the tissues with lowest lfsr 

```{r}
sort(lfsr_mat[,indices], decreasing = FALSE)[1:5]
```

```{r}
par(mfrow=c(3,3))
for(j in 1:53){
  plot(uncos_times_1, person_tissue_genes_1[,j, indices],
       ylab = dimnames(person_tissue_genes)[[2]][j])
  #lines(loess.smooth(uncos_times_1,person_tissue_genes_1[,j, indices[1]]), 
  #      col="red", lty=2, lwd=2)
}
```

```{r}
col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
      rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
image(as.matrix(ash_cor_mat[order_index,order_index, indices]),
      col=col, main=paste0("corshrink: "), cex.main=2,
      xaxt = "n", yaxt = "n", zlim=c(-1,1))
axis(1, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)
axis(2, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)
```



