---
title: "CorShrink : Adaptive correlation shrinkage with applications"
output:
  html_document:
    toc: false
---




## What is CorShrink? 

`CorShrink` is an adaptive correlation shrinkage algorithm inspired from the adaptive 
shrinkage (*ash*) method of [Stephens 2016](https://academic.oup.com/biostatistics/article/18/2/275/2557030/False-discovery-rates-a-new-deal). It provides a way to efficiently handle large scale missing data while adjusting the degree of shrinkage of the correlations or cosine similarities. Unlike other popular correlation shrinkage algorithms such as *corpcor*, *CovShrinkMat* and *GLASSO*, *CorShrink* does not need to resort to imputing the data before applying shrinkage, thereby making it free of any imputation error. Also unlike other methods which globally shrink the observed correlation matrix to a target, our method has a a specific shrinkage level for each cell of the correlation matrix, making it more flexible. 

## Sneak peek at workflow

In our workflow, we replicate the examples and the analysis referenced in the paper. Primarily, the analysis is focused on comparison `Corshrink` with other correlation shrinkage methods like GLASSO, `corpcor` etc, application of our method to gene wise human tissue-tissue correlation matrix shrinkage based on bulk RNA-seq data from Genotype Tissue Expression (GTEx) project and how dimension reduction methods are related to our shrinkage method. We also present applications to motivate how `CorShrink` can be used to produce more robust rankings from the word2vec output in Natural Language Processing. 

Okay, so lets get started !!


## Detailed workflow 

### Comaprison of CorShrink with corpcor and GLASSO

In this section, we perform simulation experiments to compare out `CorShrink` approach with that
of the more popular approaches to correlation shrinkage - namely `corpcor` and `GLASSO`. We focus on two simulation designs with missing data. We consider two approaches to imputing the missing data. Approach 1 is an ad-hoc way of replacing the missing values by resampling from each tissue distribution over the non-NA samples for each gene. Approach 2 was to use SoftImpute that uses 
a low dimensional structure representation to impute the values.

- [CorShrink vs corpcor vs GLASSO : Approach 1](../docs/corshrink_corpcor_glasso.html)
- [CorShrink vs corpcor vs GLASSO : Approach 2](../docs/CorShrink_softimpute.html)

### CorShrink and CountClust clusters on GTEx data

[Dey et al 2017](http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1006599)
obtained clusters of GTEx tissue samples from the gene expression data using a Grade of Membership approach. Here we try to relate the clusters obtained there and the genes that were found to drive these clusters with their tissue-tissue correlation structure patterns. Also of 
interest is how this correlation structure depends on the expression patterns of the genes.

- [CorShrink of top genes driving GTEx Countclust clusters](../docs/gtex_corshrink.html)
- [Curious case of high correlation in Brain](../docs/corshrink_brain_func.html)
- [GSEA of CorShrink correlation patterns on all tissues ](../docs/gsea_corshrink_corpcor.html)
- [GSEA of CorShrink correlation patterns on brain tissues ](../docs/gsea_corshrink_corpcor_2.html)

### Dimension Reduction and CorShrink

In the GTEx data, CorShrink was previously applied one gene at a time on its tissue-tissue correlation matrix. However, one would assume that there will be sharing of information among genes and we can pool in this shared information while performing shrinkage to get a low dimensionla structyre of the correlation matrix together with shrinkage. To answer that question, we perform PCA, PMD and FLASH on the vectorized CorShrink matrices across genes. For the latter (FLASH), we also compare between FLASH on vectorized CorShrink and sample correlation matrices.

- [Principal Component Analysis on CorShrink matrix](../docs/corshrink_pca.html)
- [Penalized Matrix Decomposition on CorShrink matrix](../docs/corshrink_pmd.html)
- [FLASH on CorShrink matrix](../docs/corshrink_flash.html)
- [FLASH on sample correlation matrix](../docs/cor_flash.html)




