---
title: "p53 and p63 gene analysis"
author: "Kushal K Dey"
date: "6/5/2017"
output: html_document
---

## Getting to know these genes 

In this script, we take a close look at the p53 and p63 genes that were found
to be highly correlated with age in the GTEx data.

The genes related to p53 are *ENSG00000161513*, *ENSG00000130513*, *ENSG00000134574*, *ENSG00000185088*, *ENSG00000147889*, *ENSG00000172667*.

*ENSG00000161513*, also called **FDXR**, sensitizes cells to oxidative stress-induced apoptosis and is regulated by the p53 family [Article](https://www.ncbi.nlm.nih.gov/pubmed/12370809). So, it makes sense it is associated with aging since p53 family plays a role in aging.

*ENSG00000130513* also called **GDF-15** (or growth differentiation factor)
is a p53- and demethylation-upregulating gene represses cell proliferation, invasion, and tumorigenesis in bladder carcinoma cells. [Article](https://www.nature.com/articles/srep12870), [Article2](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1975780/).

*ENSG00000134574* or  Xeroderma Pigmentosum DDB2 **DDB2** gene is activated by p53 only in humans and not in mice. [Article](http://mcb.asm.org/content/22/10/3247).

*ENSG00000185088* or **RPS27L** or Ribosomal protein S27-like and S27 interplay with p53-MDM2 axis s a target, a substrate, and a regulator. [Article](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3077453/).

*ENSG00000147889* or **CDKN2A** . Somatic mutations of CDKN2A are common in the majority of human cancers, with estimates that CDKN2A is the second most commonly inactivated gene in cancer after p53. The gene codes for two proteins, including the INK4 family member p16 (or p16INK4a) and p14arf. Both act as tumor suppressors by regulating the cell cycle. p14ARF (known as p19ARF in the mouse) activates the p53 tumor suppressor.

*ENSG00000172667* orZMAT3 or Wig-1 is a p53 target gene that encodes an RNA-binding zinc-finger protein involved in the regulation of mRNA stability through binding to AU-rich elements (AREs). Wig-1 regulates cell cycle arrest and cell death through the p53 targets FAS. [Article](http://www.nature.com/onc/journal/v33/n35/full/onc2013594a.html).

## Reads expression profile

We take these genes one by one and look for the reads expression across the 
tissues. Here are the plots from the GTEx portal site.

[ENSG00000172667](https://gtexportal.org/home/gene/ENSG00000172667)
[ENSG00000147889](https://gtexportal.org/home/gene/ENSG00000147889)
[ENSG00000185088](https://gtexportal.org/home/gene/ENSG00000185088)
[ENSG00000134574](https://gtexportal.org/home/gene/ENSG00000134574)
[ENSG00000130513](https://gtexportal.org/home/gene/ENSG00000130513)
[ENSG00000161513](https://gtexportal.org/home/gene/ENSG00000161513)

In all these examples, it seems the cell lymphocytes and cell fibroblasts seem to be higher expressed compared to most tissues. 

We know p53 prevent apoptosis and aids in the survival of stem cells. We also know that it plays a role in the survival of the stem cells. Now we know that fibroblasts play a very important role in wound healing which is linked with stem cells, so there seems to be a connection there. 

We also look at the p63 genes in the pathways we found and their reads expression profiles in GTEx. The genes we consider are *ENSG00000161513*, *ENSG00000130513* and *ENSG00000147889*. All these genes are already studied in their patterns above.

We now look at the patterns of expression of each of these genes for the different samples across the 53 tissues of interest.

```{r}
sample_attributes <- get(load("../output/sample_attributes_filtered.rda"))
person_tissue_genes <- get(load("../output/person_tissue_genes_voom.rda"))
```

```{r}
indices <- match(c("ENSG00000161513", "ENSG00000130513",
                   "ENSG00000134574", "ENSG00000185088",
                   "ENSG00000147889", "ENSG00000172667"), dimnames(person_tissue_genes)[[3]])
```

## ENSG00000161513

```{r}
par(mfrow = c(3,3))
for(j in 1:53){
  temp1 <- person_tissue_genes[,j,indices[1]]
  plot(temp1, sample_attributes$AGE, ylab = dimnames(person_tissue_genes)[[2]][j])
  lines(loess.smooth(temp1,sample_attributes$AGE, family = "gaussian"), 
        col="red", lty=2, lwd=2)
}
```

## ENSG00000130513

```{r}
par(mfrow = c(3,3))
for(j in 1:53){
  temp1 <- person_tissue_genes[,j,indices[2]]
  plot(temp1, sample_attributes$AGE, ylab = dimnames(person_tissue_genes)[[2]][j])
  lines(loess.smooth(temp1,sample_attributes$AGE, family = "gaussian"), 
        col="red", lty=2, lwd=2)
}
```

## ENSG00000134574

```{r}
par(mfrow = c(3,3))
for(j in 1:53){
  temp1 <- person_tissue_genes[,j,indices[3]]
  plot(temp1, sample_attributes$AGE, ylab = dimnames(person_tissue_genes)[[2]][j])
  lines(loess.smooth(temp1,sample_attributes$AGE, family = "gaussian"), 
        col="red", lty=2, lwd=2)
}
```

## ENSG00000185088

```{r}
par(mfrow = c(3,3))
for(j in 1:53){
  temp1 <- person_tissue_genes[,j,indices[4]]
  plot(temp1, sample_attributes$AGE, ylab = dimnames(person_tissue_genes)[[2]][j])
  lines(loess.smooth(temp1,sample_attributes$AGE, family = "gaussian"), 
        col="red", lty=2, lwd=2)
}
```

## ENSG00000147889

```{r}
par(mfrow = c(3,3))
for(j in 1:53){
  temp1 <- person_tissue_genes[,j,indices[5]]
  plot(temp1, sample_attributes$AGE, ylab = dimnames(person_tissue_genes)[[2]][j])
  lines(loess.smooth(temp1,sample_attributes$AGE, family = "gaussian"), 
        col="red", lty=2, lwd=2)
}
```

## ENSG00000172667

```{r}
par(mfrow = c(3,3))
for(j in 1:53){
  temp1 <- person_tissue_genes[,j,indices[6]]
  plot(temp1, sample_attributes$AGE, ylab = dimnames(person_tissue_genes)[[2]][j])
  lines(loess.smooth(temp1,sample_attributes$AGE, family = "gaussian"), 
        col="red", lty=2, lwd=2)
}
```

## CorShrink matrices

Now we check how the CorShrink matrices look like for these genes.

```{r}
ash_cor_mat <- get(load("../output/ash_cor_only_voom_pearson_halfuniform_gtex_tissues.rda"))
```

```{r}
common_samples <- get(load("../output/common_samples.rda"))
tissue_labels <- read.table(file = "../data/GTEX_V6/samples_id.txt")[,3]

#library(data.table)
#data <- data.frame(fread("../data/GTEX_V6/cis_gene_expression.txt"))
#matdata <- t(data[,-c(1,2)])

gene_names <- as.character(read.table(file = "../data/GTEX_V6/gene_names_GTEX_V6.txt")[,1])
gene_names_1 <- as.character(sapply(gene_names, function(x) return(strsplit(x, "[.]")[[1]][1])))


U <- unique(tissue_labels)
```

```{r echo=FALSE, eval=TRUE}
order_index <- c();
order_index <- c(order_index, grep("Brain", U))
order_index <- c(order_index, grep("fibroblast", U))
order_index <- c(order_index, grep("EBV", U))
order_index <- c(order_index, grep("Spleen", U))
order_index <- c(order_index, grep("Whole Blood", U))
order_index <- c(order_index, grep("Muscle - Skeletal", U))
order_index <- c(order_index, grep("Liver", U))
order_index <- c(order_index, grep("Pancreas", U))
order_index <- c(order_index, grep("Stomach", U))
order_index <- c(order_index, grep("Kidney - Cortex", U))
order_index <- c(order_index, grep("Adrenal Gland", U))
order_index <- c(order_index, grep("Colon - Transverse", U))
order_index <- c(order_index, grep("Small Intestine - Terminal Ileum", U))
order_index <- c(order_index, grep("Heart - Atrial Appendage", U))
order_index <- c(order_index, grep("Heart - Left Ventricle", U))
order_index <- c(order_index, grep("Minor Salivary Gland", U))
order_index <- c(order_index, grep("Skin - Sun Exposed", U))
order_index <- c(order_index, grep("Skin - Not Sun Exposed", U))
order_index <- c(order_index, grep("Lung", U))
order_index <- c(order_index, grep("Ovary", U))
order_index <- c(order_index, grep("Thyroid", U))
order_index <- c(order_index, grep("Pituitary", U))
order_index <- c(order_index, grep("Testis", U))
order_index <- c(order_index, grep("Nerve - Tibial", U))
order_index <- c(order_index, grep("Breast - Mammary Tissue", U))
order_index <- c(order_index, grep("Adipose - Visceral", U))
order_index <- c(order_index, grep("Adipose - Subcutaneous", U))
order_index <- c(order_index, grep("Artery - Coronary", U))
order_index <- c(order_index, grep("Artery - Tibial", U))
order_index <- c(order_index, grep("Artery - Aorta", U))
order_index <- c(order_index, grep("Esophagus - Mucosa", U))
order_index <- c(order_index, grep("Vagina", U))
order_index <- c(order_index, grep("Cervix - Endocervix", U))
order_index <- c(order_index, grep("Esophagus - Gastroesophageal Junction", U))
order_index <- c(order_index, grep("Colon - Sigmoid", U))
order_index <- c(order_index, grep("Esophagus - Muscularis", U))
order_index <- c(order_index, grep("Cervix - Ectocervix", U))
order_index <- c(order_index, grep("Fallopian", U))
order_index <- c(order_index, grep("Prostate", U))
order_index <- c(order_index, grep("Uterus", U))
order_index <- c(order_index, grep("Bladder", U))
```


```{r}
col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
      rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
image(as.matrix(ash_cor_mat[order_index,order_index, indices[1]]),
      col=col, main=paste0("corshrink: "), cex.main=2,
      xaxt = "n", yaxt = "n", zlim=c(-1,1))
axis(1, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)
axis(2, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)
```


```{r}
col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
      rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
image(as.matrix(ash_cor_mat[order_index,order_index, indices[2]]),
      col=col, main=paste0("corshrink: "), cex.main=2,
      xaxt = "n", yaxt = "n", zlim=c(-1,1))
axis(1, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)
axis(2, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)
```

```{r}
col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
      rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
image(as.matrix(ash_cor_mat[order_index,order_index, indices[3]]),
      col=col, main=paste0("corshrink: "), cex.main=2,
      xaxt = "n", yaxt = "n", zlim=c(-1,1))
axis(1, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)
axis(2, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)
```



```{r}
col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
      rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
image(as.matrix(ash_cor_mat[order_index,order_index, indices[4]]),
      col=col, main=paste0("corshrink: "), cex.main=2,
      xaxt = "n", yaxt = "n", zlim=c(-1,1))
axis(1, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)
axis(2, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)
```


```{r}
col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
      rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
image(as.matrix(ash_cor_mat[order_index,order_index, indices[5]]),
      col=col, main=paste0("corshrink: "), cex.main=2,
      xaxt = "n", yaxt = "n", zlim=c(-1,1))
axis(1, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)
axis(2, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)
```


```{r}
col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
      rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
image(as.matrix(ash_cor_mat[order_index,order_index, indices[6]]),
      col=col, main=paste0("corshrink: "), cex.main=2,
      xaxt = "n", yaxt = "n", zlim=c(-1,1))
axis(1, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)
axis(2, at = seq(0, 1, length.out = 53), labels = U[order_index], las=2, cex.axis = 0.5)
```

## Correlation scatter plot (Fibroblasts)

### ENSG00000130513

```{r}
##  23 is the index for fibroblasts in person_tissue_genes tissue labels
par(mfrow=c(3,3))
for(j in 1:53){
  plot(person_tissue_genes[,23, indices[[2]]], person_tissue_genes[,j, indices[2]], col="red", pch=20, cex=1, xlab = "Fibroblasts",
     ylab = dimnames(person_tissue_genes)[[2]][j])
}
```

### ENSG00000185088

```{r}
##  23 is the index for fibroblasts in person_tissue_genes tissue labels
par(mfrow=c(3,3))
for(j in 1:53){
  plot(person_tissue_genes[,23, indices[[4]]], person_tissue_genes[,j, indices[2]], col="red", pch=20, cex=1, xlab = "Fibroblasts",
     ylab = dimnames(person_tissue_genes)[[2]][j])
}
```


## Correlation scatter plot (Lymphoblasts)

### ENSG00000130513

```{r}
##  22 is the index for lymphoblasts in person_tissue_genes tissue labels
par(mfrow=c(3,3))
for(j in 1:53){
  plot(person_tissue_genes[,22, indices[[2]]], person_tissue_genes[,j, indices[2]], col="red", pch=20, cex=1, xlab = dimnames(person_tissue_genes)[[2]][22],
     ylab = dimnames(person_tissue_genes)[[2]][j])
}
```

### ENSG00000185088

```{r}
##  23 is the index for fibroblasts in person_tissue_genes tissue labels
par(mfrow=c(3,3))
for(j in 1:53){
  plot(person_tissue_genes[,22, indices[[4]]], person_tissue_genes[,j, indices[2]], col="red", pch=20, cex=1, xlab = dimnames(person_tissue_genes)[[2]][22],
     ylab = dimnames(person_tissue_genes)[[2]][j])
}
```

It seems that for fibroblasts and lymphocytes, there seems to be slightly negative to 0 correlation between the tissues based on the samples for the genes of interest. We have to make sure 

## Correlation plots for brain spinal cord

### ENSG00000130513

```{r}
##  19 is the index for brain spinal cord in person_tissue_genes tissue labels
par(mfrow=c(3,3))
for(j in 1:53){
  plot(person_tissue_genes[,19, indices[[2]]], person_tissue_genes[,j, indices[2]], col="red", pch=20, cex=1, xlab = dimnames(person_tissue_genes)[[2]][19],
     ylab = dimnames(person_tissue_genes)[[2]][j])
}
```





