---
title: "Reads expression vs CorShrink"
author: "Kushal K Dey"
date: "5/18/2017"
output: html_document
---

<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
```{r read-chunk, include=FALSE, cache=FALSE}
knitr::read_chunk("chunks.R")
```

<!-- Update knitr chunk options -->
```{r knitr-opts-chunk, include=FALSE}
```

<!-- Insert the date the file was last updated -->
```{r last-updated, echo=FALSE, results='asis'}
```

<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
```{r code-version, echo=FALSE, results='asis'}
```

In this script, we look at how the correlations between tissues is affected by the reads
expression of the genes and is it true that higher read expression tend to drive higher
correlations. 

We normalized the reads expression data by library size to obtain the log CPM data and then 
looked at the correlations between the normalized expression.

The log CPM normalized data per person per tissue for each gene.


```{r}
person_tissue_genes <- get(load("../output/person_tissue_genes_voom.rda"))
```

Next we load the CorShrink data (Pearson correlation).

```{r}
corshrink_data <- get(load("../output/ash_cor_only_voom_pearson_gtex_tissues.rda"))
cor_data <- get(load("../output/cor_tissues_non_ash_voom_pearson.rda"))
```

```{r}
data <- get(load("../output/ash_cor_only_voom_pearson_gtex_tissues.rda"))

gene_names <- as.character(read.table(file = "../data/GTEX_V6/gene_names_GTEX_V6.txt")[,1])
gene_names_1 <- as.character(sapply(gene_names, function(x) return(strsplit(x, "[.]")[[1]][1])))
 
person_label=read.table("../data/GTEX_V6/person_identifier_labels_with_numbers.txt");
samples_id <- read.table(file = "../data/GTEX_V6/samples_id.txt")[,1]

samples_person <- sapply(samples_id, function(x) return(paste0(strsplit(as.character(x), "-")[[1]][1:2], collapse ="-")))
tissue_labels <- read.table(file = "../data/GTEX_V6/samples_id.txt")[,3]

unique_persons <- unique(samples_person)
unique_tissues <- unique(tissue_labels)
```

```{r echo=FALSE, eval=TRUE}
order_index <- c();
U <- unique_tissues
order_index <- c(order_index, grep("Brain", U))
order_index <- c(order_index, grep("fibroblast", U))
order_index <- c(order_index, grep("EBV", U))
order_index <- c(order_index, grep("Spleen", U))
order_index <- c(order_index, grep("Whole Blood", U))
order_index <- c(order_index, grep("Muscle - Skeletal", U))
order_index <- c(order_index, grep("Liver", U))
order_index <- c(order_index, grep("Pancreas", U))
order_index <- c(order_index, grep("Stomach", U))
order_index <- c(order_index, grep("Kidney - Cortex", U))
order_index <- c(order_index, grep("Adrenal Gland", U))
order_index <- c(order_index, grep("Colon - Transverse", U))
order_index <- c(order_index, grep("Small Intestine - Terminal Ileum", U))
order_index <- c(order_index, grep("Heart - Atrial Appendage", U))
order_index <- c(order_index, grep("Heart - Left Ventricle", U))
order_index <- c(order_index, grep("Minor Salivary Gland", U))
order_index <- c(order_index, grep("Skin - Sun Exposed", U))
order_index <- c(order_index, grep("Skin - Not Sun Exposed", U))
order_index <- c(order_index, grep("Lung", U))
order_index <- c(order_index, grep("Ovary", U))
order_index <- c(order_index, grep("Thyroid", U))
order_index <- c(order_index, grep("Pituitary", U))
order_index <- c(order_index, grep("Testis", U))
order_index <- c(order_index, grep("Nerve - Tibial", U))
order_index <- c(order_index, grep("Breast - Mammary Tissue", U))
order_index <- c(order_index, grep("Adipose - Visceral", U))
order_index <- c(order_index, grep("Adipose - Subcutaneous", U))
order_index <- c(order_index, grep("Artery - Coronary", U))
order_index <- c(order_index, grep("Artery - Tibial", U))
order_index <- c(order_index, grep("Artery - Aorta", U))
order_index <- c(order_index, grep("Esophagus - Mucosa", U))
order_index <- c(order_index, grep("Vagina", U))
order_index <- c(order_index, grep("Cervix - Endocervix", U))
order_index <- c(order_index, grep("Esophagus - Gastroesophageal Junction", U))
order_index <- c(order_index, grep("Colon - Sigmoid", U))
order_index <- c(order_index, grep("Esophagus - Muscularis", U))
order_index <- c(order_index, grep("Cervix - Ectocervix", U))
order_index <- c(order_index, grep("Fallopian", U))
order_index <- c(order_index, grep("Prostate", U))
order_index <- c(order_index, grep("Uterus", U))
order_index <- c(order_index, grep("Bladder", U))
```

```{r fig.height = 6}
plot.new()
num <- grep("ENSG00000244734", gene_names_1)
par(mfrow=c(1,1))
col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
          rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
    image(diag(1, 53) + cor_data[order_index, order_index,num],
          col=col, main=paste0("corr structure: ", gene_names_1[num]), cex.main=0.7,
          xaxt = "n", yaxt = "n", zlim=c(-1,+1))
    axis(1, at = seq(0, 1, length.out = 53), labels = unique_tissues[order_index], las=2, cex.axis = 0.5)
    axis(2, at = seq(0, 1, length.out = 53), labels = unique_tissues[order_index], las=2, cex.axis = 0.5)
```

```{r fig.height = 4}
plot.new()
par(mfrow=c(1,1))
col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
          rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
    image(corshrink_data[order_index, order_index,num],
          col=col, main=paste0("corr structure: ", gene_names_1[num]), cex.main=0.7,
          xaxt = "n", yaxt = "n", zlim=c(-1,+1))
    axis(1, at = seq(0, 1, length.out = 53), labels = unique_tissues[order_index], las=2, cex.axis = 0.5)
    axis(2, at = seq(0, 1, length.out = 53), labels = unique_tissues[order_index], las=2, cex.axis = 0.5)

```

We get a lot of red !!  Is this for real. We take up a couple of tissues to test.

```{r fig.height= 12, fig.weight = 4}
mat <- person_tissue_genes[,,num]
par(mfrow=c(6,2))
k <- 53
l <- 36
plot(mat[,k], mat[,l], col="red", pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0("corshrink est: ", round(corshrink_data[k,l,num], 2)))

k <- 53
l <- 14
plot(mat[,k], mat[,l], col="red", pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0("corshrink est: ", round(corshrink_data[k,l,num], 2)))

k <- 53
l <- 8
plot(mat[,k], mat[,l], col="red", pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0("corshrink est: ", round(corshrink_data[k,l,num], 2)))

k <- 53
l <- 47
plot(mat[,k], mat[,l],  col="red", pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0("corshrink est: ", round(corshrink_data[k,l,num], 2)))

k <- 53
l <- 23
plot(mat[,k], mat[,l], col="red", pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0("corshrink est: ", round(corshrink_data[k,l,num], 2)))

k <- 53
l <- 22
plot(mat[,k], mat[,l], col="red", pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0("corshrink est: ", round(corshrink_data[k,l,num], 2)))

k <- 13
l <- 14
plot(mat[,k], mat[,l], col="red", pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0("corshrink est: ", round(corshrink_data[k,l,num], 2)))

k <- 12
l <- 11
plot(mat[,k], mat[,l], col="red", pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0("corshrink est: ", round(corshrink_data[k,l,num], 2)))


k <- 41
l <- 1
plot(mat[,k], mat[,l], col="red", pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0("corshrink est: ", round(corshrink_data[k,l,num], 2)))

k <- 41
l <- 8
plot(mat[,k], mat[,l], col="red", pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0("corshrink est: ", round(cor_data[k,l,num], 2)))

k <- 53
l <- 41
plot(mat[,k], mat[,l], col="red", pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0("corshrink est: ", round(cor_data[k,l,num], 2)))

k <- 41
l <- 47
plot(mat[,k], mat[,l], col="red", pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0("corshrink est: ", round(cor_data[k,l,num], 2)))

```

Do we get back the gene expression distribution as expected from the GTEx plot.

Now we look at another gene, a testis specific gene called PRM1 (ENSG00000175646).

```{r}
plot.new()
num <- grep("ENSG00000175646", gene_names_1)
par(mfrow=c(1,1))
col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
          rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
    image(diag(1, 53) + cor_data[order_index, order_index,num],
          col=col, main=paste0("corr structure: ", gene_names_1[num]), cex.main=0.7,
          xaxt = "n", yaxt = "n", zlim=c(-1,+1))
    axis(1, at = seq(0, 1, length.out = 53), labels = unique_tissues[order_index], las=2, cex.axis = 0.5)
    axis(2, at = seq(0, 1, length.out = 53), labels = unique_tissues[order_index], las=2, cex.axis = 0.5)
```

```{r}
plot.new()
par(mfrow=c(1,1))
col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
          rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
    image(corshrink_data[order_index, order_index,num],
          col=col, main=paste0("corr structure: ", gene_names_1[num]), cex.main=0.7,
          xaxt = "n", yaxt = "n", zlim=c(-1,+1))
    axis(1, at = seq(0, 1, length.out = 53), labels = unique_tissues[order_index], las=2, cex.axis = 0.5)
    axis(2, at = seq(0, 1, length.out = 53), labels = unique_tissues[order_index], las=2, cex.axis = 0.5)

```

```{r fig.height= 12, fig.weight= 6}
mat <- person_tissue_genes[,,num]
par(mfrow=c(4,2))
k <- 49
l <- 22
plot(mat[,k], mat[,l], col="red", pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0("corshrink est: ", round(corshrink_data[k,l,num], 2)))

k <- 49
l <- 53
plot(mat[,k], mat[,l], col="red", pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0("corshrink est: ", round(corshrink_data[k,l,num], 2)))

k <- 49
l <- 14
plot(mat[,k], mat[,l], col="red", pch=20, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0("corshrink est: ", round(corshrink_data[k,l,num], 2)))

k <- 14
l <- 13
plot(mat[,k], mat[,l],  col="red", pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0("corshrink est: ", round(corshrink_data[k,l,num], 2)))

k <- 8
l <- 14
plot(mat[,k], mat[,l],  col="red", pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0("corshrink est: ", round(corshrink_data[k,l,num], 2)))


k <- 16
l <- 12
plot(mat[,k], mat[,l],  col="red", pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0("corshrink est: ", round(corshrink_data[k,l,num], 2)))

```

The final example we consider os of the MBP gene (ENSG00000197971).

```{r}
plot.new()
num <- grep("ENSG00000197971", gene_names_1)
par(mfrow=c(1,1))
col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
          rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
    image(diag(1, 53) + cor_data[order_index, order_index,num],
          col=col, main=paste0("corr structure: ", gene_names_1[num]), cex.main=0.7,
          xaxt = "n", yaxt = "n", zlim=c(-1,+1))
    axis(1, at = seq(0, 1, length.out = 53), labels = unique_tissues[order_index], las=2, cex.axis = 0.5)
    axis(2, at = seq(0, 1, length.out = 53), labels = unique_tissues[order_index], las=2, cex.axis = 0.5)
```

```{r}
plot.new()
par(mfrow=c(1,1))
col=c(rev(rgb(seq(1,0,length=1000),1,seq(1,0,length=1000))),
          rgb(1,seq(1,0,length=1000),seq(1,0,length=1000)))
    image(corshrink_data[order_index, order_index,num],
          col=col, main=paste0("corr structure: ", gene_names_1[num]), cex.main=0.7,
          xaxt = "n", yaxt = "n", zlim=c(-1,+1))
    axis(1, at = seq(0, 1, length.out = 53), labels = unique_tissues[order_index], las=2, cex.axis = 0.5)
    axis(2, at = seq(0, 1, length.out = 53), labels = unique_tissues[order_index], las=2, cex.axis = 0.5)
```

```{r fig.height = 12, fig.weight = 6}
mat <- person_tissue_genes[,,num]
par(mfrow=c(4,2))
k <- 8
l <- 12
plot(mat[,k], mat[,l], col="red", pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0("corshrink est: ", round(corshrink_data[k,l,num], 2)))

k <- 32
l <- 33
plot(mat[,k], mat[,l], col="red", pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0("corshrink est: ", round(corshrink_data[k,l,num], 2)))

k <- 31
l <- 7
plot(mat[,k], mat[,l], col="red", pch=20, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0("corshrink est: ", round(corshrink_data[k,l,num], 2)))

k <- 35
l <- 24
plot(mat[,k], mat[,l],  col="red", pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0("corshrink est: ", round(corshrink_data[k,l,num], 2)))

k <- 15
l <- 19
plot(mat[,k], mat[,l],  col="red", pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0("corshrink est: ", round(corshrink_data[k,l,num], 2)))

k <- 12
l <- 19
plot(mat[,k], mat[,l],  col="red", pch=20, cex = 1.5, xlab = paste0(unique_tissues[k]),
     ylab = paste0(unique_tissues[l]), main = paste0("corshrink est: ", round(corshrink_data[k,l,num], 2)))

```


## Session information

<!-- Insert the session information into the document -->
```{r session-info}
```
