---
title: "CorShrink : Adaptive correlation shrinkage with applications"
output:
  html_document:
    toc: false
---


## What is CorShrink? 

`CorShrink` is an adaptive correlation shrinkage algorithm inspired from the adaptive 
shrinkage (*ash*) method of [Stephens 2016](https://academic.oup.com/biostatistics/article/18/2/275/2557030/False-discovery-rates-a-new-deal). It provides a way to efficiently handle large scale missing data while adjusting the degree of shrinkage of the correlations or cosine similarities. Unlike other popular correlation shrinkage algorithms such as *corpcor*, *CovShrinkMat* and *GLASSO*, *CorShrink* does not need to resort to imputing the data before applying shrinkage, thereby making it free of any imputation error. Also unlike other methods which globally shrink the observed correlation matrix to a target, our method has a a specific shrinkage level for each cell of the correlation matrix, making it more flexible. 

## Sneak peek at workflow

In our workflow, we replicate the examples and the analysis referenced in the paper. Primarily, the analysis is focused on comparison `Corshrink` with other correlation shrinkage methods like GLASSO, `corpcor` etc, application of our method to gene wise human tissue-tissue correlation matrix shrinkage based on bulk RNA-seq data from Genotype Tissue Expression (GTEx) project and how dimension reduction methods are related to our shrinkage method. We also present applications to motivate how `CorShrink` can be used to produce more robust rankings from the word2vec output in Natural Language Processing. 

Okay, so lets get started !!


## Detailed workflow 

### Comaprison of CorShrink with other shrinkage methods

In this section, we perform simulation experiments to compare out `CorShrink` approach with that
of the more popular approaches to correlation shrinkage - namely `corpcor` and `GLASSO`. We focus on two simulation designs with missing data. We consider two approaches to imputing the missing data. Approach 1 is an ad-hoc way of replacing the missing values by resampling from each tissue distribution over the non-NA samples for each gene. Approach 2 was to use SoftImpute that uses 
a low dimensional structure representation to impute the values.

- [Literature Review](literature_review.html)
- [CorShrink vs corpcor vs GLASSO : Approach 1](corshrink_corpcor_glasso.html)
- [CorShrink vs corpcor vs GLASSO : Approach 2](CorShrink_softimpute.html)
- [CorShrink vs thresholding estimators](thresholding_Corshrink.html)
- [CorShrink vs PDSCE](PDSCE_vs_CorShrink.html)
- [CorShrink vs Sparse Cov Estimation (SPCOV)](spcov_CorShrink.html)
- [CorShrink dsc simulation data: Diagonal](corshrink_dsc.html)
- [Corshrink dsc simulation data: Toeplitz](corshrink_dsc_toeplitz.html)
- [CorShrink dsc simulation data: Graph](corshrink_dsc_graph.html)

### CorShrink and tissues/CountClust clusters on GTEx data

[Dey et al 2017](http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1006599)
obtained clusters of GTEx tissue samples from the gene expression data using a Grade of Membership approach. Here we try to relate the clusters obtained there and the genes that were found to drive these clusters with their tissue-tissue correlation structure patterns. Also of 
interest is how this correlation structure depends on the expression patterns of the genes.

- [CorShrink demo on one gene in GTEx](one_gene_correlation.html)
- [Genewide vs Tissue wide CorShrink](genewide_vs_tissuewide_corshrink.html)
- [tissue wide CorShrink of top genes driving GTEx Countclust clusters](gtex_corshrink.html)
- [gene wide CorShrink of top genes driving GTEx CountClust clusters](gtex_genewide_CorShrink.html)
- [Curious case of high correlation in Brain](corshrink_brain_func.html)
- [GSEA of CorShrink correlation patterns on brain tissues ](gsea_corshrink_corpcor.html)
- [GSEA of CorShrink correlation patterns on all tissues ](gsea_corshrink_corpcor_2.html)
- [Enrichment of tissue subsets based on correlation design](correlation_analysis_subset_enrichment.html)
- [Pathway Grouping and Heatmap Representation](pathway_grouping.html)

We dig deeper into the enrichment analysis 

- Tissue wide CorShrink Enrichment Analysis
    - [Enrichment analysis of high correlation genes in Arteries](enrichment_arteries.html)
    - [Enrichment analysis of high correlation genes inside Brain](enrichment_inside_brain.html)
    - [Enrichment analysis of high correlation genes in non brain](enrichment_non_brain.html)
    - [Enrichment analysis of high correlation in Brain vs other tissues](enrichment_brain_other_tissues.html)
    - [Enrichment analysis of high correlation in Brain vs non brain](enrichment_brain_nonbrain.html)
    - [Enrichment analysis of high correlation in Blood vs Spleen](enrichment_blood_spleen.html)
    - [Enrichment analysis of high correlation in Breast and Adipose](enrichment_breast_adipose.html)
    - [Enrichment analysis of high correlation in Esophagus and Colon](enrichment_esophagus_colon.html)
    - [Enrichment analysis of high correlation in Heart vs other tissues](enrichment_heart_nonheart.html)
    - [Enrichment analysis of high correlation in Skin vs other tissues](enrichment_skin_nonskin.html)
    - [Enrichment analysis of high correlation in Stomach and Pancreas](enrichment_stomach_pancreas.html)
    - [Enrichment analysis of high correlation in Pituitary and Testis](enrichment_pituitary_testis.html)
    - [Enrichment analysis of high,low correlation genes in Fibroblasts and Lymphoblasts](enrichment_fibroblast_lymphoblast.html)
    



- Gene wide CorShrink Enrichment Analysis
    - [Enrichment analysis inside brain](enrichment_inside_brain_2.html)
    - [Enrichment analysis in non brain](enrichment_non_brain.html)
    - [Enrichment analysis in brain and non brain](enrichment_brain_nonbrain_2.html)
    - [Enrichment analysis in Breast and Adipose](enrichment_breast_adipose_2.html)
    - [Enrichment analysis in Skin and Non skin](enrichment_skin_nonskin_2.html)
    - [Enrichment analysis in Stomach and Pancreas](enrichment_stomach_pancreas_2.html)
    - [Enrichment analysis in Arteries](enrichment_arteries_2.html)
    - [Enrichment analysis in Blood and Spleen](enrichment_blood_spleen_2.html)
    - [Enrichment analysis of high,low correlation genes in Fibroblasts and Lymphoblasts](enrichment_fibroblast_lymphocyte_2.html)
    


### Expression vs CorShrink

In this section, we check how the reads expression of the genes in the GTEx data are connected to the CorShrink correlations and how small sample size can indeed produce big correlation values.

- [Reads expression and CorShrink](reads_expression_CorShrink.html)
- [Reads data for Muscle and Heart genes that show low overall correlation](muscle_heart_genes_reads_expression.html)


### Systemic effects analysis in GTEx

CorShrink identifies genes that show high correlation across all tissues, 
both brain and non brain. In this section, we try to dig deeper into learning
more about systemic effects. 

The idea is to compute the gene and tissue specific correlation across samples with respect to sample meta-characteristics such as age, sex, time of death, body temperature etc. We then check if there are genes that show significant effect in regression coefficient across most tissues an whether they can be traced back to the pathways we obtained from tissue-tissue correlations in CorShrink.

First we used a quantile based pathway enrichment analysis of the genes based on the effects of the genes on age, sex, death time and BMI.

- [BMI systemic effects analysis](bmi_compare_study.html)
- [Circadian rhythm/clock trajectories](circadian_rhythm_analysis.html)
- [Circadian rhythm/clock effects analysis](death_time_comparison.html)
- [Age specific systemic analysis](age_comparison_study.html)
- [Sex specific systemic analysis](sex_comparison_study.html)
- [Sex specific systemic analysis - part 2](sex_comparison_study_2.html)

We then applied adaptive shrinkage across tissues for each gene and then compared the
number of tissues for which a given lfsr threshold is exceeded by a gene. This is a 
preliminary analysis before the more formal MASH analysis.

 - [BMI systemic effects study with tissue wide adaptive shrinkage](bmi_compare_study_ash.html)
 - [Age systemic effects study with tissue wide adaptive shrinkage](age_compare_study_ash.html)
 - [Sex systemic effects study with tissue wide adaptive shrinkage](sex_compare_study_ash.html)
 - [Circadian systemic effects study with tissue wide adaptive shrinkage](circadian_compare_study_ash.html)

We additionally observed how the expression trjectories for Nicotine pathways look and whether they cluster by individuals clearly.

- [Is there visible effect of Nicotine pathways](nicotine_study.html)

We also looked at genes that show trends with respect to body temperature. But we did not find anything interesting.

- [Is there a trend in expression of genes with respect to body temperature](heat_effects_analysis.html)

p53 pathway is an extremely important pathway for aging. So, we took a peak into the genes in this pathway and how they look like in this data.

- [p53 and p63 gene studies](p53_p63_analysis.html)

We also studied whether we can use detect some differences in gene expression between two causes of death - say Suicide and Natural Death, or Accident and Natural Death. We present the analyses corresponding to that below.

- [Suicide vs Natural Death genetic effects](suicide_compare_study_ash.html)
- [Accident vs Natural Death genetic effects](accident_compare_study_ash.html)

We also checked between anoxia and heart disease and anoxia and cerebrovascular problems.
The goal was to check if there are certain genes expressed more in one tissue in anoxia compared to heart disease and vice versa. Same for anoxia and cva.

 - [Anoxia vs Cardiac disease](anoxia_v_heart_compare_study.html)
 - [Anoxia vs CVA - cerbro vascular ](anoxia_v_cva_compare_study.html)
 
 We performed FLASH to find factors and factor loadings to get an idea about which tissues are highly loaded in a factor and 
 which genes are driving that factor from the rest.
 
  - [FLASH systemic effects study: age](flash_age_systemic.html)
  - [FLASH systemic effects study: sex](flash_sex_systemic.html)
  - [FLASH systemic effects study: circadian](flash_circadian_systemic.html)
  - [FLASH systemic effects study: suicide/natural death](flash_suicide_systemic.html)
  - [FLASH systemic effects study: BMI](flash_bmi_systemic.html)
  
We followed up this analysis by the MASH model, which is more adapted to performing analyses such as this. We report below the results of MASH for the canonical correlations for Sex, Age, Death time and BMI.

  - [MASH systemic effects study (canonical): age](mash_age_systemic.html)
  - [MASH systemic effects study (canonical): sex](mash_sex_systemic.html)
  - [MASH systemic effects study (canonical): circadian](mash_circadian_systemic.html)
  - [MASH systemic effects study (canonical): BMI](mash_bmi_systemic.html)

We found that the extreme deconvolution matrices were not capturing the details of the tissue level expression. We present an example study for Sex.

  - [MASH systemic effects study (extr deconv + canonical): age](mash_sex_systemic_2.html)
  
We added few additional prior cov matrices to the MASH input. Those included all 1s in brain and 0 everywhere else, all 1s in the two adipose tissues and 0 everywhere else, all 1s in the cerebellum and cerebellar hemisphere and 0 everywhere else, all 1s in the heart tissues and 0 everywhere else, all 1s in the artery tissues and 0 everywhere else, all 1s in the skin tissues and 0 everywhere else.

The results are as follows 

  - [MASH systemic effects study (tissue prior case 1): age](mash_age_systemic_pool.html)
  - [MASH systemic effects study (tissue prior case 1): sex](mash_sex_systemic_pool.html)
  - [MASH systemic effects study (tissue prior case 1): death time](mash_circadian_systemic_pool.html)
  - [MASH systemic effects study (tissue prior case 1): BMI](mash_bmi_systemic_pool.html)
  
We also next added two new correlation structures for each multiple tissue block cov matrix (one with all correlations equal to 0.5 and the other with all correlations equal to 1). Also we added a separate non-brain specific cov matrix to the mix as well.

 -  [MASH systemic effects study (tissue prior case 2): age](mash_age_systemic_pool_2.html)
  - [MASH systemic effects study (tissue prior case 2): sex](mash_sex_systemic_pool_2.html)
  - [MASH systemic effects study (tissue prior case 2): death time](mash_circadian_systemic_pool_2.html)
  - [MASH systemic effects study (tissue prior case 2): BMI](mash_bmi_systemic_pool_2.html)
  
We tried to look into the sinusoidal patterns of expression of the liver specific genes and the circadian genes in general.

 - [sinusoidal patterns of gene expression in Liver](cellcycler_patterns.html)
 - [faulty cellcycleR application on liver expression data](cellcycler_on_circadian.html)

 
### Dimension Reduction and CorShrink

In the GTEx data, CorShrink was previously applied one gene at a time on its tissue-tissue correlation matrix. However, one would assume that there will be sharing of information among genes and we can pool in this shared information while performing shrinkage to get a low dimensionla structyre of the correlation matrix together with shrinkage. To answer that question, we perform PCA, PMD and FLASH on the vectorized CorShrink matrices across genes. For the latter (FLASH), we also compare between FLASH on vectorized CorShrink and sample correlation matrices.

- [Principal Component Analysis on CorShrink matrix](corshrink_pca.html)
- [Penalized Matrix Decomposition on CorShrink matrix](corshrink_pmd.html)
- [FLASH on CorShrink matrix](corshrink_flash.html)
- [FLASH on sample correlation matrix](cor_flash.html)
- [Correlation matrix estimation from FLASH imputed data](flash_imputation_CorShrink.html)




